\mainmatter

# Context

```{r echo=FALSE}
knitr::asis_output(if (knitr:::is_latex_output()) {
"\\begin{chapabstract}{Abstract:}
Complex diseases arise from a mixture of genetic and environmental factors. In this thesis I focus on the identification of the former. This is done via genome-wide association studies (GWAS), when univariate association is studied, and genome-wide association interaction studies, when interactions between genetic factors are considered (GWAIS). Both experimental settings involve the genotyping of hundreds of thousands of SNPs over thousands of individuals, both with the disease and from the healthy population. Yet, they still present some challenges, namely low statistical power, difficult interpretation, andarbitrary choices at multiple points of the study. Approaching the problem from a network perspective might help tackling these issues. In essence, this involves considering both the individual association with the disease of each gene and its biological context, modelled as a network built from known pairwise interactions. In this thesis I focus on how network-based methods can boost biomarker discovery and interpretability on GWAS and GWAIS. I apply these methods to two complex diseases: familial breast cancer and inflammatory bowel disease.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}

\\end{chapabstract}"
})
```

## The common disease/common variant framework

Complex diseases are those caused by a mixture of genetic, environmental and lifestyle factors. The object of study of this thesis are the methodologies to identify such genetic factors. This is of paramount importance for disease prevention, understanding the etiology of diseases, and providing better treatments. 

The genetic architecture of a trait includes the variants that contribute to the risk, as well as their allelic frequencies, effect sizes, and their genetic mode of action (e.g. dominant or recessive). From this point of view, complex diseases are easier to understand in contrast with Mendelian ones. The latter are caused by a single locus with a strong effect, and hence follow the Mendelian rules of inheritance. In essence, and barring considerations on reduced penetrance, whether an individual will develop a Mendelian disease or not depends exclusively on the two alleles at that particular locus, and their genetic mode of action. By contrast, the genetic architecture of complex is modeled by the *liability-threshold model*, an extension to binary traits of the infinitesimal model used to describe the genetics of continuous phenotypes like height. Under the infinitesimal model, a continuous trait is shaped by many Mendelian alleles, each of them with a small contribution to the trait [@barton_infinitesimal_2017]. Specifically, it computes a score for an individual based on the contribution of each of the alleles in the genetic architecture, plus the contribution of the environment. The liability-threshold model computes an equivalent latent score and, if it takes a value above a given threshold, disease will ensue.

(ref:fig-size-freq-caption) Relationship between the minor allele frequency of an allele, and its effect size. As a consequence of evolutionary forces, highly deleterious alleles which cause Mendelian diseases are infrequent, while alleles with small effects that increase slightly the susceptibility to a complex disease are more widespread. Figure adapted from @bush_chapter_2012, licensed under CC BY.

```{r size-freq, echo=FALSE, fig.cap='(ref:fig-size-freq-caption)'}
knitr::include_graphics('fig/dl_frequency_vs_effect_size.png', dpi = NA)
```

Because the risk alleles have such small effect sizes, they are not under strong purifying selection (Figure \@ref(fig:size-freq)). This allows them to be common (>1-5% of the population), unlike the mutations causing Mendelian diseases, which are rare, as they strongly decrease the fitness of the individual [@manolio_finding_2009]. In consequence, the study of the genetics of complex diseases relies on the *common disease, common variant* hypothesis: common diseases are partly attributable to allelic variants present in more than 1-5% of the population, which cause, by themselves or in combinations, small increments in risk (1.1 - 1.5-fold). However, another consequence of this limited effect size is that only weak associations between causal variants and phenotypes can be expected. Again, this notion is radically different from Mendelian diseases, where every carrier of the risk allele develops the disease under complete penetrance. Therefore, the study of the genetics of complex diseases requires the identification of many risk variants among the 88 millions known ones [@the_1000_genomes_project_consortium_global_2015]. 

The most common source of genetic variation in humans are single base-pair changes in the DNA sequence, called single-nucleotide polymorphisms or SNPs [@the_1000_genomes_project_consortium_global_2015]. They usually involve two alleles, meaning that in a population there are two possible base-pairs for a genetic position. SNPs are characterized by their minor-allele frequency, that is, the frequency of the least common allele in the population. Although in this thesis I focused my work on SNPs, other forms of genetic variation exist. These are the structural variants, which involve variation in the structure and the quantitative arrangement of the chromosomes [@spielmann_structural_2018]. Copy number variants (CNVs), a type of structural variants, consist on a repeated segment of the genome, where the specific number of repeats changes from person to person. CNVs are relevant for human health. For instance a neurological disorder known as Huntington's disease ensues when a specific tri-nucleotide in the huntingtin gene is repeated more than 36 times [@macdonald_novel_1993].

## Genome-wide association studies {#intro-gwas}

Genome-wide association studies (GWAS) are experiments that explore large cohorts, systematically surveying both a high number of genetic variants, and phenotypes [@bush_chapter_2012]. Their goal is to find associations between genotypes and the studied phenotype. These associations might lead to earlier diagnoses, choices of treatment appropriate for a patient's genetic background, and better understanding of the etiology of the disease. For that purpose, a classical GWAS involves a statistical test of association between each variant and the phenotype. Often that statistical test is a logistic regression, which allows using additional variables that might act as confounders. For instance, for a SNP $i$:

\begin{equation}
logit \left(p_i\right) = \alpha + \beta_i g_i + \gamma X
(\#eq:gwas-logreg)
\end{equation} 

where $g_i$ is the vector of genotypes at SNP $i$, $\beta_i$ is the coefficient, and $X$ and $\gamma$ are respectively the matrix of covariates and the vector with the covariates' coefficients. A statistical test can be conducted on the value of $\beta$ by transforming it into a Z-score. In order to evaluate the significance of the associated P-value, an appropriate threshold which accounts for multiple testing is needed. Often, that P-value is chosen by setting the family-wise error rate to 0.05 i.e. $0.05/\text{\# SNPs}$ (e.g. 10^-7^ if 500 000 SNPs are tested). Then, genome-wide significant SNPs undergo a follow-up study on an independent cohort.

The 1000 Genome Projects catalogued 84.7 million SNPs across multiple human populations [@the_1000_genomes_project_consortium_global_2015]. However GWAS do not need to survey all of them to obtain a representative view of the genome. Instead, they exploit the correlations between the variants consequence of the genetic history, a phenomenon known as linkage disequilibrium (LD). Thanks to LD, GWAS survey the whole genome in an inexpensive fashion using SNP arrays that contain only a small fraction of the known variants [@visscher_10_2017]. In consequence, the SNPs associated with a disease are likely not the causal ones, but SNPs close to, and in LD with the causal variants.

Since the first GWAS in the late 2000s, more than 5 600 studies have shed light into the genetics of complex traits, identifying more than 70 000 variant-trait associations [@buniello_nhgri-ebi_2019]. From this community effort we took a few lessons about the architecture of complex traits. First, GWAS confirmed the infinitesimal model in all studied complex traits, whose variance can only be explained but by many loci with small effect sizes [@visscher_10_2017]. Such explanatory variants tend to be located in chromatin that is open and expressed in the tissues relevant for the disease [@boyle_expanded_2017]. Also, even genes functionally implicated in the disease explain a small fraction of the trait variance [@boyle_expanded_2017]. In fact GWAS has revealed widespread pleiotropy, as the same genomic regions are often found in association to multiple traits [@visscher_10_2017]. This has profound implications, showing how interrelated different biomolecules are. In fact, holistic models that add nuances to the infinitesimal model have been suggested, like the *omnigenic model* [@boyle_expanded_2017]. This model postulates that only a few core genes are directly implicated in the disease, and alterations on them have a strong effect. But alterations in many other, unrelated, genes can also lead to disease as they propagate through biological networks to affect the functionality of those core genes. A more complete view of biological networks and disease is available in Chapter \@ref(genesis).

### Challenges {#intro-gwas-challenges}

The discovery of the genetic basis of complex diseases is hindered by several intrincacies in both the GWAS setting and the underlying biology. That is why, despite its success in identifying associated variants, all of them together do not explain the whole genetic variance of the disease. Possible explanations include the involvement of a large number of common variants with small effects; of rare variants with large effects not covered in GWAS; of epistasis; and of unaccounted gene-environment interactions [@manolio_finding_2009]. My thesis involved developing methodologies that tackle these challenges, which I present below.

#### Low statistical power

Due to the low effects sizes implied by the common disease / common variant hypothesis, GWAS would require very large sample sizes. Yet, for practical limitations, they have traditionally remained in the thousands, with the latest studies raising them up to hundreds of thousands [@sudlow_uk_2015]. However, the number of biomarkers required to scan the whole genome is even larger, from hundreds of thousands to millions. In consequence, GWAS is conducted in an ultradimensional setting which, jointly with the small effect sizes, leads to low statistical power [@wray_pitfalls_2013;@visscher_heritability_2008;@button_power_2013]. Statistical power also takes a hit due to the partial correlations between the tests, as multiple test correction procedures like Bonferroni consider the statistical tests independent, and hence overcorrect [@wang_statistical_2018]. Statistical power can be further reduced when the causal SNPs are in weak LD with the closest genotyped variants, for instance if they are rare, and when the phenotype is heterogeneous [@visscher_10_2017], as is common in complex diseases. This has several implications. First, that by definition we have a small chance of discovering effects that are true. Second, it raises the probability of a discovery to be a false positive. Third, when an underpowered study discovers a true effect it is more likely to overestimate its effect size. One practical consequence of this low power in GWAS is the difficulty to reproduce results [@visscher_heritability_2008].

#### Choice of encoding {#encoding}

The most commonly used association tests in GWAS require making assumptions on the mode of action of the SNPs (dominance, codominance, etc.). For instance, the logistic regression presented in Equation \@ref(eq:gwas-logreg) needs a single number that represents the two alleles of each individual. Converting the genotype into such a number is known as encoding a genotype. There are several such encodings, and its choice have implications on study: choosing an encoding which diverges from how the SNP acts will reduce the statistical power [@romagnoni_comparative_2019]. A common one is the additive encoding, which assumes that the minor allele is responsible for the phenotype, proportionally to the number of copies; hence, the major allele in homozygosis is represented by a 0, the heterozygous genotype by a 1, and the minor allele in homozygosis by a 2. In Appendix \@ref(block-hsic-lasso) we explore a feature selection algorithm which, when applied to GWAS, makes no assumptions on the mode of action of the SNP.

#### Estimating individual risk

Once the genetic architecture is fully understood, it will be possible to estimate the full genetic component of a patient. This involves moving moving from population-level associations, to an individual assessment [@wray_pitfalls_2013]. Conventionally these are done through polygenic risk scores, which consist on a linear combination of genotypes from the associated loci in a patient, weighted by their effect sizes. However, their utility is far from clinical applications [@visscher_10_2017]. In this thesis, I apply machine learning algorithms for sample classification in Chapter \@ref(genesis) (Section \@ref(results-benchmark)) and in Appendix \@ref(block-hsic-lasso). Also, it is worth stressing than even when the whole genetic architecture of a disease is uncovered, the $R^2$ of a linear predictor would be upper bounded by the heritability ($h^2$). A complete prediction will require fully understanding epistasis (Section \@ref(gwais)), environmental, and the interaction between environment and genetics.

#### Population structure {#pop-structure}

As explained before, the GWAS exploits LD to avoid genotyping all known variants. However, these correlations between SNPs depend on the evolutionary history of each sample, and hence are population specific. Therefore, GWAS designs must account for samples with different ancestries or, in other words, capture the population structure in the data. Failure to do so might lead to overestimating allelic and genotypic frequencies, reducing statistical power and producing spurious associations [@wang_statistical_2018]. Population structure can be captured by the principal components of the genotype matrix, and hence they are often used to account for it [@price_principal_2006]. For instance, a logistic regression using the main principal components as covariates is often used to obtain measures of association at the SNP level correcting by potentially confounding population structure [@michailidou_genome-wide_2015;@michailidou_association_2017;@ellinghaus_analysis_2016].

Another consideration is that most GWAS have been carried out on populations of European ancestry, which have a reduced variability in comparison to other human populations [@visscher_heritability_2008]. In fact, studies on non-European populations have yielded a big number of new, intriguing variants.

#### Interpretability {#intro-interpretability}

By design, a genotyped SNP acts as a tag for the region in the genome in high LD with it. Hence, even if that tag SNP shows statistical association with a disease, fine-mapping studies are needed to pinpoint the specific SNP that is involved in the susceptibility to it. A frequent strategy is identifying the genes under the influence of that genomic region [@lehne2011snps;@Jorgenson], as those are considered the functional unit of inheritance. For instance, a SNP might affect a gene by provoking an amino acid change in the protein product, altering its gene expression or its splicing. Yet, mapping genomic regions to the genes they might impact is not trivial. In the literature, we find three ways of of doing so. The first one is the physical mapping, which maps that region to the genes that whose genomic coordinates overlap with it. Often the gene boundaries are expanded by a fixed number of kilobases, as SNPs in promoters or nearby enhancers can affect gene expression [@segre_common_2010]. Nonetheless, physical mapping can be ambiguous due to the overlap between genes (Section \@ref(results:consensus)). The second SNP-gene mapping is through gene expression regulation, when SNPs in the associated genomic region are eQTLs of a gene. eQTLs are loci that are significantly associated to the expression of a gene. In this regard, the GTEx project [@gtex_consortium_genetic_2017], a project that genotypes multiple human subjects, and extracts expression information from their different tissues, is a useful source of tissue-specific eQTL. Gene expression mapping is not exempt of the overlap problem that occurs in the physical mapping. Solutions which consider LD patterns and the association across the whole genomic region have been proposed [@liu_abundant_2019]. The third SNP-gene mapping is based on the 3D structure of the genome, which causes distant genomic regions to be close in space [@spielmann_structural_2018]. In this situation, SNPs in a genomic region are associated to genes in the neighboring region.

Once a set of SNPs is reliable mapped to a gene, multiple tests extist to aggregate their results into a gene statistic, that tests the association of the gene as a whole with the disease. Aggregation of such SNP statistics into gene statistics is likely to increase the statistical power when dealing with complex diseases [@wu2010novel].

## Epistasis {#intro-epistasis}

Epistasis is the phenomenon where the effect of one locus on the phenotype depends on the state of one or more additional loci. It has two variants: biological and statistical [@moore_traversing_2005]. Biological epistasis refers to the physical interaction occurring between the loci, for instance via their protein products. It is possibly a consequence of the redundancy of biological mechanisms, which requires altering all redundant biological mechanisms to alter a biological function [@niel_survey_2015]. Multiple cases of biological epistasis contributing to phenotypes in model organisms have been reported in the literature [@required]. Statistical epistasis, by contrast, is the observation that the association between one locus and the phenotype changes across the level of the other locus. In essence, statistical epistasis refers to the biological epistasis detectable at the population level. In this regard, links between epistasis and complex diseases like Alzheimer's disease [@combarros_epistasis_2009], inflammatory bowel disease [@cho_identification_1998] and hypertension [@kimura_multilocus_2012] have been found. As in this thesis I worked exclusively in the detection of the latter kind, for brevity's sake I will refer to it simply as *epistasis*.

Despite the links between epistasis and complex disease mentioned above, estimating the magnitude of its contribution to complex diseases in humans is hard [@gusareva_practical_2014]. Nonetheless, @zuk_mystery_2012 proposed that incorrectly accounting for epistasis might be behind the so-called *missing heritability* of complex traits. Additionally, studies of traits model organisms suggest that epistasis plays a key role of their genetic architecture [@required]. This motivates further studying the involvement of epistasis in complex diseases.

## Genome-wide association interaction studies {#gwais}

Genome-wide association interaction studies (GWAIS) share the experimental design with GWAS, but focus on the detection of epistasic associations. As opposed to GWAS, no standard GWAIS protocol exist yet, although some general recommendations have been issued [@gusareva_practical_2014; @ritchie_search_2018].

Due to their similar experimental design, GWAIS and GWAS share a number of challenges (Chapter \@ref(intro-gwas-challenges)). Nonetheless, such problems are often aggravated in GWAIS. For instance, a larger number of tests implies a further reduction of the statistical power. Interpretation is also more complicated, as two or more SNPs need to be mapped to their respective genes [@gusareva_practical_2014]. Nonetheless, the identification of gene-gene interactions is crucial to understand the functional basis of the association: if two interacting SNPs alter the same gene, and both alterations are required to modify its function, if they affect different genes from the same pathway, from two different pathways that cross-talk, etc. However it presents two additional considerations, related to the multiple genetic scenarios in which epistasis can occur. For instance [@li_complete_2000] estimated that there are 50 different fully penetrant disease models involving two loci in epistasis.

First, in any epistasis related study, a number of arbitrary choices must be made, like the order of the explored interactions, whether to pre-filter the data according to function or detectable main effects, or the genetic encoding (see Section \@ref(encoding)) [@romagnoni_comparative_2019]. For instance, fourth order epistasis involves four different loci jointly contributing to a phenotype. Nonetheless, as the number of interactions grows exponentially with the epistasis order, most methods and studies focus on second order epistasis. Second, epistasis introduces the challenge of quantifying and assessing the significance of an statistical interaction. Multiple strategies to detect epistasis have been proposed, from logistic regression with an interaction term to deep learning. [@niel_survey_2015] reviews the main families of epistasis detection strategies. In general, this diversity comes from tackling different aspects of the computational and statistical issues that arise from the large number of potential interactions. For instance, logistic regression with an interaction term is an exhaustive method with a parallelized implementation [@chang_second-generation_2015], but makes strong assumptions on the underlying relationship between the genotype and the phenotype. Hence, statistical power is compromised when that model is inaccurate. On the other hand, MDR [@moore_flexible_2006] is model-free and exhaustive, but is limited to case-control phenotypes and the inability to compute P-values analytically makes it slower. In Chapter \@ref(epi-network) we examine different epistasis detection methods.

## Diseases studied in this thesis

The bulk of my work in this thesis revolved around two complex diseases: breast cancer and inflammatory bowel disease.

### Breast cancer

Cancer is the name of a collection of related diseases. Specifically, all cancers undergo an uncontrolled proliferation of the patient cells, which spread into surrounding tissues. In a normal organism, cells grow and divide to maintain the tissue. As cells grow old, or accumulate too much damage, they undergo cell death and new cells will take their place. However, in cancer, this orderly process breaks down. Cells refuse to die when they get old, or accumulate damage. New cells are formed even if they are not needed. In consequence they form purpose-less growths called tumors.

This abnormal behavior occurs as consequence of the alteration of crucial genes. These alterations can be inherited from our parents, or acquired during our lifetime, due to replication errors or exposure to DNA-damaging substances. As with any other phenotypic trait, the likelihood of developing cancer will be determined by the interplay between our genetic background and the environment: genetic backgrounds may favor or hinder the acquisition of mutations, and so do environmental factors.

Breast cancer occurs when breast cells undergo this uncontrolled proliferation. In most of the cases they begin in the ducts that carry the milk to the nipple. However the tumor can originate in other tissues, mainly the milk-producing gland.

Breast cancer is the second most commonly diagnosed cancer among women, after non-melanoma skin cancer. It is also the second leading cause of cancer deaths after lung cancer. It is mostly a women's disease: only about 1% of the diagnosed cases are in men. Among the most important risk factors for breast cancer we can highlight age, family history, reproductive history, usage of oral contraceptives and exposure to radiation. Most breast cancers occur after age 50.

Breast cancer is a very heterogeneous disease: while all the tumors appear in the same organ, the tissue where they originate, the molecular mechanism involved, the response to therapy, etc. vastly differ. In general, clinical decisions are based on the expression of 3 molecular markers: the expression of the endocrine receptors for estrogen and progesterone (ER and PgR, respectively) and the expression of the HER2 gene. The proteins these three genes code for are targets for chemotherapy. Based on the results, we distinguish three main breast cancer subtypes: hormone receptor positive, HER2 positive and triple negative.

- Hormone receptor positive: Hormone receptor positive tumors include the tumors expressing ER and/or PR, which respectively depend on estrogen and/or progesterone to grow. They happen mostly in postmenopausal women. HR+/HER2- also known as LuminalA are the majority of breast cancers (60-75%) and they present the best prognosis.
- HER2 positive: HER2+ tumors depend on the protein HER2/neu (human epidermal growth factor receptor 2) to proliferate, which they over-express. HR+/HER2+ (also known as LuminalB) constitutes 10% of the cases, while HR-/HER2 (also known as HER2-enriched) involves 5% of them. There are a couple of very effective drugs against it.
- Triple-negative: Triple-negative tumors lack the expression of all three of ER, PgR and HER2. These patients present a worse prognosis than the rest, due to the aggressiveness of the tumor and the lack of a clear molecular target. Still, the main treatment is chemotherapy.

In the mid-19th century a French medical doctor, Pierre Paul Broca, reported for the first time a case of familial breast cancer [@nielsen_hereditary_2016]. Indeed, his wife acquired breast cancer, as many women in her family had for, at least, 4 generations. Cases of familiar breast cancer usually occur in women younger than 50 years, and bilateral primary breast tumors are frequent. Epidemiological studies later quantified the relative risk conferred by a the presence of multiple breast cancers in the family at 2.7. Moreover, they exhibit a higher likelihood of acquiring triple-negative breast cancer.

It wasn't until the late 20th century that two genes involved in DNA repair, BRCA1 and BRCA2, were associated with hereditary breast and ovarian cancer (HBOC). Some mutations in these genes increase the risk of developing breast cancer, giving respectively a 57–65% or 45–55% risk of developing breast cancer by age 70 among women. For that reason, BRCA1 and BRCA2 mutations are rare in most populations (1 of 400). 

HBOC follows an autosomal dominant inheritance pattern. While approximately 5–10% of all patients with breast cancer exhibit a monogenic predisposition to breast and ovarian cancer, only about 25% of them harbor BRCA1/2 mutations. An additional 23 genes have been associated with familial breast and/or ovarian cancer (Table \@ref(tab:hboc-genes)).

Nearly all known HBOC susceptibility genes encode tumor suppressors that participate in genome stability pathways (homologous recombination repair, replication fork stability, transcription–replication collisions, mismatch repair, and DNA damage signaling, checkpoints and cell death; see Appendix \@ref(hboc-genes) for more information).

#### The GENESIS dataset {#genesis-dataset}

The GENE Sisters (GENESIS) study was designed to investigate risk
factors for familial breast cancer in the French population
[@sinilnikova_genesis:_2016]. Index cases are patients with infiltrating
mammary or ductal adenocarcinoma, who had a sister with breast cancer,
and who have been tested negative for BRCA1 and BRCA2 pathogenic
variants. Controls are unaffected colleagues and/or friends of the
cases, born around the year of birth of the corresponding case ($\pm 3$
years). We focused on the 2 577 samples of European ancestry, of which
1 279 are controls and 1 298 are cases. The genotyping was performed
using the iCOGS array, a custom Illumina array designed to study genetic
susceptibility of hormone-related cancers [@sakoda_turning_2013]. It
contains 211 155 SNPs, including SNPs putatively associated with breast,
ovarian, and prostate cancers, SNPs associated with survival after
diagnosis, and SNPs associated to other cancer-related traits, as well
as functional candidate variants in selected genes and pathways.

### Inflammatory bowel disease

Inflammatory bowel disease (IBD) is a group of complex diseases that, as the name indicates, share a common theme of inflammation of the intestines. The two main subtypes are ulcerative colitis and Chron's disease. Clinically, these two share a lot of the symptoms, mainly intermittent abdominal pain and diarrhea [@liu_genetics_2016]. However, they differ in the specific regions of the digestive tract that get affected, as well as the specific lesions. IBD's incidence worldwide has been continually growing, specially in newly industrialized countries, althouth after decades of growth it has stabilized in North America, Oceania and Europe [@ng_worldwide_2017]. In these latter countries, the prevalence is slightly above 0.3\%. 

The genetic component of IBD was recognized more than a century ago [@ek_history_2014]. However, it was not until the 2001 that the first gene, NOD2, was linked to IBD susceptibility. Ever since, hundreds loci have been associated to IBD as well, in positions related to immune system genes, both innate and adaptive [@loddo_inflammatory_2015;@ellinghaus_analysis_2016;@liu_genetics_2016]. For instance NOD2, and other susceptibility genes like IL23R, and PTPN2, are related to cell signalling in immune cells. However, most of the loci associated to genes are rare variants (< 0.5\%) with large effect sizes. By contrast, several GWAS have identified very common SNPs (20-50\%), with small effect sizes (odds ratio < 1.1) [@liu_genetics_2016], but which do not encode any coding change [@jostins_hostmicrobe_2012]. This raises questions about the underlying biology (see Section \@ref(intro-interpretability)). For a comprehensive view of the genetics of IBD, interested readers can read [@liu_genetics_2016] and [@loddo_inflammatory_2015].

#### The IIBDGC dataset {#iibdgc-dataset}

The International Inflammatory Bowel Disease Genetics Consortium carried out the largest case-control GWAS on ulcerative colitis and Chron's disease to date [@jostins_hostmicrobe_2012]. The dataset contain 66 280 samples, out of which 32 622 are cases and 33 658 are controls. The Immunochip SNP array was used for the genotyping [@cortes_promise_2010], which contains 196 524 polymorphisms, with a special focus on immunogenetics.

## Network view of complex diseases

<!-- motivation (link to challenges, biology as networks) -->

Human biology is notoriously complicated, as sheer numbers demonstrate: to form a 70kg man, 3.0 \texttimes 10^13^ cells [@sender_revised_2016] assemble and interact to produce and maintain 79 organs. To achieve that level of complexity, human cells depend on their genetic material, carefully tuned by epigenetics and enabled by a favorable environment. In terms of genetics, the object of my work, a human diploid genome is 6.4 billion base pairs long, and encodes 44 393 genes, of which 20 444 encode for a protein and 23 949 are RNA genes. DNA, proteins and RNA are in constant interplay with each other, with the metabolites, and with the environment. Proteins physically interact with each other in highly specific ways (protein-protein interactions or PPIs). If such interactions are stable enough, proteins can assemble into large complexes to carry out particular functions. But proteins also interact with DNA to regulate gene expression (transcription factor - DNA interactions). And so on: enzymes interact with their metabolites, hormones with their receptors, the individual with their environment, etc. At a fundamental level, a person and their traits are just the emerging pattern born from the interaction of all these factors. Hence, biology, from ecosystems to molecular biology, cannot be understood if not as an interplay. Mindful of this, and enabled by the omics technologies of the 21st century, researchers have strove to capture and understand these relationships. The goal of systems biology is achieving a global understanding of the complex interplay that drives biology. Among all the possible biological relationships, in this thesis I focused on protein-protein interactions, including protein complexes, as their coverage is larger and their properties better understood. Such interactions are available in databases like HINT [@das_hint_2012], The BioGRID [@oughtred_biogrid_2019] or IntAct [@hermjakob_intact_2004].

Relationships between pairs of entities can be mathematically formalized as a network, which makes them analytically approachable. In such networks, often proteins or genes are the nodes, which are connected by edges in a pairwise fashion when they are related. Although the edges might have directionality, often PPIN are undirected, as the direction of the edges is often unknown (transitory interactions), or inexistent (co-complexes) [@barabasi_network_2011].

The field of systems biology relies on the assumption that the network accurately captures the context it requires to carry out its biological function. We can distinguish three levels of network properties: properties of individual nodes (local), the joint properties of groups of nodes (mesoscale), and the properties over the whole network (global). Indeed, at all three levels biological networks are structured, very differently from random networks [@barabasi_network_2011;@Chaiboonchoe2013].

At the global level, for instance, the degree, which represents the number of edges per node, follows a power-law distribution [@barabasi_network_2011]. This implies that, at the local level, a few genes participate in the majority of the edges. Such nodes are called *hubs*. Importantly, this node property is informative of the gene function: *in utero* essential genes, like knots preventing the network from falling apart, tend to be hubs. An important consequence of the structured degree distribution at the mesoscale level is the emergence of *modules*, subsets of nodes densely interconnected to each other, and sparsely to the rest. Such modules are often associated with biological function towards which all nodes jointly contribute [@mitra_integrative_2013]. 

Another global property of a graph is the distribution of distances between pairs of nodes, that is, how many edges must be passed to travel from one node to another. Such is the notion of path between nodes. Often we are interested in the *shortest* path, for they reveal the fastest way information can flow from one node to another. An examination of the distribution of shortest paths in biological networks, shows that all nodes are close to each other [@barabasi_network_2011]. Such networks are called *small world networks*. This structure makes the flow of information resilient to the removal of nodes or edges [@Chaiboonchoe2013], a cause of biological epistasis [@niel_survey_2015].

Modules are connected groups of nodes which are densely interconnected to each other and sparsely to the rest of the network. A mesoscale property of biological networks is that they have a modular structure i.e. a strong division into modules. Such modules often constitute functional units within the network, where the nodes jointly contribute to a specific function [@mitra_integrative_2013].

### Networks in disease

Examining how these biological networks relate to disease, and the topology around disease genes produces a nuanced approach to disease: cut the knot in the center of the web that keeps it all together, and it all will fall off; cut a bunch of peripheral, less important nodes around it, the outcome might be the same. In other words, there are many ways of producing the same disease [@leiserson_network_2013]. 

The properties of biological networks enumerated above lay down the foundations of the use of biological networks to study disease genes. For instance, they justify the *local hypothesis*, which expects genes involved in a disease to interact with each other [@barabasi_network_2011]. They also justify the disease module hypothesis, which expects genes involved in the same disease to share a module. Experimentally, network propagation experiments highlight the differential topological properties of disease genes and biological networks. Network propagation refers to methods that use all the possible paths in the network to re-rank the genes on it [@cowen_network_2017]. They include heat diffusion, random walk, graph kernels, and even Google's search algorithm. In essence, for these methods, association of a node with a phenotype can be thought of as a volume of water: the more strongly associated, the more voluminous. Generically, each node starts with an initial volume which, iteratively, gets distributes its liquid among its neighbors. Equivalently, that node will receive a share from its neighbors'. The expectation is that truly associated genes will be densely interconnected to each other, forming cycles and modules that will keep the water from diffusing to other, uninvolved genes. After a number of steps the volume of water in each node is re-evaluated, and used to re-prioritize the genes. Using network propagation [@huang_systematic_2018] recently showcased the differential topological properties of disease genes. Across different biological networks, they were able to retrieve disease-related genes with varying success using only a subset of them and random walk with restart.

<!-- TODO double check 
tend to be non-essential genes located in the periphery
-->

Indeed, disease genes exhibit different properties than non-disease genes in disease [@pinero_uncovering_2016; @cai_broker_2010; @furlong_human_2013; @barabasi_network_2011]. For starters, disease genes are not hubs [@cai_broker_2010; @das_hint_2012], but tend to be non-essential genes located in the periphery. This is coherent with a evolutionary mindset, where mutations in essential genes would be highly deleterious, even resulting in embryonic lethality. However, this enrichment is driven mainly by cancer genes [@pinero_uncovering_2016]. Additionally, disease genes tend to be bottlenecks i.e. the sole link between many peripheral genes and the rest of the network [@cai_broker_2010], suggesting disease arise when these vulnerable regions of the network break down. 

These properties, however, do not affect equally all types of disease genes. Many of the aforementioned properties do not extend to genes identified through GWAS [@cai_broker_2010]. However, differences arise even when comparing genes involved in complex diseases with those involved in Mendelian ones.

### Network-guided approaches to disease study 

As exemplified in the previous section, networks can be leveraged on to gain insight of the biology of the disease. Below I summarize several ways of doing that I have worked on during my thesis.

#### High-score subnetwork search {#intro-hs3}

One of the focuses of my thesis was the study of networks where the nodes have been scored by their association to the disease. The scores might come from omics experiments, or from *a priori* knowledge of disease genes. In essence, such methods look for connected subnetworks made of nodes with high scores. Lacking a term broadly agreed upon in the community, I refer to such algorithms as high-score subnetwork search. Although heterogeneous, to some extent all existing approaches are based on the *guilt-by-association principle* (Figure \@ref(fig:network-gwas): nodes nearby other nodes associated to the disease are suspect of being associated themselves, even if their association is non-significant by conventional standards. However, taking only the genes associated with those associated would be prone to false positives, as often networks include edges that are not relevant for the biological problem at hand. Several high-score subnetwork search methods are available in the literature. Essentially, they differ in the considerations they make on what the solution looks like. In Chapter \@ref(genesis) we describe a representative set of these methods, critically discuss their performance at biomarker discovery on the GENESIS dataset, and discuss their shortcomings.

(ref:fig-network-gwas-caption) Overview of the high-score subnetwork search applied to GWAS. First, **(A)** SNP P-values are assigned to genes (e.g. by physical mapping. **(B)** The significance of the phenotype-association for each gene is based on the assigned P-value, which is overlaid on a co-functional network. **(C)** The significance of each gene is boosted by guilt-by-association, resulting in additional candidate genes (e.g., Gene D) for the phenotype **(D)**. Figure and caption adapted from @lee_aragwab_2018, licensed under CC BY 4.0.

```{r network-gwas, echo=FALSE, fig.cap='(ref:fig-network-gwas-caption)'}
knitr::include_graphics('fig/dl_network_gwas.jpg', dpi = NA)
```

<!--
TODO
MOVE TO THE GENESIS PAPER
Nonetheless, these methods are not exempt of problems. For instance, these methods are heavily affected by the *small world* nature of biological networks, which make it relatively easy to find that two associated nodes are close [@pinero_uncovering_2016]. Additionally, when the biological network contains edges that are irrelevant for a particular disease the complexity of the problem is increased, and the signal decreased. Also [@mitra_integrative_2013].
-->

#### Module detection

In the context of disease, densely interconnected subnetworks (modules) of consistently altered nodes might represent the mechanisms that lead to it. Hence, their identification can provide insights into its etiology. In Appendix \@ref(smartas) we apply module detection techniques to a subnetwork of genes with abnormal splicing in cancer.

#### Aggregation of networks

During my research, often I obtained multiple high-scoring subnetworks using different approaches, which provided complementary perspectives of the same disease. In consequence, I was interested in integrating them into a single subnetwork, which I would analyze. In Chapter \@ref(genesis) we discuss a naive way of aggregating subnetworks from different high-score subnetwork search algorithms, were the edges are unweighted. [@wang_similarity_2014] and [@glass_passing_2013] used information theory to integrate find the closest network to a set of input networks. In Chapter \@ref(epi-network), we discuss how to apply such methods to networks coming from different epistasis detection methods, whose edges are weighed by the confidence they exist.

## Contributions

The object of my thesis was the methodological study and application of network methods to GWAS data. In essence, networks contain prior information, which can be traded for statistical astringency in our analysis. In other words, if genes strongly associated to a disease, albeit non-significantly so, are interconnected in an underlying biological network, we are more likely to believe they represent a consistently altered biological mechanism.

In collaboration with Nadine Andrieu and Fabienne Lesueur (Institut Curie), and working closely with Christine Lonjou (Institut Curie), I applied six high-score subnetwork search to the GENESIS dataset (Section \@ref(genesis-dataset)). Our goal was to find new biomarkers for breast cancer susceptibility. My work on GENESIS is explained in Chapter \@ref(genesis). In summary, I applied six different network-based, biomarker discovery methods to the GENESIS dataset. The methods provide a representative view of the high-score subnetwork search field. We performed a methodological comparison and a benchmark of the methods, highlighting their strengths and weaknesses. Finally, we conclude that combining the methods provide a more complete answer than any of the individual solutions. Our network analysis recovers both genes and genomic regions previously found in association with breast cancer susceptibility, as well as new genes. Importantly, all of those genes are tied in a subnetwork, providing a rationale on how alterations on those genes might lead to disease.

My first approach to the problem involved working on SConES [@azencott_efficient_2013], one of the high-score subnetwork search method examined in Chapter \@ref(genesis). I give a more detailed explanation of this algorithm, together with my work on it in Chapter \@ref(martini). In summary, I worked on applying SConES to case-control datasets, and strategies to parametrize it. Regarding the former, as SConES implemened exclusively the regression version of SKAT [@wu_rare-variant_2011;@ionita-laza_sequence_2013], I implemented two ways of measuring association between a SNP and a binary phenotype. Regarding the latter, SConES has two parameters, $\eta$ and $\lambda$, which control the sparsity and the inter-connectedness of the selected SNPs, respectively. Originally the parameters that produced the most stable solution were selected. I explored the impact of penalized-likelihood measures like BIC, AIC, and AICc, which score a set of features based on both their sparsity and the accuracy of a linear classifier built on them. The product of my work is published as *martini* [@martini], an R package published in Bioconductor [@bioconductor], a peer-reviewed R repository.

In 2019, when I started working on epistasis detection, we established a collaboration with Kristel Van Steen (University of Liège, Belgium). Specifically, I worked closely with Diane Duroux, a PhD student in her research group. Our goal was to build an epistasis gene network of inflammatory bowel disease. We discuss our efforts in this regard in Chapter \@ref(epi-network). For that purpose, we surveyed suitable epistasis detection methods, applied them to the IIBDGC dataset (Section \@ref(iibdgc-dataset)), and integrated the solution. However, as a previous step we needed to appropriately map epistatic SNP networks to epistatic gene networks. I describe this work in Chapter \@ref(epi-snp2gene). In summary, we examined and evaluated four different mappings (physical, eQTL, chromatin, and the three together).

In collaboration with Makoto Yamada (RIKEN AIP, Japan) and Samuel Kaski (Aalto University, Finland), I developed block HSIC Lasso, a general-purpose non-linear feature selector. The work involved modifying an existing algorithm, HSIC Lasso, to reduce its memory consumption. On top of that, we worked on improving its performance, and on solving numerical issues in edge cases. The algorithm is implemented as the Python package pyHSICLasso, available on both PyPI and GitHub. Then, I characterized the algorithm and applied it to several biological datasets. Crucially, three of the datasets were GWAS, which was a milestone in terms of the number of features block HSIC Lasso can handle. Thanks to this work, we analyzed the impact of considering non-redundancy and non-linear models when selecting SNPs for patient classification. We describe the algorithm and our conclusions in the article *Block HSIC Lasso: model-free biomarker detection for ultra-high dimensional data* published in the proceedings of the ISMB/ECCB 2019 [@climente-gonzalez_block_2019]. The full manuscript is available in Appendix \@ref(block-hsic-lasso). It was made possible by a scholarship from RIKEN AIP, which allowed me to spend 3 months in Makoto Yamada's laboratory in Kyoto.

Additionally, I worked on the involvement of alternative splicing in cancer with Eduardo Eyras (Australian National University), a continuation of my previous research. During 2016 and early 2017, we prepared the answer to reviewers for the article *The functional impact of alternative splicing in cancer*, which was published in Cell Reports in August 2017 [@climente-gonzalez_functional_2017]. The full manuscript is available in Appendix \@ref(smartas). I also explored evidence of epistasis in cancer, looking for mutual exclusivity between alterations in alternative splicing and somatic mutations. We searched for evidence of mutual exclusivity at both the gene and the pathway level. Among others, we found a compelling pattern in GATA3. This finding was published in Cell Reports in March 2018, in the broader article *Systematic analysis of splice-site-creating mutations in cancer* [@jayasinghe_systematic_2018]. The full manuscript is available in Appendix \@ref(scm). In addition, I put compiled the code required for the analyses in [@climente-gonzalez_functional_2017], and created a Python package with a clean interface. The package, *spada*, is available in both PyPI and GitHub. This package searches alterations of alternative splicing, and maps them to functional consequences at the protein level. Specifically, such consequences are the removal/addition of functional modules to the protein (e.g. domains) and the loss or gain of protein-protein interactions. Lastly, I applied *spada* to two leukemia datasets from TARGET, an NIH program that aims to understand the molecular basis of several childhood cancers. This last analysis is still on-going.

Lastly, I carried out my research committed to open, reproducible science. As such, all my projects have an associated, version-controlled, laboratory notebook, which includes as much data as I am allowed to share. All such laboratory notebooks are made out public when the paper is. All the scripts I developed are open source, under permissive MIT license. Specifically, I made an effort to develop project-independent tools, which are useful to anyone which needs similar to the ones I had. These tools are published on GitHub https://github.com/hclimente/gwas-tools.