# Unraveling the genetics of complex diseases: a network approach

## The common disease/common variant framework

Complex diseases are those caused by a mixture of genetic, environmental and lifestyle factors. The object of study of this thesis is the discovery of such genetic factors. Identifying the genetic factors that increase the risk of complex disease are paramount for prevention, understanding the etiology, and better treatment. In this dimension, complex diseases are easier to explain through by contrasting their genetic architecture with that of Mendelian traits. The genetic architecture of a trait is the number of variants that contribute to the risk, as well as their allelic frequencies, effect sizes, and their genetic mode of action (e.g. dominant or recessive). In this regard, Mendelian diseases are caused by a single locus with strong effects, and hence they follow the Mendelian rules of inheritance. In essence, and barring considerations on reduced penetrance, whether an individual will develop a Mendelian disease or not depends exclusively on the two alleles on that particular locus, and how they relate to each other. In contrast, the genetic architecture of complex is modelled by the *liability-threshold model*. The liability-threshold is an extension to binary traits of the infinitesimal model used to describe the genetics of continuous phenotypes like height. Under the infinitesimal model, a trait is shaped by many Mendelian alleles, each of them with a small contribution to the trait. The liability-threshold model computes a risk score for an individual based on the contribution of each of the alleles in the genetic architecture, plus the contribution of the environment. If such risk score surpasses a given threshold, disease will ensue.

Interestingly, because each of the involved alleles have small effect sizes, they are not under strong purifying selection. This allows them to be common (>1-5% of the population), unlike the mutations causing Mendelian diseases, which are rare. In consequence, the study of the genetics of complex diseases relies on the *common disease, common variant* hypothesis: common diseases are partly attributable to allelic variants present in more than 1-5% of the population, which cause, by themselves or in combinations, small increments in risk (1.1 - 1.5-fold). Due to this limited effect size, only partial correlations between causal variants and phenotypes are expected. This notion is radically different from rare diseases associated to rare variants with big effect sizes, where every carrier of the risk allele develops the disease (complete correlation). In fact, this is consistent with population genetic theory: variants that explain a large part of the disease by themselves would reduce the reproductive fitness of the individual and, in consequence, would be infrequent [@manolio_finding_2009]. In summary, the study of the genetics of complex diseases requires the identification of a large number of risk variants, among a humbling 686 million known variants in humans [@required]. 

The modern unit of genetic variation are single-nucleotide polymorphisms (SNPs) [@bush_chapter_2012]. SNPs are single base-pair changes in the DNA sequence that occur with high frequency in the human genome. They are, by far, the most common form of genetic variation in humans. SNPs usually involve two alleles, meaning that in a population there are two possible base-pairs for a genetic position. SNPs are characterized by their minor-allele frequency (MAF), that is, the frequency of the least common allele in the population.

<!-- explain epistasis as well -->

## Genome-wide association studies

<!-- ![genetic_studies](../figures/genetic_studies.png) --> 

Genome-wide association studies (GWAS) are experiments where a high number of genetic variants are systematically surveyed on large cohorts [@bush_chapter_2012]. Their goal is to identify genomic regions associated with a disease. In the classical GWAS setting, a statistical test for association is conducted between each variant and the phenotype of interest. Often, that statistical test is a logistic regression, which allows using additional variables that might act as confounders. Then the "best hits", conventionally variants with a genome-wide P-value < 5 \texttimes 10^{-8}^, undergo a follow-up on an independent cohort. Many fields that profit from this ability to survey a sample genome are using GWAS, especially in personalized medicine (e.g. pharmacogenomics). In consequence, GWAS has become a paragon of translational bioinformatics. The ultimate goal is to be able to identify individuals at genetic risk, as well as understand the molecular basis of the disease.

The build 153 of dbSNP compiled over 667 million variants in humans. However GWAS do not need to survey all of them to obtain a representative view of the genome. Instead, GWAS exploits a phenomenon called linkage disequilibrium (LD), consising on correlations between the variants as a consequence of the genetic history. Thanks to LD, GWAS uses SNP arrays to survey only a fraction of them, which act as markers for large genomic regions. In consequence, GWAS allows to survey the whole genome quite cheaply [@visscher_10_2017]. The SNPs associated to the disease can have functional consequences by themselves (e.g. change an amino acid in the protein). However, in the context of genetic studies, SNPs are used as markers of a chromosomic region. In consequence, the SNPs associated with a disease are not necessarily causal: they might just be close and in LD with the causal variant.

Since the first GWAS in the late 2000s, more than 70000 variant-trait associations have been discovered [@buniello_nhgri-ebi_2019], shedding light into the genetics of complex diseases. One finding is that many loci with small effect sizes contribute to shape any complex trait [@visscher_10_2017]. GWAS have also revealed widespread pleiotropy, as variants are often found in association to multiple traits [@visscher_10_2017]. This has profound implications, showing how interrelated different biomolecules are. In fact, [@wray_common_2018]. 

recently holistic models that go beyond the classical polygenic model have been suggested, like the /omnigenic model/ [@boyle_expanded_2017]. The omnigenic model postulates that only a few core genes are directly implicated in the disease, and alterations on them have a strong effect. But alterations in many other, unrelated, genes can also lead to disease as they propagate through biological networks to affect the functionality of those core genes. A more complete view of biological networks and disease is available in Chapter TODO. 

### Challenges {#gwas-challenges}

In the case of height, 80% of the intra-population variability in height is due to genetic factors (h^2 = 0.8) [@visscher_heritability_2008]. Three studies with a total sample size of 63k individuals (14k + 16k + 34k) identified 54 variants that are reliably associated with height. However, they only get to explain 5% of the phenotypic variance [@manolio_finding_2009]. The rest of the estimated genetic component is something alike "dark matter": we know it is there, as we can see its effects, we just cannot see it nor know what it is exactly. This situation is actually common to many complex phenotypes, and has been called the missing heritability problem. There are several candidates out there to explain this dark matter: a larger than expected number of variants with mild effects (difficult to find in underpowered setting of GWAS); rarer variants with strong effects; but also an overestimation of h<sup>2</sup> by unconsidered epigenetics, dominance and epistatic effects; and even an inadequate accounting of shared environment among relatives. Of course, different traits will have different allelic architectures (number of variants, effects, type...). But even then, our knowledge is too limited to really assess it. The enigma of missing heritability is especially relevant when we deal with complex diseases. Missing heritability is a consequence of a few, inter-related challenges, which I describe below. My thesis involved developing methodologies that work around these issues.

#### Low statistical power

Due to the low effects sizes implied by the CD/CV hypothesis, GWAS require large sample sizes, on the thousands. However, the number of biomarkers required to scan the whole genome is even larger, from hundreds of thousands to the millions. In consequence, GWAS is conducted in an ultradimensional setting, which altogether with the small effect sizes, leads to low statistical power [@button_power_2013]. Statistical power can be further reduced if the causal SNPs are in weak LD with the genotyped variants, for instance if they are rare [@visscher_10_2017]. Lastly, also when the phenotype is heterogeneous [@visscher_10_2017]. 

<!-- STILL RELEVANT? Of the main suspects of missing heritability, variants not covered by standard experiment designs are regarded as the most likely cause. Although candidates like structural variants have been proposed, rare SNPs (MAF < 0.05) are the option that gathers more support. They are often not included in the genotyping arrays, while their mild effect sizes (say, odds-ratios of 2-3) make them difficult to find in classical linkage analyses. Increased sample sizes and more exhaustive arrays could potentially help find these variants.-->

This has with several implications. First, it implies by definition a small chance of discovering effects that are true. Second, it raises the probability of a discovery to be a false positive. Third, when an underpowered study discovers a true effect, it is likely that the estimated is exaggerated. One practical consequence of this low power in GWAS studies, for example, is the difficulty to reproduce a GWAS result [@visscher_heritability_2008].

#### Identification of causal variants

Visscher conducted one of the most iconic GWAS to date, to inquire about the heritability of height [@visscher_heritability_2008]. Three studies with a total sample size of ∼63k individuals (14k + 16k + 34k) identified 54 variants that are reliably associated with height. However, there was an alarmingly low overlap between the causal SNPs identified in the different studies. The reasons for this are multiple. First, we are dealing with SNPs that have a very limited effect size (% of variation explained). In consequence, we need sample sizes of the order of tens of thousands to reliably identify one causal SNP. Second, we are conducting the hundreds of thousands of association tests. Thus, we need to be very stringent with the significant threshold in order to minimize type I errors (false positives). Lastly, the number of SNPs in a population is between one and two orders of magnitude above the number of SNPs surveyed in a chip [@wray_pitfalls_2013]. Hence, it is likely that the association we measure between most of the markers and the phenotype is due to the linkage disequilibrium with the true causal SNP(s). This further dilutes the association that we intend to measure.

Another consideration regarding GWAS is that most studies have been carried out on European ancestry populations, which has a reduced variability in comparison to other human populations [@visscher_heritability_2008]. In fact, studies on non-European populations have yielded a big number of new, intriguing variants.

#### Estimating individual risk

Ultimately, our goal is to estimate the full effect of the genetic factors for a patient. This involves moving moving from population-level associations, to an individual assessment [@wray_pitfalls_2013]. However, even in an hypothetical case where we know all the markers involved in a phenotype and their respective effect sizes, the $R^2$ of a linear predictor would be upper bounded by the heritability ($h^2$). This means that, at the patient level, we would only have a probabilistic estimate of their risk.

#### Population structure

The LD structure is population specific. Hence, in GWAS designs must account for samples with different ancestries or, in other words, capture the population structure in the data. Failure to do so might lead to overestimating allelic and genotypic frequencies, reduce statistical power and produce spurious associations [@required]. Population structure can be captured by the principal components of the genotype matrix, and hence they are often used to account for it. For instance, a logistic regression using the main principal components as covariates is often used to obtain measures of association at the SNP level correcting by potentially confounding population structure. Approaches like EIGENSTRAT also use these PCs to detect population structure in the data.

#### Interpretability

Every SNP genotyped in a GWAS represents a whole genomic region, which is in LD with it. Hence, when an association is found between an SNP and a disease, the causal SNPs might be any SNP in LD with the surveyed one. Functional analyses are needed to find out the real causal one. However, if the LD block is large, that's hard.

The problem is exacerbated by the unclear mapping of SNP to a biological function. As the gene is considered the functional unit of inheritance, mapping SNPs to genes is a crucial step. However, that is not a trivial problem either. In the literature, we find three ways of mapping a SNP to a gene. The classical one is the physical mapping, where a SNP is mapped to a gene when their genomic coordinates overlap. Often a certain window around the gene boundaries are also included, to deal with the LD problem, and because SNPs outside the coding region are also known to control the expression of the gene. In this regard, another mapping is through eQTL. Thanks to projects like ??? we find those in different tissues. Lastly, 3D structure.

Genes also overlap. Mention that paper that Makoto sent.

Non-coding SNPs.

## Genome-wide association interaction studies

Genome-wide association interaction studies (GWAIS) share the experimental design with GWAS, but focus on the detection of epistasis.

### Challenges

GWAIS share the same challenges as GWAS enumerated in Section \@ref(gwas-challenges).

#### Definition of epistasis

## Diseases under study

In this thesis I worked in two complex diseases: breast cancer, and inflammatory bowel disease.

### Breast cancer

Cancer is the name of a collection of related diseases. Specifically, all cancers undergo an uncontrolled proliferation of the patient cells, which spread into surrounding tissues. In a normal organism, cells grow and divide to maintain the tissue. As cells grow old, or accumulate too much damage, they undergo cell death and new cells will take their place. However, in cancer, this orderly process breaks down. Cells refuse to die when they get old, or  accumulate damage. New cells are formed even if they are not needed. In consequence they form purpose-less growths called tumors.

This abnormal behavior occurs as consequence of the alteration of crucial genes. These alterations can be inherited from our parents, or acquired during our lifetime, due to replication errors or exposure to DNA-damaging substances. As with any other phenotypic trait, the likelihood of developing cancer will be determined by the interplay between our genetic background and the environment: genetic backgrounds may favor or hinder the acquisition of mutations, and so do environmental factors.

Breast cancer occurs when breast cells undergo this uncontrolled proliferation. In most of the cases they begin in the ducts that carry the milk to the nipple. However the tumor can originate in other tissues, mainly the milk-producing gland.

Breast cancer is the second most commonly diagnosed cancer among women, after non-melanoma skin cancer. It is also the second leading cause of cancer deaths after lung cancer. It is mostly a women's disease: only about 1% of the diagnosed cases are in men. Among the most important risk factors for breast cancer we can highlight age, family history, reproductive history, usage of oral contraceptives and exposure to radiation. Most breast cancers occur after age 50 years.

Breast cancer is a very heterogeneous disease: while all the tumors appear in the same organ, the tissue where they originate, the molecular mechanism involved, the response to therapy, etc. vastly differ. In general, clinical decisions are based on the expression of 3 molecular markers: the expression of the endocrine receptors for estrogen and progesterone (ER and PgR, respectively) and the overexpression of the HER2 gene. The proteins these three genes code for are targets for chemotherapy. Based on the results, we distinguish three main breast cancer subtypes: hormone receptor positive, HER2 positive and triple negative.

- Hormone receptor positive :: Hormone receptor positive tumors include the tumors expressing ER and/or PR, which respectively  depend on estrogen and/or progesterone to grow. They happen mostly in postmenopausal women. HR+/HER2- also known as LuminalA are the majority of breast cancers (60-75%) and they present the best prognosis.
- HER2 positive :: HER2+ tumors depend on the protein HER2/neu (human epidermal growth factor receptor 2) to proliferate, which they over-express. HR+/HER2+ (also known as LuminalB) constitutes 10% of the cases, while HR-/HER2 (also known as HER2-enriched) involves 5% of them. There are a couple of very effective drugs against it.
- Triple-negative :: Triple-negative tumors lack the expression of all three of ER, PgR and HER2. These patients present a worse prognosis than the rest, due to the aggressiveness of the tumor and the lack of a clear molecular target. Still, the main treatment is chemotherapy.

In the mid-19th century a French medical doctor, Pierre Paul Broca, reported for the first time a case of familial breast cancer [@nielsen_hereditary_2016]. Indeed, his wife acquired breast cancer, as many women in her family had for, at least, 4 generations. Cases of familiar breast cancer usually occur in women younger than 50 years, and bilateral primary breast tumors are frequent. Epidemiological studies later quantified the relative risk conferred by a the presence of multiple breast cancers in the family at 2.7. Moreover, they exhibit a higher likelihood of acquiring triple-negative breast cancer.

It wasn't until the late 20th century that two genes involved in DNA repair, BRCA1 and BRCA2, were associated with hereditary breast and ovarian cancer (HBOC). Some mutations in these genes increase the risk of developing breast cancer, giving respectively a 57–65% or 45–55% risk of developing breast cancer by age 70 among women. For that reason, BRCA1 and BRCA2 mutations are rare in most populations (1 of 400). 

HBOC follows an autosomal dominant inheritance pattern. While approximately 5–10% of all patients with breast cancer exhibit a monogenic predisposition to breast and ovarian cancer, only about 25% of them harbor BRCA1/2 mutations. Other 23 genes have been associated with familial breast and/or ovarian cancer (Table \ref{tab:HBOC_genes}, in Appendix A). 

Nearly all known HBOC susceptibility genes encode tumor suppressors that participate in genome stability pathways (homologous recombination repair, replication fork stability, transcription–replication collisions, mismatch repair, and DNA damage signaling, checkpoints and cell death; see Appendix A for more information).

#### The GENESIS dataset

GENESIS (GENE SISter) is a French project that aims to shed some light on familial breast cancer. The index cases are patients with a breast cancer affected sister and no BRCA1/2 mutations [@sinilnikova_genesis:_2016]. The controls are unaffected colleagues and/or friends of the cases. This dataset is specially interesting for us: the heritability of the disease among the cases is not driven by BRCA1/2, but by rarer variants which are enriched in this experimental setting.

GENESIS used the iCOGS genotyping platform [@sakoda_turning_2013]. iCOGS is a custom Illumina array designed by four consortia that study genetic susceptibility of three hormone-related cancers: breast (BCAC and CIMBA), ovarian (OCAC) and prostate (PRACTICAL). Its explicit purpose is to facile the genotyping in large case-control studies for these tumors. The criteria to include the SNPs was (i) previously associated with cancer susceptibility or survival; (ii) fine mapping of genomic regions of interest; (iii) associated with cancer-related quantitative traits; (iv) in selected candidate genes or pathways; (v) associated with other cancers. The final array included ~211,000 SNPs.

GENESIS was examined in Chapter \@ref(genesis). 

### Inflammatory bowel disease

## Network view of complex diseases

<!-- motivation (link to challenges, biology as networks) -->

Human biology is notoriously complicated, as sheer numbers demonstrate: to form a 70kg man, 3.0 \texttimes 10^13^ cells [@sender_revised_2016] assemble and interact to produce and maintain 79 organs. To achieve that level of complexity, human cells depend on their genetic material, carefully tuned by epigenetics and enabled by a favorable environment. in terms of genetics, the object of my work, a human diploid genome is 6.4 billion base pairs long, and encodes 30,000 genes, of which X encode for a protein and Y are RNA genes. DNA, proteins and RNA are in constant interplay between themselves, with the metabolites, and with the environment. At some basic level, a person and their traits are just the emerging pattern born from the interaction of all these factors. Proteins physically interact with each other in highly specific ways (protein-protein interactions or PPIs). If such interactions are stable enough, proteins can assemble into large complexes to carry out particular functions. But proteins also interact with DNA to regulate gene expression (transcription factor - DNA interactions). And so on: enzymes interact with their metabolites, hormones with their receptors, the individual with their environment, etc. Biology, from ecosystems to molecular biology, cannot be understood if not as an interplay. Mindful of this, and enabled by the omics technologies of the 21st century, researchers have strived to capture these relationships. Among all the possible biological relationships, in this thesis I focused on protein-protein interactions, including protein complexes, as their coverage is larger and their properties better understood. Such interactions are available in databases like HINT [@das_hint_2012], The BioGRID [@oughtred_biogrid_2019] or IntAct [@required].

Relationships between pairs of proteins can be mathematically formalized as a network, which makes them analitically approachable. The field of systems biology relies on the assumption that the network accurately captures the context it requires to carry out its biological function. Indeed, biological networks are structured, very differently from random networks [@barabasi_network_2011]. For instance, the degree distribution, which represents the distribution of the number of edges of the nodes in a network, follows a power-law distribution. That is, a few genes have the majority of the edges in the network. In graph theory, such nodes are called *hubs*. Indeed, we observe that this node property is informative of the gene function: *in utero* essential genes, like knots preventing the network from falling apart, tend to be hubs. Another important consequence of the structured degree distribution is the emergence of modules, subsets of highly interconnected nodes. In biological networks, modules are often associated with biological function towards which all nodes jointly contribute [@required].

Another view on graphs relies on how far two nodes are from each other. For that we use the notion of paths between nodes: number of edges required to pass in order to go from one to the other. Small world network

Talk about local mesoscale and global?

### Networks in disease

The network view of biology depicted above requires a nuanced approach to disease: cut the knot in the center of the web that keeps it all together, and it all will fall off; cut a bunch of peripheral, less important nodes around it, the outcome might be the same. In other words, there are many ways of producing the same disease.

The observations on biological networks enumerated above lay down the foundations to use biological networks to study disease genes. They explain the local hypothesis, which expects genes involved in a disease to interact with each other [@barabasi_network_2011]. The disease module hypothesis, which expects genes involved in the same disease to share a module.

Indeed, disease genes exhibit different properties than non-disease genes in disease [@pinero_uncovering_2016; @cai_broker_2010; @furlong_human_2013; @barabasi_network_2011]. For starters, disease genes are not hubs [@cai_broker_2010; @das_hint:_2012], but tend to be non-essential genes located in the periphery. This is coherent with a evolutionary mindset, where mutations in essential genes would be highly deleterious, even resulting in embryonic lethality. However, this enrichment is driven mainly by cancer genes [@pinero_uncovering_2016]. Additionally, disease genes tend to be bottlenecks i.e. the sole link between many peripheral genes and the rest of the network [@cai_broker_2010], suggesting disease arise when these vulnerable regions of the network break down. 

These properties, however, do not affect equally all types of disease genes. Many of the aforementioned properties do not extend to genes identified through GWAS [@cai_broker_2010]. However, differences arise even when comparing genes involved in complex diseases with those involved in Mendelian ones. 

** Network propagation experiments 

Network propagation includes a set of methods which use all the possible paths in the network to re-rank the genes on it [@cowen_network_2017]. In essence, for these methods, association of a node with a phenotype can be thought of as a volume of water: the more strongly associated, the more voluminous. Generically, each node starts with an initial volume which, iteratively, gets distributes its liquid among its neighbors. Equivalently, that node will receive a share from its neighbors'. The expectation is that truly associated genes will be densely interconnected to each other, forming cycles and modules that will keep the water from diffusing to other, uninvolved genes. After a number of steps the volume of water in each node is re-evaluated, and used to re-prioritize the genes. An added benefit of network propagation method is that they account for biases introduced by highly connected nodes. The outcome of different network propagation methods, that is, the vector p containing the scores of the nodes after k steps or in stationary state, can be expressed by the equation [@cowen_network_2017]:

p = S p_0,

where p_0 is the vector containing the association of each node to the phenotype, and S is a network-derived similarity matrix. S_ij represents the information propagated from node j to node i. Among network propagation methods, we include heat diffusion, random walk, graph kernels and even Google's PageRank. Below we describe different network propagation methods. Readers interested in network propagation can read [@cowen_network_2017].

Proved that they are important [@huang_systematic_2018].

<!-- Hindrances -->

Undirected edges [@barabasi_network_2011].

Biases introduced by technology [@pinero_uncovering_2016].

Posed by small world network, when the edges considered are not relevant for the particular disease [@pinero_uncovering_2016].

### Network-guided disease studies 

As exemplified in the previous section, networks can be leveraged on to gain insight of the biology of the disease.

#### High-score subnetwork search

One of the focuses of my thesis was the study of networks where the nodes have been scored by their association to the disease. Such scoring might come from omics experiments, or from /a priori/ knowledge of disease genes. In essence, such methods look for connected subnetworks made of nodes with high scores. Lacking a term broadly agreed in the community, I refer to such algorithms as high-score subnetwork search (HS3). Although heterogeneous, to some extend all such approaches are based on the guilt-by-association principle: nodes nearby associated nodes are suspect of being associated themselves. Hence, if they display some level of association, even if it is not significant by conventional standards, we are likely to believe it. However, taking only the genes associated with those associated would be prone to false positives, as often networks include edges that are not relevant for the biological problem at hand.

From a machine learning perspective, these methods are feature selectors. THat is, algorithms that select features based on potential utility as predictors of a class. To that end, they assign weights to features, rank them, or select a subset.

There are plenty of such methods available in the literature. Essentially, they differ in the considerations they make on how the solution looks like. Hence, it makes sense to exploit that criteria to create a classification of the methods. In Chapter \@ref(genesis) we describe a representative set of these methods and critically discuss their performance at biomarker discovery on the GENESIS dataset. 

#### Consensus clustering

In Chapter TODO we discuss a naive way of aggregating subnetworks from different high-score subnetwork search algorithms, were the edges are unweighed. In Chapter TODO, we discuss how to apply consensus clustering to networks coming from different epistasis detection methods, whose edges are weighed by the confidence they exist.

#### Community detection

In Appendix \@ref(smartas) we apply community detection techniques to a subnetwork of genes with abnormal splicing.

#### Modelling

Based on these interactions, the complex interplay described above can be formalized as a mathematical network. In such a network, selected biomolecules are nodes, and selected relationships between them, edges. Model building.

However, parametrizing the network is hard. And a whole model, impossible. Just recently it was done on Mycoplasma (Luis Serrano?). Nonetheless, the network itself provides a good framework to study disease. Interesting conclusions can be drawn from such networks.

Most of the edges are undirected, because of reasons.

Pinhero et al.

## Contributions

<!-- TODO The bulk of my thesis revolved around GWAS and networks. -->

In collaboration with Dr. Makoto Yamada and Dr. Samuel Kaski, I developed block HSIC Lasso, a general-purpose non-linear feature selector. The work involved modifying an existing algorithm, HSIC Lasso, to reduce its memory consumption. On top of that, we worked on improving its performance, and on solving numerical issues in edge cases. The algorithm is implemented as the Python package pyHSICLasso, available on both PyPI and GitHub. Then, I characterized the algorithm and applied it to several biological datasets. Crucially, three of the datasets were GWAS, which was a milestone in terms of the number of features block HSIC Lasso can handle. Thanks to this work, we analyzed the impact of considering non-redundancy and non-linear models when selecting SNPs for patient classification. We describe the algorithm and our conclusions in the article /Block HSIC Lasso: model-free biomarker detection for ultra-high dimensional data/ published in the proceedings of the ISMB/ECCB 2019. The full manuscript is available in Appendix 1. It was made possible by a scholarship from RIKEN AIP, which allowed me to spend 3 months in Dr. Yamada's laboratory in Kyoto.

Additionally, I continued working on the involvement of alternative splicing in cancer. During 2016 and early 2017, we prepared the answer to reviewers for the article *The functional impact of alternative splicing in cancer*, which was published in Cell Reports in August 2017. The full manuscript is available in Appendix 2. In the same dates, I explored evidence of epistasis in cancer, looking for mutual exclusivity between alterations in alternative splicing and somatic mutations. We searched evidence of mutual exclusivity at both the gene and the pathway level. Among others, we found a compelling pattern in GATA3. This finding was published in Cell Reports in March 2018, in the broader article *Systematic analysis of splice-site-creating mutations in cancer*. In addition, I put compiled the code required for the analyses in *The functional impact of alternative splicing in cancer*, and created a Python package with a clean interface. The package, *spada*, is available in both PyPI and GitHub. This package searches alterations of alternative splicing, and maps them to functional consequences at the protein level. Specifically, such consequences are the removal/addition of functional modules to the protein (e.g. domains) and the loss or gain of protein-protein interactions. Lastly, I applied *spada* to two leukemia datasets from TARGET. This last analysis is still on-going.

As part of the IC-3i International Ph.D. Program, I was part of a doctoral mission, unrelated to my research, which would take one month of work every year. In 2017, it involved the preparation of the "3rd course on Breast Cancer, from clinics to biology". In 2018, it consisted on leading my own collaboration described above with Dr. Eyras. In 2019, I helped mantaining the website of the Training Unit of Institut Curie. 

Lastly, I carried out my research as committed as I could to open, reproducible science. As such, all my projects have an associated, version-controlled, laboratory notebook, which includes as much data as I am allowed to share. All such laboratory notebooks are made out public when the paper is. All the scripts I developed are open source, under permissive MIT license. Specifically, I made an effort to develop project-independent tools, which are useful to anyone which needs similar to the ones I had. These tools are published on GitHub https://github.com/hclimente/gwas-tools. 