# Combining network-guided GWAS to discover susceptibility mechanisms for breast cancer {#genesis}

The contents of this chapter are based on *Héctor Climente-González, Christine Lonjou, Fabienne Lesueur, GENESIS investigators, Dominique Stoppa-Lyonnet, Nadine Andrieu and Chloé-Agathe Azencott*, Combining network-guided GWAS to discover susceptibility mechanisms for breast cancer (manuscript in preparation). Parts of this work were presented in ASHG 2018 in the poster *Héctor Climente-González, Christine Lonjou, Fabienne Lesueur, GENESIS investigators, Dominique Stoppa-Lyonnet, Nadine Andrieu and Chloé-Agathe Azencott*, Judging genetic loci by the company they keep: comparing network-based methods for biomarker discovery in familial breast cancer. 

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
library(magrittr)
library(kableExtra)

cat("\\begin{chapabstract}{Abstract:}
Systems biology provides a comprehensive approach to biomarker discovery and biological hypothesis building. It does so by jointly considering the statistical association between a gene and a phenotype, and the biological context of each gene, represented as a network. In this chapter we study the utility of six network methods to discover new biomarkers for breast cancer susceptibility by searching subnetworks highly associated to a phenotype. We interrogate a familial breast cancer genome-wide association study (GWAS) focused on BRCA1/2 negative French women. By trading statistical astringency for biological meaningfulness, most network methods get more compelling results than standard SNP- and gene-level analyses, recovering causal subnetworks tightly related to cancer susceptibility. We perform an in-depth benchmarking of the methods with regards to size of the solution subnetwork, their utility as biomarkers, and the stability and the runtime of the methods. Interestingly, a combination of solution subnetworks provided a concise subnetwork of 93 genes, enriched in known breast cancer susceptibility genes (BABAM1, BLM, CASP8, FGFR2, and TOX3, Fisher's exact test P-value = $7.8 \\times 10^{-5}$) and more central than average. Additionally, it includes subnetworks of mechanisms related to cancer, like protein folding (HSPA1A, HSPA1B, and HSPA1L) or mitochondrial ribosomes (MRPS30, MRPS31, MRPS18B). We also observed a general dysregulation in the neighborhood of COPS5, a gene related to multiple hallmarks of cancer.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}

\\end{chapabstract}")
```

## Introduction

Network-based biomarker discovery methods provide a comprehensive
statistical framework which includes the structure of biological data
tackle the the challenges in the GWAS field
(Section \@ref(intro-gwas-challenges)). In essence, each SNP has a measure of
association with the disease, given by the experiment, and functionally
biological relationships, given by a network built on prior knowledge.
Then, the problem becomes finding a functionally-related set of genes
that is highly associated with the disease. Such set is easier to 
interpret, as it is related to known biological mechanisms, and trade 
statistical significance for biological relevance, potentially improving
the power of the study. This problem is addressed by the high-scoring 
subnetwork search methods (Section \@ref(intro-hs3)) which use a 
guilt-by-association strategy to identify disease 
susceptibility genes [@azencott_network-guided_2016].
Different solutions have been proposed, often stemming from divergent different
mathematical frameworks and considerations of what the optimal solution
looks like. Some methods strongly constrain the problem to certain kinds
of subnetworks. Such is the extreme case of LEAN
[@gwinner_network-based_2016], which focuses on star subnetworks, i.e.
instances were both a gene and its direct interactors are associated
with the disease. Other algorithms, like dmGWAS [@jia_dmgwas:_2011] and
heinz [@dittrich_identifying_2008], focus on interconnected genes with
high association with the disease. However, they differ in their
tolerance to the inclusion of lowly associated nodes, and the possible
number of disconnected subnetworks in the solution. Lastly, other
methods also consider the topology of the network, favoring solutions
that are densely interconnected; such is the case of HotNet2
[@leiserson_pan-cancer_2015], SConES [@azencott_efficient_2013], and
SigMod [@liu_sigmod:_2017].

In this chapter, we analyze the effectiveness of these six methods to
discover new biomarkers on GWAS data. We focus on the GENESIS dataset
[@sinilnikova_genesis:_2016], a study of familial breast cancer
conducted in the French population (Section \@ref(genesis-dataset)). 
After following a classical GWAS
approach, we use these network-based methods to recover additional
familial breast cancer biomarkers. Some of them are known, while others
are specific to this dataset. Lastly, we carry out a comparison of the
solutions obtained by the different methods, and aggregate them to
obtain a consensus network of predisposition to familial breast cancer.

## Methods

### Data preprocessing and quality control

In this study we used the GENESIS dataset, described in Section \@ref(genesis-dataset).
We discarded SNPs with a minor allele frequency lower than 0.1%, those
not in Hardy - Weinberg equilibrium in controls (P-value \<0.001), and
those missing on more than 10% of the samples. A subset of 20 duplicated
SNPs in *FGFR2* were also removed. In addition, we removed the samples
with more than 10% missing genotypes, and an additional 28 samples with
TODO. The final dataset included 1\,271 controls and 1\,280 cases,
genotyped over 197\,083 SNPs.

We looked for population structure that could create confounding
associations (Section \@ref(pop-structure)). A PCA revealed no differential 
population structure between cases and controls (Figure \@ref(fig:pcs)).
Independently, we did not find evidence of genomic inflation ($\lambda$
= 1.05) either, further confirming the absence of confounding population
structure.

(ref:fig-pcs-caption) GENESIS shows no differential population structure between cases and controls. **(A,B,C,D)** Eight main principal components computed on the genotypes of GENESIS. Cases are colored in green, controls in orange.

```{r pcs, echo=FALSE, fig.cap='(ref:fig-pcs-caption)'}
knitr::include_graphics('fig/sfigure_1', dpi = NA)
```

### High-score subnetwork search algorithms {#sec:org6ced6c9}

#### SNP and gene association {#methods:node_score}

To measure association between a genotype and the phenotype, we
performed a per-SNP 1 d.f. $\chi$^2^ allelic test using PLINK v1.90
[@chang_second-generation_2015]. Then, we used VEGAS2v2 to compute the
gene-level association score from the SNP P-values [@mishra_vegas2:_2015]. 
In order to map SNPs to genes we used their overlap on the
genome: all SNPs located within the boundaries of a gene, $\pm 50$ kb, were
mapped to that gene. To compute the gene association we used the 10% of
SNPs linked to the gene with lowest P-values. We used the 62\,193 genes
described in GENCODE 31 [@frankish_gencode_2019], although only 54\,612
could be mapped to at least one SNP. Out of those, we focused
exclusively on the 32\,767 that could be mapped to an HGNC symbol. Out of
the SNPs 197\,083 remaining after quality control, 164\,037 mapped to at
least one of these genes.

We use such mapping to compare the outputs of methods who produce SNP-
to those that produce gene-lists, and vice versa. In the former case, we
consider any gene that can be mapped to any of the selected SNPs as
selected as well. In the latter, we consider all the SNPs that can be
mapped to that gene as selected by the method.

#### Mathematical notation {#methods:notation}

In this chapter, we use undirected, vertex-weighted networks, or graphs,
$G = (V,E,w)$. $V = \{v_{1}, \dots{}, v_{n}\}$ refers to the vertices,
with weights $w: V \rightarrow \mathbb{R}$. Equivalently,
$E \subseteq \{\{x,y\} | x,y \in V \wedge x \neq y\}$ refers to the
edges. When referring to a subnetwork S, $V_{S}$ is the set of nodes in
S and $E_{S}$ is the set of edges in S. A special case of subgraphs are
*connected* subgraphs, which occur when every node in the subgraph can
be reached from any other node.

On top of a weight, nodes have other properties provided by the topology
of the graph. In this chapter we focus on two: degree centrality, and
betweenness centrality. The degree centrality, or degree, is the number
of edges that a node has. The betweenness centrality, or betweenness, is
the number of times a node participates in the shortest paths between
two other nodes.

In addition, we use several matrices that describe different properties
of a graph. The described matrices are square, and have as many rows and
columns as nodes are in the network. The element $(i,j)$ represents a
selected relationship between $v_i$ and $v_j$. The *adjacency matrix*
$W_G$ contains a 1 when the corresponding nodes are connected through an
edge, and 0 otherwise; the diagonal is zero. The *degree matrix* $D_G$
is a diagonal matrix which contains the degree of the different nodes.
Lastly, the *Laplacian matrix* $L_G$ is defined as $L_G = D_G - W_G$.

#### Methods used {#methods:methods}

Beyond the assumption that genes that contribute to the same function
will be nearby in the protein-protein interaction network (PPIN), they
might be topologically related to each other in diverse ways (densely
interconnected modules, nodes around a hub, a path, etc.). That is not
the only choice to make: how to score the nodes, whether the affected
mechanisms form a single connected component or several, how to frame
the problem in a computationally efficient fashion, what is the best
network to use, etc. In consequence, multiple solutions have been
proposed. In this chapter, we examine six of them: five that explore the
protein-protein interaction network, and one which explores SNP
networks. We selected methods that were open source, had an
implementation available, and an accessible documentation. Their main
differences are summarized in Table \@ref(tab:method-comparison).

(ref:tab-method-comparison-footnote) *Field*: field in which the algorithm was developed. *Nodes*, the type of network, either gene (protein-protein interaction network usually) or a SNP network. *Exhaustive*: whether all the possible solutions given the selected hyperparameters are explored. *Solution*: additional properties are enforced on the solution subnetwork, other than being dense in high scores and connected. *Components*: number of connected subnetworks in the solution. *Input*: genotype data or GWAS summary statistics. *Scoring*: how SNP/gene P-values are transformed into node scores.

(ref:dmgwas) @jia_dmgwas:_2011

(ref:heinz) @dittrich_identifying_2008

(ref:hotnet) @leiserson_pan-cancer_2015

(ref:lean) @gwinner_network-based_2016

(ref:scones) @azencott_efficient_2013

(ref:sigmod) @liu_sigmod:_2017

(ref:logten) -log~10~(P)

(ref:chisq) $\chi^2$

```{r method-comparison, echo=FALSE}
method_summary <- rbind(
  c('dmGWAS', 'GWAS', 'Genes', 'No', '', '1', 'Summary', '(ref:logten)', '(ref:dmgwas)'),
  c('heinz', 'Omics', 'Genes', 'Yes', '', '1', 'Summary', 'BUM', '(ref:heinz)'),
  c('HotNet2', 'Omics', 'Genes', 'Yes', 'Modular', '> 1', 'Summary', 'Local FDR', '(ref:hotnet)'),
  c('LEAN', 'Omics', 'Genes', 'Yes', 'Star', '> 1', 'Summary', '(ref:logten)', '(ref:lean)'),
  c('SConES', 'GWAS', 'SNPs', 'Yes', 'Modular', '> 1', 'Genotypes', '(ref:chisq)', '(ref:scones)'),
  c('SigMod', 'GWAS', 'Genes', 'Yes', 'Modular', '1', 'Summary', '(ref:logten)', '(ref:sigmod)')
)

colnames(method_summary) <- c('Method', 'Field', 'Nodes', 'Exhaustive', 'Solution', 'Components', 'Input', 'Scoring', 'Reference')

kable(method_summary, 
      caption = 'Summary statistics on the results of multiple network methods on the gene-gene interaction network. The first row contains the summary statistics on the whole network.', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = c("striped", "scale_down")) %>%
  footnote(general = '(ref:tab-method-comparison-footnote)',
           threeparttable = TRUE, fixed_small_size = TRUE)
```

dmGWAS

:   dmGWAS searches the subgraph with the highest local density in low
    P-values [@jia_dmgwas:_2011]. To that end it searches candidate
    subnetwork solutions using a greedy, "seed and extend", heuristic:

    1.  Select a seed node.

    2.  Compute Stouffer's Z-score Z~m~ for the current subgraph as

        $$Z_m = \frac{\sum z_i}{\sqrt{k}}$$

        where *k* is the number of genes in the subgraph, z~i~ =
        $\phi$^-1^(1 - P-value~i~), and $\phi$^-1^ is the inverse normal
        distribution function.

    3.  Identify neighboring nodes i.e. nodes at distance $\le$ *d*. We
        set d = 2.

    4.  Add the neighboring nodes whose inclusion increases the Z~m+1~
        more than Z~m~ (1 + r). In our experiments, we set r = 0.1.

    5.  Repeat 2-4 until no increment Z~m~ (1 + r) is possible.

    Lastly, the module's Z-score is normalized as

    $$Z_{N}=\frac{Z_{m}-\operatorname{mean}\left(Z_{m}(\pi)\right)}{\operatorname{SD}\left(Z_{m}(\pi)\right)}$$

    where Z~m~($\pi$) represent a vector containing 100000 random
    subsets of the same number of genes.

    We used the implementation of dmGWAS in the dmGWAS 3.0 R package
    [@dmgwas]. We used the function *simpleChoose* to select the
    solution subnetwork, which aggregates the top 1% modules into the
    solution subnetwork.

heinz

:   The goal of heinz is to identify the highest-scored connected
    subgraph on the network [@dittrich_identifying_2008]. The authors
    propose a transformation of the genes' P-value into a score that is
    negative under no association with the phenotype, and positive value
    when there is. This transformation is achieved by modelling the
    distribution of P-values by a beta-uniform model (BUM) parameterized
    by the desired FDR. Thus formulated, the problem is NP-complete. To
    solve it efficiently it is re-casted as the Prize-Collecting Steiner
    Tree Problem (PCST), which seeks to select the connected subnetwork
    S that maximizes the *profit* p(S):

    $$p(S) = \sum_{v \in V_S} p(v) - \sum_{e \in E_S} c(e).$$

    were p(v) = w(v) - w' is the *profit* of adding a node, c(e) = w' is
    the *cost* of adding an edge, and $w' = min_{v \in V_{G}} w(v)$. All
    three are positive quantities. heinz implements the algorithm from
    [@ljubic_algorithmic_2006], which in practice is often fast and
    optimal, neither is guaranteed. We used BioNet's implementation of
    heinz, available on Bioconductor [@beisser_bionet:_2010; @heinz].

HotNet2

:   HotNet2 was developed to find connected subgraphs of genes
    frequently mutated in cancer [@leiserson_pan-cancer_2015]. To that
    end, it considers both the local topology of the network and the
    scores of the nodes. The former is captured by an insulated heat
    diffusion process: at the beginning, the score of the node
    determines its initial heat; iteratively each node yields heat to
    its "colder" neighbors, and receives heat from its "hotter"
    neighbors, while retaining part of its own (hence, *insulated*).
    This process continues until equilibrium is reached, and results in
    a similarity matrix F. F is used to compute the similarity matrix E
    that accounts also for similarities in node scores as

    $$E = F \operatorname{diag}(w(V)),$$

    where $\operatorname{diag}(w(V))$ is a diagonal matrix with the node
    scores in its diagonal. We scored the nodes as in
    @nakka_gene_2016, assigning a score of 0 for the genes with low
    probability of being associated to the disease, and
    -log~10~(P-value) to those likely to be. In this dataset, the
    threshold separating both was a P-value of 0.125, which was obtained
    using a local FDR approach [@scheid_twilight_2005]. To obtain
    densely connected subnetworks, HotNet2 prunes E, only preserving
    edges such that w(E) \> $\delta$. Lastly, HotNet2 evaluates the
    statistical significance of the subnetworks by comparing their size
    to the size of networks obtained by permuting the node scores.
    HotNet2 has two parameters: the restart probability $\beta$, and the
    threshold heat $\delta$. Both parameters are set automatically by
    the algorithm, and are robust [@leiserson_pan-cancer_2015]. HotNet2
    is implemented in Python [@hotnet2].

LEAN

:   LEAN searches disregulated "star" gene subnetworks, that is,
    subnetworks composed by one central node and all its interactors
    [@gwinner_network-based_2016]. By imposing this restriction, LEAN is
    able to exhaustively test all such subnetworks (one per node). For a
    particular subnetwork of size *m*, the P-values corresponding to the
    involved nodes are ranked as p~1~ $\le$ ... $\le$ p~m~. Then, *k*
    binomial tests are conducted, to compute the probability of having
    *k* out of *m* P-values lower or equal to p~k~ under the null
    hypothesis. The minimum of these *k* P-values is the score of the
    subnetwork. This score is transformed into a P-value through an
    empirical distribution obtained via a subsampling scheme, where sets
    of *m* genes are selected randomly, and their score computed.
    Lastly, P-values are corrected for multiple testing through a
    Benjamini-Hochberg correction. We used the implementation of LEAN
    from the LEANR R package [@leanr].

SConES

:   SConES searches the minimal, modular, and maximally associated
    subnetwork in a SNP graph [@azencott_efficient_2013]. Specifically,
    it solves the problem
    
    \begin{equation}
    \underset{S \subseteq G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} + \underbrace{\lambda \sum_{v \in V_S} \sum_{u \not\in V_S} L_{vu} }_{\text { connectivity }}-\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}
    (\#eq:scones)
    \end{equation} 

    where $\lambda$ and $\eta$ are hyperparameters that control the
    sparsity and the connectivity of the model. Given two
    hyperparameters, the aforementioned problem has a unique solution,
    that SConES finds using a graph min-cut procedure. We used the
    version on SConES implemented in the R package martini [@martini].
    As in @azencott_efficient_2013, we selected $\lambda$ and $\eta$
    by cross-validation, choosing the values that produce the most
    stable solution across folds. Note that the solution to the above
    problem can consist of several connected subnetworks which are
    disconnected from each other. In this case, the selected
    hyperparameters were $\eta$ = 3.51, $\lambda$ = 210.29 for SConES
    GS; $\eta$ = 3.51, $\lambda$ = 97.61 for SConES GM; and $\eta$ =
    3.51, $\lambda$ = 45.31 for SConES GI.

SigMod

:   SigMod aims at identifying the most densely connected gene
    subnetwork that is most strongly associated to the phenotype
    [@liu_sigmod:_2017]. It addresses an optimization problem similar to
    that of SConES (Equation \@ref(eq:scones)), but using the Laplacian 
    matrix rather than the adjacency matrix (Section \@ref(methods:notation)), 
    to quantify solutions containing many edges.

    $$\underset{S \in G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} + \underbrace{\lambda \sum_{v \in V_S} \sum_{u \in V_S} W_{vu} }_{\text { connectivity }} -\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}.$$

    As SConES, this optimization problem can also be solved by a graph
    min-cut approach.

    SigMod presents three important differences with SConES. First it is
    designed for gene-gene networks. Second, by replacing the adjacency
    by the Laplacian matrix, it favors subnetworks containing many
    edges. SConES, instead, penalizes connections between the selected
    selected and unselected nodes. Third, it returns a single connected
    subnetwork, which it achieves by exploring a grid of hyperparameters
    and processing their respective solutions. Specifically, for the
    range of $\lambda$ = $\lambda$~min~, ..., $\lambda$~max~ for the
    same $\eta$, it prioritizes the solution with the largest change in
    size from $\lambda$~n~ to $\lambda$~n+1~. Such a large change
    implies that the network is strongly interconnected. This results in
    one candidate solution for each $\eta$, which are processed by
    removing any node not connected to any other. A score is assigned to
    each candidate solution by summing their node scores and normalizing
    by size. The candidate solution with the highest standardized score
    is the chosen solution. SigMod is implemented in an R package
    [@sigmod].

#### Gene-gene network {#methods:gene-network}

Out of the six methods tested, five use a gene-gene interaction network
(Section \@ref(methods:methods)). Although their respective statistical
frameworks are compatible with any type of network (protein
interactions, gene coexpression, regulatory, etc.), for practical
reasons we focused on a PPIN, as they are interpretable, well
characterized, and most of the methods were designed to scale
appropriately to it. We built our PPIN from both binary and co-complex
interactions stored in the HINT database (release April 2019)
[@das_hint_2012]. Unless specified otherwise, we used only interactions
coming from high-throughput experiments to avoid biasing the topology of
the network by well-studied genes with more known interactions on
average. Out of the 146\,722 interactions from high-throughput
experiments that HINT stores, we were able to map 142\,541 to a pair of
HGNC symbols. The scoring function for the nodes changed from method to
method (Section \@ref(methods:methods)).

Additionally, we compared the results of the aforementioned PPIN with
those obtained on another PPIN built using interactions coming from both
high-throughput and targeted studies. In that case, out of the 179\,332
interactions in HINT, we mapped 173\,797 to a pair of HGNC symbols.

#### SNP networks {#methods:snp-network}

SConES [@azencott_efficient_2013] is the only of the studied methods
designed to handle SNP networks. As in gene networks, two SNPs are
linked in a SNP network when there is evidence of shared functionality
between two SNPs. The authors suggested three ways of building these
networks: connecting the SNPs consecutive in the genomic sequence ("GS
network"); interconnecting all the SNPs mapped to the same gene, on top
of GS ("GM network"); and interconnecting all SNPs mapped to two genes
for which a protein-protein interaction exists ("GI network"). We
focused on the GI network, as it is the network that fits better the
scope of this chapter. However, at different stages of the chapter we
also used GS and GM. For the GM network, we used the mapping described
in Section \@ref(methods:node_score). For the GI network, we used the PPI as
described in Section \@ref(methods:gene-network). For all three networks the 
node score used is the association of the individual SNPs with the phenotype;
specifically, we used the 1 d.f. $\chi$^2^.

#### Consensus network {#methods:consensus}

The different high-weight subnetwork discovery algorithms make different
assumptions on the properties of the solutions, and employ different
strategies to find them. Hence, combining the outcome of the different
approaches might provide a more complete outlook on the specific
alterations on the GENESIS dataset. We built such consensus network by
retaining the nodes that were selected by at least two of the methods.
We combined the results of 6 methods: dmGWAS, heinz, HotNet2, LEAN,
SConES GI, and SigMod.

### Evaluation of methods {#sec:orgc968928}

#### Classification accuracy of selected biomarkers {#methods:comparison}

A desirable solution is one that is sparse, while offering a good
predictive power on unseen samples. We evaluated the predicting power of
the SNPs selected by the different methods through the performance of an
L1-penalized logistic regression trained exclusively on those SNPs to
predict the outcome (case/control). The L1 penalty helps to account for
LD to reduce the number of SNPs included in the model (size of the
active set), while improving the generalization of the classifier. The
value of the regularization parameter, which controls both the magnitude
and the sparsity of the coefficients, was set by cross-validation. To
that end, we used the different network-methods on a random subset of
80% of the samples. On this same subset we trained our classifier
exclusively on the SNPs selected by a particular method. When the method
retrieved a list of genes (all of them except SConES), all the SNPs
mapped to any of those genes were used. Then we
evaluated performance of the classifier on the remaining 20% of the
dataset. We repeated this procedure 5 times to estimate the average and
the deviation of the different performance measures. The 
performance measures we used were: size of the solution, size of the
active set, specificity, and sensitivity. The size of the active set
provides an estimate of a plausible, more sparse solution with a
comparable predictive power to the original solution.

Additionally, for each of the methods, we evaluated their stability and
their runtime. The stability of an algorithm is its sensitivity to small
changes of the input, and is measured using the Pearson's correlation
between different runs as suggested in @nogueira_measuring_2016. To
obtain a baseline, we also performed the procedure using all the SNPs.
Lastly, another desirable property is that the method retrieves a good
candidate causal subnetwork. In consequence, we compared the outcome of
each of the methods to the consensus subnetwork of all the solutions
(Section \@ref(methods:consensus)).

#### Biological relevance of the genes {#methods:bcac}

An alternative way to validate the results is comparing our results to
an external dataset. For that purpose, we recovered a list of 153 genes
associated to familial breast cancer from DisGeNET
[@pinero_disgenet:_2017]. Across this chapter we refer to these genes as
*breast cancer susceptibility genes*.

Additionally, we used the summary statistics from the Breast Cancer
Association Consortium (BCAC) [@michailidou_genome-wide_2015]. BCAC has
conducted one of the largest efforts in GWAS, involving over 120\,000
women of European ancestry. As opposed to GENESIS, samples were not
selected based on family history, and hence the study is enriched in sporadic
breast cancers. Another difference is that BCAC is a relatively
heterogeneous study on a pan-European sample, while GENESIS is a
homogeneous dataset focused on the French population. Despite these
differences, there should be shared genetic architecture. On top of
that, that overlap should become larger when the results are aggregated
at the gene level. For that purpose, we computed the gene association as
in Section \@ref(methods:node_score). The iCOGS array was used for genotyping in
BCAC [@sakoda_turning_2013], the same array as for GENESIS
[@sinilnikova_genesis:_2016]. Although imputed data is available, we
used exclusively the SNPs available on GENESIS after quality control to
make the results comparable.

### Code availability {#sec:org1d907c0}

This work required developing computational pipelines for several GWAS
analyses, such physically mapping SNPs to genes, computing gene scores,
and performing six different network analyses. For each of those
processes, a streamlined, project-agnostic pipeline with a clear
interface was created. They are compiled in the following GitHub
repository: <https://github.com/hclimente/gwas-tools>. The code that
applies these pipelines to the GENESIS project, as well as the code that
reproduces all the analyses in this chapter are available at
<https://github.com/hclimente/genewa>. Although the GENESIS dataset is
not publicly available, the published code should work on any other GWAS
dataset. All the produced gene subnetworks were deposited on NDEx
(<http://www.ndexbio.org>), under the UUID
e9b0e22a-e9b0-11e9-bb65-0ac135e8bacf.

## Results {#sec:orgc718f04}

### A conventional GWAS shows that FGFR2 is strongly associated with familial breast cancer {#results:conventional}

(ref:fig-snp-gene-manhattan-caption) Association in GENESIS. The red line represents the Bonferroni threshold. **(A)** SNP association, measured from the outcome of a 1df $\chi^2$ allelic test. Significant SNPs that are within a coding gene, or within 50 kilobases of its boundaries, are annotated. The Bonferroni threshold is $2.54 \times 10^{-7}$. **(B)** Gene association, measured by P-value of VEGAS2v2 [@mishra_vegas2:_2015] using the 10% of SNPs with the lowest P-values. The Bonferroni threshold is $1.53 \times 10^{-6}$.

```{r snp-gene-manhattan, echo=FALSE, fig.cap='(ref:fig-snp-gene-manhattan-caption)'}
knitr::include_graphics('fig/figure_1.png', dpi = NA)
```

We conducted association analyses in the GENESIS dataset at both the SNP
and the gene levels (Section \@ref(methods:node_score)). Two genomic regions 
have a P-value lower than the Bonferroni threshold in chromosomes 10 and 16 
(Figure \@ref(fig:snp-gene-manhattan) A). The former overlaps with gene
*FGFR2*; the latter with *CASC16*, and it is located near the
protein-coding gene *TOX3*. Variants in both *FGFR2* and *TOX3* were
related to breast cancer susceptibility in other cohorts negative for
*BRCA1/2* [@rinella_genetic_2013]. Only the peak in chromosome 10
replicated in the gene-level analysis, with *FGFR2* just above the
threshold of significance (Figure \@ref(fig:snp-gene-manhattan)B).

These results show the overlap in the genetic architecture of the
disease between the studied French population sample and other
populations, especially at the gene level. In addition, there are other
SNPs whose P-values, although higher than the conventional threshold of
significance, show a strong association with familial breast cancer. The
most prominent of such regions are 3p24 and 8q24, both of which have
been associated to breast cancer susceptibility in the past
[@brisbin_meta-analysis_2011; @search_newly_2009]. This motivates
exploring network methods, which trade statistical significance for
biological relevance.

### Network methods successfully identify genes associated with breast cancer {#results:separate-networks}

(ref:tab-gene-solutions-footnote) *Num genes*: number of genes selected out of those that are part of the PPIN; for SConES GI the total number of genes, including RNA genes, was added in parentheses. $\overline{\mbox{Betweenness}}$: mean betweenness of the selected genes in the PPIN. $\hat{\mbox{P}}$~gene~: median P-value of the selected genes. $\rho$~consensus~: Pearson\'s correlation between the subnetwork and the consensus network.

(ref:betweenness) $\overline{\mbox{Betweenness}}$

(ref:p-gene) $\hat{\mbox{P}}_{gene}$

(ref:rho) $\rho_{consensus}$

```{r gene-solutions, echo=FALSE}
solution_summary <- rbind(
      c('HINT HT', 13619, 142541, 16706, 0.46,0.066),
      c('Consensus', 55, 117, 74062, 0.0051, 1),
      c('dmGWAS', 194, 450, 49115, 0.19, 0.41),
      c('heinz', 4, 3, 113633, 0.0012, 0.21),
      c('HotNet2', 440, 374, 7739, 0.048, 0.31),
      c('LEAN', 0, 0, NA, NA, NA),
      c('SConES GI', '0 (1)', 0, NA, NA, NA),
      c('SigMod', 142, 249, 92603, 0.0083, 0.73)
  )
colnames(solution_summary) <- c('Network','Num genes','Num edges', 
                                '(ref:betweenness)',
                                '(ref:p-gene)', '(ref:rho)')

footnote(kable(solution_summary, 
               caption = 'Summary statistics on the results of multiple network methods on the gene-gene interaction network. The first row contains the summary statistics on the whole network.', 
               booktabs = TRUE) %>% 
           kable_styling(latex_options = c("striped", 'scale_down')),
         general = '(ref:tab-gene-solutions-footnote)', 
         threeparttable = TRUE, escape = FALSE, fixed_small_size = TRUE)
```

(ref:tab-snp-solutions-footnote) $\overline{\mbox{Betweenness}}$: mean betweenness of the selected SNPs in the corresponding full network; $\hat{\mbox{P}}$~SNP~: median P-value of the selected SNPs.

(ref:p-snp) $\hat{\mbox{P}}$~SNP~

```{r snp-solutions, echo=FALSE}
solution_summary <- read.table(text = "
                  Network SNPs Edges Subnetworks Betweenness P
                  GS 197083 197060 NA 20300000 0.49
                  'SConES GS' 1590 1585 5 25200000 0.023
                  GM 197083 6442446 NA 3990000 0.49
                  'SConES GM' 1692 177611 5 4400000 0.055
                  GI 197083 28733720 NA 1460000 0.49
                  'SConES GI' 408 539 5 9330000 0.076
                  ", header = TRUE, stringsAsFactors = FALSE)
solution_summary$Edges <- formatC(solution_summary$Edges, digits = 2, format = 'e')
solution_summary$Betweenness <- formatC(solution_summary$Betweenness, digits = 2, format = 'e')

colnames(solution_summary) <- c('Network', 'SNPs', 'Edges', 'Subnetworks',
                                '(ref:betweenness)', '(ref:p-snp)')

kable(solution_summary, 
      caption = 'Summary statistics on the results of SConES on the three SNP-SNP interaction networks. The first row within each block contains the summary statistics on the whole network.', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = c("striped", 'scale_down')) %>%
  footnote(general = '(ref:tab-snp-solutions-footnote)', 
           threeparttable = TRUE, escape = FALSE, fixed_small_size = TRUE)
```

We applied six network methods to the GENESIS dataset (Section
\@ref(methods:methods)), obtaining six solutions (Figure 
\@ref(fig:overview-solutions), Supplementary Files 1 and 2): one
for each of the five gene-based methods (Section
\@ref(methods:gene-network)), and one for SConES GI (Section
\@ref(methods:snp-network)). The solutions are very heterogeneous
(Table \@ref(tab:gene-solutions) and Table
\@ref(tab:snp-solutions)): none of the subnetworks examined by
LEAN is significant (adjusted P-value \< 0.05), while HotNet2 produced
the largest solution subnetwork with 440 genes. SConES GI failed to
recover genes in the PPIN, but it recovered one genomic region mapped to
RNA gene *RNU6-420P*. All solution subnetworks except LEAN's are, on
average, more strongly associated to breast cancer than the whole PPIN
(median P-values $\ll 0.46$), despite containing genes with higher
P-values (Figure \@ref(fig:manhattan-solutions)). This exemplifies the 
trade-off between statistical significance and biological relevance. However,
there are nuances between solutions: heinz strongly favored highly
associated genes, while dmGWAS is less conservative (median gene
P-values 0.0012 and 0.19, respectively); SConES tended to select whole
LD-blocks; and HotNet2 and SigMod were less likely to select lowly
associated genes.

(ref:fig-overview-solutions-caption) Overview of the subnetworks produced by the different network methods. **(dmGWAS, heinz, HotNet2, LEAN, and SigMod)** contain gene subnetworks; **(SConES GI)**, SNP subnetworks.

```{r overview-solutions, echo=FALSE, fig.cap='(ref:fig-overview-solutions-caption)'}
knitr::include_graphics('fig/sfigure_2', dpi = NA)
```

(ref:fig-manhattan-solutions-caption) Manhattan plots showing the biomolecules selected by each method. In **(Consensus, dmGWAS, heinz, HotNet2, and SigMod)** datapoints are genes; in **(SConES GS, GM, and GI)**, SNPs. LEAN was excluded, as it did not select any gene.

```{r manhattan-solutions, echo=FALSE, fig.cap='(ref:fig-manhattan-solutions-caption)', out.height='80%', fig.align='center'}
knitr::include_graphics('fig/sfigure_3.png', dpi = NA)
```

The solution subnetworks present other desirable properties. First, four
of the methods succeeded at recovering genes involved in the disease
(Figure \@ref(fig:disgenet-comparison)), as their subnetworks were
enriched in breast cancer susceptibility genes (dmGWAS, heinz, HotNet2,
and SigMod, Fisher's exact test one-sided P-value \< 0.03). We also
compared the outcome of the network methods to the association tests
conducted on the population of European ancestry from the Breast Cancer
Association Consortium (BCAC) [@michailidou_genome-wide_2015]
(Figure \@ref(fig:bcac-comparison)). Encouragingly, every solution
subnetwork is enriched in genes or SNPs that are Bonferroni-significant
in BCAC. This confirms the capability of network methods to find the
same signal as in more powered studies by leveraging on prior knowledge.
Second, the genes in four solution subnetworks display on average a
higher betweenness centrality than the rest of the genes, a difference
that is significant in three solutions (dmGWAS, and SigMod, Wilcoxon
rank-sum test P-value \< 1.4 10^-21^). This agrees with the notion that
disease genes are more central than other, non-essential genes
[@pinero_uncovering_2016]. We observe that this conclusion holds in this
disease, as known breast cancer susceptibility genes have higher
betweenness centrality than others (one-tailed Wilcoxon rank-sum test
P-value = 2.64 10^-5^, Figure \@ref(fig:consensus-stats)C). Interestingly, 
SConES' selected SNPs are also more central than the average SNP (Table
\@ref(tab:snp-solutions)), suggesting that causal SNPs are 
also more central than unrelated SNPs. However, very central nodes are also
more likely to be connecting a random pair of nodes, making then more
likely to be selected by the examined methods. Hence, further work is
needed draw conclusions.

(ref:fig-disgenet-comparison-caption) Proportion of the selected genes by each of the methods on the GENESIS data that is a known breast cancer susceptibility gene (Section \@ref(methods:bcac)). Only genes present in the protein-protein interaction network were considered. LEAN is not displayed as it did not select any gene. The presented network methods recover a higher proportion of breast cancer susceptibility genes than of other genes, despite their lack of significance in GENESIS.

```{r disgenet-comparison, echo=FALSE, fig.cap='(ref:fig-disgenet-comparison-caption)', out.height='80%', fig.align='center'}
knitr::include_graphics('fig/sfigure_5', dpi = NA)
```

(ref:fig-bcac-comparison-caption) Proportion of the Bonferroni significant biomolecules (in either the GENESIS or the BCAC datasets) selected by each of the methods on the GENESIS data. **(Consensus, dmGWAS, heinz, HotNet2, and SigMod)** involve significant genes, only among those present in the protein-protein interaction network. **(SConES GS, GM and GI)** involve significant SNPs. LEAN is not displayed as it did not select any gene. The presented network methods recover a higher proportion of significant genes than of non-significant genes in both datasets, despite their lack of significance in GENESIS.

```{r bcac-comparison, echo=FALSE, fig.cap="(ref:fig-bcac-comparison-caption)", out.height='80%', fig.align='center'}
knitr::include_graphics('fig/sfigure_4', dpi = NA)
```

(ref:fig-consensus-stats-caption) Genes on the consensus network. Breast cancer susceptibility genes are colored in pink; the rest are colored in grey. **(A)** Number of methods selecting every gene in the subnetwork. **(B)** VEGAS P-values of association of the genes, with regards to the number of methods that selected them. **(C)** Comparison of betweenness centrality of the genes in the consensus network and the other genes in the PPIN and not in the consensus network. To improve visualization, we removed outliers. **(D)** Relationship between the log~10~ of the betweenness centrality and the -log~10~ of the VEGAS P-value of the genes in the consensus network. The blue line represents a fitted generalized linear model.

```{r consensus-stats, echo=FALSE, fig.cap='(ref:fig-consensus-stats-caption)', out.height='75%', fig.align='center'}
knitr::include_graphics('fig/sfigure_8', dpi = NA)
```

As the solutions were quite different from each other it is hard to draw
joint conclusions. The 4-gene solution selected by heinz includes the
breast cancer susceptibility gene *TOX3*, in region 16q12. By dealing
with SNP networks, SConES studies the association of non-coding regions,
as well as SNPs in any gene, coding or not. In fact, SConES GI, which
adds to GM the interactions between genes, retrieves 4 subnetworks in
intergenic regions, and 1 overlapping an RNA gene (*RNU6-420P*). SigMod,
despite being related to SConES, produces a vastly different, large
solution. On top of recovering three breast cancer susceptibility genes,
a keratin-based region of its subnetwork affects the cytoskeleton
(*structural constituent of cytoskeleton*, GO enrichment's adjusted
P-value = 9.10 10^-4^), a potentially novel susceptibility mechanism for
cancer susceptibility. Interestingly, dmGWAS solution is also related to
cytoskeleton (*tubulin binding*, GO enrichment's adjusted P-value =
0.031). But, additionally, it includes a submodule of proteins related
to *unfolded protein binding* (GO enrichment's adjusted P-value =
0.045), which has been previously related to cancer susceptibility
[@calderwood_heat_2016]. Lastly, HotNet2 produced 135 subnetworks, 115
of which have less than five genes. The second largest subnetwork (13
nodes), contains the two breast cancer susceptibility genes *CASP8* and
*BLM*.

### heinz retrieves a small, highly informative set of biomarkers in a fast and stable fashion {#results-benchmark}

(ref:fig-benchmark-caption) Comparison of network-based GWAS methods on GENESIS. Each method was run 5 times of a random subset of the samples, and tested on the remaining samples (Section \@ref(methods:comparison)). **(A)** Number of SNPs selected by each method and number of SNPs on the active set used by the Lasso classifier. Points are the average over the 5 runs; lines represent the standard error of the mean. A grey diagonal line with slope 1 is added for comparison. For reference, the active set of Lasso using all the SNPs included, on average, 154\,117.4 SNPs. **(B)** Sensitivity and specificity on test set of the L1-penalized logistic regression trained on the features selected by each of the methods. In addition, the performance of the classifier trained on all SNPs is displayed. Points are the average over the 5 runs; lines represent the standard error of the mean. **(C)** Pairwise Pearson\'s correlations of the solutions used by different methods. A Pearson's correlation of 1 means the two solutions are the same. A Pearson's correlation of 0 means that there is no SNP in common between the two solutions. **(D)** Runtime of the evaluated methods, by type of network used (gene or SNP). For gene network-based methods, inverted triangles represent the runtime of the algorithm itself, and circles the total time, which includes the algorithm themselves and the additional 119\,980 seconds (1 day and 9.33 hours) which took VEGAS2v2 on average to compute the gene scores from SNP summary statistics.

```{r benchmark, echo=FALSE, fig.cap='(ref:fig-benchmark-caption)', out.width='80%', fig.align='center'}
knitr::include_graphics('fig/figure_4', dpi = NA)
```

As the employed methods produced such different results, we compared
their solutions in a 5-fold subsampling setting (Section
\@ref(methods:comparison)). Specifically, we measured 
four properties (Figure \@ref(fig:benchmark)): 
size of the solution subnetwork,
sensitivity and specificity of an L1-penalized logistic regression on
the selected SNPs, stability, and computational runtime.

Both solution size and active set of SNPs selected by Lasso varies
greatly between the different methods (Figure \@ref(fig:benchmark)A). 
heinz has the smallest solutions, with an
average of 182 selected SNPs are selected by
Lasso. The largest solutions come from SConES GI (6256.6 SNPs), and
dmGWAS (4255.0 SNPs). Interestingly, heinz has the highest proportion of
the selected SNPs that go into the active set (99.9%), although it is
high for all the methods (\> 86%). This suggests methods are selecting
informative SNPs on average.

To determine whether the selected SNPs could be used for patient
classification we computed the sensitivity and the specificity of the
classifier on the testing data (Figure \@ref(fig:benchmark)B). 
All classifiers' sensitivities were in the
0.42 -- 0.51 range; the specificities, between 0.54 and 0.62. On
average, SigMod had the highest sensitivity (0.51); dmGWAS, the highest
specificity (0.52). Both heinz and SigMod had on average better
sensitivity than the classifier trained on all the SNPs, but none had
superior specificity. However, the differences are negligible, well
within the 95% confidence interval.

Another desirable quality of an algorithm is stability (Section
\@ref(methods:comparison)). Both heinz and LEAN displayed a high
stability in our benchmark, consistently selecting the same genes and no
genes over the 5 subsamples, respectively (Figure \@ref(fig:benchmark)C). 
Conversely, the other methods displayed similarly low stabilities.

In terms of computational runtime, the fastest method was heinz (Figure
\@ref(fig:benchmark)D), which leverages on its ability to find
efficiently the solution in a few seconds. The slowest method was dmGWAS
(1 day and 17 hours on average) followed by SConES GI (1 day and 4.32
hours on average). However, including the time required to compute the gene
scores slows down considerably gene-based methods; on this
benchmark, that step took on average 1 day and 9.33 hours. Considering
that, it took 3 days and 2.4 hours on average for dmGWAS' to produce
results.

### No solution is perfect {#genesis-issues}

(ref:fig-issues-caption) Drawbacks confronted when using network guided methods. **(A)** dmGWAS solution subnetwork. Genes with a P-value \< 0.1 are highlighted in red. **(B)** Centrality degree and -log~10~ of the VEGAS P-value for the nodes in SigMod solution subnetwork. **(C)** Genomic regions where either SConES GS, GM or GI select SNPs.

```{r issues, echo=FALSE, fig.cap='(ref:fig-issues-caption)'}
knitr::include_graphics('fig/figure_2', dpi = NA)
```

In practice, and despite their similarities and their involvement in
cancer mechanisms, the solutions are remarkably different 
(Figure \@ref(fig:pearson-methods)A). That is due to the particularities
of the methods which directly or indirectly provide information about
the dataset. For instance, the fact that LEAN did not provide any
biomarkers implies that there is no gene such that both itself and its
environment are on average strongly associated with the disease.

(ref:fig-pearson-methods-caption) Pearson's correlation between the different solution subnetworks. **(A)** Correlation between selected SNPs. **(B)** Correlation between selected genes. In general, the solutions display a very low overlap.

```{r pearson-methods, echo=FALSE, fig.cap='(ref:fig-pearson-methods-caption)'}
knitr::include_graphics('fig/sfigure_6', dpi = NA)
```

In this dataset, heinz's solution is very conservative, providing a
small solution with the lowest median P-value for the subnetwork (Table
\@ref(tab:gene-solutions)). Due to this parsimonious and highly
associated solution, it was the best method to select a set of good
biomarkers for classification. (Figure \@ref(fig:benchmark)B). 
Its conservativeness stems from its
preprocessing step, which models the gene P-values as a mixture model of
a beta and a uniform distribution, controlled by an FDR parameter. Due
to the limited signal at the gene level in this dataset (Figure
\@ref(fig:snp-gene-manhattan)B), only 36 of them are retain a
positive score after applying the BUM model (Section
\@ref(methods:methods)). Hence, heinz's solution subnetwork
consists only of 4 genes, which does not provide much insight of the
biology of cancer. Importantly, it ignores genes that are strongly
associated to cancer in this dataset like *FGFR2*.

On the other end of the spectrum, we have large solutions provided by
dmGWAS, HotNet2, and SigMod. dmGWAS' subnetwork is the least associated
subnetwork on average. This is due to the greedy framework it uses,
which considers all nodes at distance 2 of the examined, and accepts
weakly associated genes if they are linked to another, strongly
associated one. This is exacerbated when the results of successive
greedy searches are aggregated, leading to a large, tightly connected
cluster of unassociated genes (Figure \@ref(fig:issues)A). 
SigMod displays the same tendency, as the
most central genes are the least associated to the disease (Figure
\@ref(fig:issues)B). This relatively low signal-to-noise ratio
combined with the large solution requires additional analyses to draw
conclusions, such as enrichment analyses. In the same line, HotNet2's
subnetwork is even harder to interpret, being composed of 440 genes
divided into 135 subnetworks. Lastly, SigMod misses some of the most
strongly associated, breast cancer susceptibility genes in the dataset,
like *FGFR2* and *TOX3*.

By virtue of using a SNP subnetwork, SConES analyzes each SNP in their
context. It therefore selects SNPs in genes none of whose interactors
are associated to the disease, as well as SNPs in non-coding regions or
in non-interacting genes. In fact, due to linkage disequilibrium, such
genes are favored by SConES, as selecting SNPs in an LD block which
overlaps with a gene favors selecting the rest of the gene. This might
explain why the GS and GM networks, heavily affected by linkage
disequilibrium, produce similar results (Figure
\@ref(fig:pearson-methods)B). On the other hand, SConES penalizes
selecting SNPs and not their neighbors. This makes it conservative
regarding SNPs with many interactions, for instance those mapped to hubs
in the PPIN. For this reason, SConES GI did not select any protein
coding gene, despite selecting similar regions as SConES GS (Figure
\@ref(fig:issues)C). In fact SConES GS and SConES GM select
regions related to breast cancer, like 16q12 (*TOX3*, Section
\@ref(results:conventional)), 3p24 (*SLC4A7*/*NEK10*,
@ahmed_newly_2009), 5p12 (*FGF10*, *MRPS30*, @quigley_5p12_2014), and
10q26 (*FGFR2*, Section \@ref(results:conventional)). On top of that only SConES
GS selects region 8q24 (*POU5F1B*, @breyer_expressed_2014). We hypothesize
that the lack of results on the PPIN network of SConES GI and LEAN due
to the same cause: the absence of joint association of a module.
Although in the case of SConES other hyperparameters could lead to a
more informative solution (e.g. lower $\lambda$, Section\@ref(methods:methods)),
it is unclear what is the best strategy
to find them. In addition, due to the iCOGS SNP array design, the genome
of GENESIS participants has not been unbiasedly surveyed: some regions
are fine-mapped --- which might distort gene structure in GM and GI
networks --- while others are under studied --- hurting the accuracy
with which the GS network captures the genome structure.

### Aggregating solutions provides insights into the biology of cancer {#results:consensus}

(ref:fig-consensus-caption) Consensus subnetwork on GENESIS (Section \@ref(methods:consensus)). Each node is represented by a pie chart, which shows the methods that selected it. The labeled genes have a VEGAS2v2 P-value \< 0.001 and/or are known breast cancer susceptibility genes (colored in pink).

```{r consensus, echo=FALSE, fig.cap='(ref:fig-consensus-caption)'}
knitr::include_graphics('fig/figure_3', dpi = NA)
```

To leverage on the strengths of each of the methods and compensate their
respective weaknesses, we built a consensus subnetwork that captures the
mechanisms most shared among the solution subnetworks (Section
\@ref(methods:consensus)). The consensus subnetwork (Figure
\@ref(fig:consensus)) contains 93 genes and is enriched in breast
cancer susceptibility genes (Fisher's exact test P-value = 7.8 10^-5^).
Due to the limited overlap between methods, only 20 genes were common to
more than two of them (Figure \@ref(fig:consensus-stats)A). Encouragingly, 
the more methods selected a gene, the higher its association was (Figure
\@ref(fig:consensus-stats)B). Globally, a GO enrichment shows the
involvement of two cellular processes: unfolded protein binding, and
structural constituent of cytoskeleton (adjusted P-values of 0.001,
0.001, respectively), which were already observed in different solutions
(Section \@ref(results:separate-networks)). Remarkably, many of the
selected genes are related to mitochondrial translation. For instance,
MRPS30 (VEGAS P-value = 0.001), encodes a mitochondrial ribosomal
protein and was also linked to breast cancer susceptibility
[@quigley_5p12_2014]. Albeit disconnected from MRPS30, the consensus
network includes a 2-node subnetwork composed of two mitochondrial
ribosomal protein (MRPS31 - VEGAS P-value = 7.67 10^-3^ - and MRPS18B -
VEGAS P-value = 7.92 10^-3^), which suggests an involvement of
mitochondrial ribosomes in carcinogenesis.

We also examined the topological properties of the nodes. The genes in
the consensus network have higher betweenness centrality than the rest
of the genes (Wilcoxon rank-sum test P-value = 4.29 10^-18^).
Interestingly, within genes in the consensus network, cancer genes are
as central as non-cancer genes (Wilcoxon rank-sum test P-value = 0.57).
Centrality, however, is weakly anti-correlated with association to the
disease (Pearson correlation coefficient = -0.26, Figure
\@ref(fig:consensus-stats)D), which suggests that some highly
central genes were selected because they were on the shortest path
between two highly associated genes. In view of this, we hypothesize
that highly central genes might contribute to the heritability through
consistent alterations of their neighborhood, consistent with the
omnigenic model of disease [@boyle_expanded_2017]. For instance, the
most central node in the consensus network is *COPS5* 
(Figure \@ref(fig:consensus-names)), a gene related to multiple hallmarks
of cancer and which is overexpressed in multiple tumors, including
breast and ovarian cancer [@liu_jab1_cops5_2018]. Despite its lack of
association in GENESIS (VEGAS P-value = 0.22), its neighbors in the
consensus subnetwork have consistently low P-values (median VEGAS
P-value = 0.006).

(ref:fig-consensus-names-caption) Consensus subnetwork on GENESIS (Section \@ref(methods:consensus)). **(A)** Each node is represented by a pie chart, which shows the methods that selected it. The labeled genes have a VEGAS2v2 P-value \< 0.001 and/or are known breast cancer susceptibility genes (colored in pink). This panel is identical to Figure \@ref(fig:consensus). **(B)** The name of every gene is indicated.

```{r consensus-names, echo=FALSE, fig.cap='(ref:fig-consensus-names-caption)', out.height='75%', fig.align='center'}
knitr::include_graphics('fig/sfigure_7', dpi = NA)
```

The consensus subnetwork is not completely connected: out of the 93
genes, the largest connected subnetwork includes only 49. A GO
enrichment analysis showed that this component is related to three major
cellular processes: unfolded protein binding, structural constituent of
cytoskeleton, and poly(U) RNA binding (adjusted P-values of 0.01, 0.04,
and 0.04, respectively). We found support in the literature of the
involvement of each of these functions in the development of cancer, as
discussed next. The consensus network also contains a protein directly
involved in caspase-mediated apoptosis, *CASP8* (VEGAS P-value = 1.95
10^-4^). This is related to the enriched activity, *unfolded protein
binding*, which inhibits caspase-dependent apoptosis, raising the
chances of developing cancer [@calderwood_heat_2016]. It involves three
Hsp70 chaperones of the consensus subnetwork: HSPA1A, HSPA1B, and
HSPA1L. They genes encoding these proteins are all near each other at
6p21. In fact, out of the 22 SNPs that map to any of these three genes,
9 map to all of them, and 4 to two, making hard to disentangle their
association. *HSPA1A* was the most strongly associated one (VEGAS P-value
= 8.37 10^-4^). Remarkably, 14 of the 93 genes are in subnetworks of
size 1 (isolated) or 2, as they do not have a consistently altered
neighborhood. One of them is the well-known breast cancer susceptibility
gene *FGFR2* (Section \@ref(results:conventional)
). Another one is the also well-known
*SLC4A7* gene (VEGAS P-value = 2.70 10^5^), which encodes a sodium
bicarbonate cotransporter. The genomic region containing both *SLC4A7*
and nearby gene *NEK10* (VEGAS P-value = 1.56 10^-5^) have been
consistently associated with breast cancer susceptibility
[@ahmed_newly_2009]. *NEK10* is a gene that might be involved in
cell-cycle control, but it is absent from the PPIN and hence it could
not be studied by gene methods. Despite that, the fact that both dmGWAS,
HotNet2 and SigMod link *SLC4A7* in their different subnetwork supports
the notion that this gene is the responsible for breast cancer
susceptibility.

## Discussion {#genewa-discussion}

In this chapter we evaluate the viability of systems biology approaches
to GWAS, and examine a GWAS dataset on familial breast cancer focused on
*BRCA1/2* negative French women. Systems biology addresses two of the
largest GWAS issues: interpretability and an overly conservative
statistical framework that hinders discovery. This is achieved by
considering the biological context of each of the genes and SNPs. Based
on divergent considerations of what the desired set of biomarkers is,
several methods for network-guided biomarker discovery have been
proposed. We reviewed the performance of six of them on GWAS. Despite
their differences, most of them produced a relevant subset of
biomarkers, recovering known familial breast cancer genes. We also
discuss the limitations of such analyses, related to the lack of known
interactions around some genes. A crucial step for the gene based
methods is the computation of the gene score. In this chapter we used
VEGAS2v2 [@mishra_vegas2:_2015] due to the flexibility it offers to use
user-specified gene annotations. However, it presents known problems
(selection of an appropriate percentage of top SNPs, long runtimes and
P-value precision limited to the number of permutations
[@nakka_gene_2016]), other algorithms might have more statistical power.

The network methods we studied differ in what the optimal solution
subnetwork looks like. On the one hand, SConES and heinz prefer small
highly associated solutions. On the other hand, SigMod and dmGWAS
gravitate towards larger, less associated solutions which provide a wide
overview of the biological context. While the former provide a reduced
set of biomarkers, the latter deepen our understanding of the disease
and provide biological hypotheses. Yet, they are not exempt of limitations.
dmGWAS and SigMod's solution's size require further analyses, which risk
oversimplifying their richness. Also, incautious practitioners might be
misled by some genes, which are very central in the solution
subnetworks, while being weakly associated. Nonetheless, they are pushed
into the solution by their privileged topological properties. On the
other end, conservative solutions, like SConES GI and heinz might not
shed much light on the etiology of the disease.

To overcome the problems posed by the individual methods while
exploiting their strengths, we propose combining them into a consensus
subnetwork. We use a straightforward aggregation to generate it,
including any node that was recovered by at least two methods. The
resulting network is a synthesis of the altered mechanism: it is smaller
than the largest solutions (SigMod and dmGWAS), which makes it more
manageable, and includes the majority of the strongly associated smaller
solutions (SConES and heinz). The consensus subnetwork captures
mechanisms and genes known to be related to cancer, recovering known
breast cancer susceptibility genes as well as genome regions associated
to breast cancer susceptibility. However, thanks to its smaller size and
its network structure, it provides compelling hypotheses of
non-canonical mechanisms involved in carcinogenesis, like mitochondrial
translation and chaperone activity.

The strength of network-based analyses comes from leveraging prior
knowledge to boost discovery. In consequence, they show their
shortcomings in front of understudied genes, especially those not in the
network. Out of the 32\,767 genes that we can map the genotyped SNPs to,
60.7% (19\,887) are not in the protein-protein interaction network. The
majority of those (14\,660) are non-coding genes, mainly lncRNA, miRNA,
and snRNA (Figure \@ref(fig:biotypes-excluded)). The importance of these genes,
like *CASC16*, is highlighted in Section \@ref(results:conventional). 
Among the excluded protein-coding
genes we find genes like *NEK10* (P-value 1.6 10^-5^) or *POU5F1B*, both
linked to breast cancer susceptibility [@ahmed_newly_2009]. However, on
average protein-coding genes absent from the PPIN are less associated
with this phenotype (Wilcoxon rank-sum P-value = 2.79 10^-8^, median
P-values of 0.43 and 0.47). As we are using interactions from
high-throughput experiments, such difference cannot be due to well-known
genes having more known interactions. As disease genes tend to be more
central [@pinero_uncovering_2016], we hypothesize that it is due to
interactions between central genes being more likely. It is worth noting
that network approaches that do not use PPIs, like SConES GS and GM, did
recover SNPs in *NEK10* and *CASC16*. Lastly, all the methods rely
heavily on how SNPs are mapped to genes. In Section
\@ref(results:conventional) we highlight ambiguities that appear
when genes overlap or are in linkage disequilibrium.

```{r biotypes-excluded, echo=FALSE, fig.cap='Biotypes of genes from the annotation that are not present in the HINT protein-protein interaction network.'}
knitr::include_graphics('fig/sfigure_9', dpi = NA)
```

As not all databases compile the same interactions, the choice of the
PPIN determines the final output. In this work we used exclusively
interactions from HINT from high-throughput experiments. This responds
to concerns of some authors about biases introduced by adding
interactions coming from targeted studies in the literature
[@cai_broker_2010; @das_hint_2012] where a "rich getting richer"
phenomenon is observed: popular genes have a higher proportion of their
interactions described. On the other hand, @huang_systematic_2018 found that the best
predictor of the performance of a network for disease gene discovery is
the size of the network. This also supports
using the largest amount of interactions. To clarify their impact on
this study, we compared the impact of using only physical interactions
from high-throughput experiment versus interactions from both
high-throughput and the literature (Section \@ref(methods:gene-network)). We 
conclude that for most of the methods a larger network did not greatly impact 
the size or the stability of the solution, the classification accuracy, or the 
runtime (Figure \@ref(fig:lc-ht-comparison)).

(ref:fig-lc-ht-comparison-caption) Comparison of benchmark on high-throughput interactions to benchmark on both high-throughput and literature curated interactions. Grey lines represent no change between the benchmarks (1 for ratios, 0 for differences). **(A)** Ratios of the selected features between both benchmarks and of the active set. **(B)** Shifts in sensitivity and specificity. **(C)** Shift in Pearson's correlation between benchmarks. **(D)** Ratio between the runtimes of the benchmarks. For gene network-based methods, inverted triangles represent the ratio of runtimes of the algorithms themselves, and circles the total time, which includes the algorithm themselves and the additional 119\,980 seconds (1 day and 9.33 hours) which took VEGAS2v2 on average to compute the gene scores from SNP summary statistics. In general, adding additional interactions slightly improves the stability of the solution, but increases the solution size, has mixed effects on the sensitivity and specificity, and impacts negatively the required runtime of the algorithms.

```{r lc-ht-comparison, echo=FALSE, fig.cap='(ref:fig-lc-ht-comparison-caption)'}
knitr::include_graphics('fig/sfigure_10', dpi = NA)
```

In order to produce the consensus network, we had to face the different
interfaces, preprocessing steps, and unexpected behaviors of the various
methods. To facilitate that other authors apply them to new datasets and
aggregate their solutions, we built six nextflow pipelines
[@di_tommaso_nextflow_2017] with a consistent interface and, whenever
possible, parallelized computation. They are available on GitHub:
<https://github.com/hclimente/gwas-tools>.