# Boosting interpretability and statistical power in epistasis detection by using prior biological knowledge {#epi-snp2gene}

The contents of this chapter are based on *Diane Duroux, Héctor Climente-González, Lars Wienbrandt, David Ellinghaus, Chloé-Agathe Azencott and Kristel Van Steen*, Boosting interpretability and statistical power in epistasis detection by using prior biological knowledge (manuscript in preparation). Parts of this work were presented in IGES 2019 in the poster *Diane Duroux, Héctor Climente-González, Lars Wienbrandt, David Ellinghaus, Chloé-Agathe Azencott and Kristel Van Steen*, Improving efficiency in epistasis detection with a gene-based analysis using functional filters.

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
library(magrittr)
library(kableExtra)

cat("\\begin{chapabstract}{Abstract:}
Detecting epistatic interactions at the gene level is essential to understanding the biological mechanisms of complex diseases. Unfortunately, genome-wide interaction association studies (GWAIS) involve many statistical challenges that make such detection hard. This chapter considers three steps to analyze the conversion of SNP-SNP interactions into a gene network through the integration of biological knowledge. First, we investigate the use of known gene-gene interactions to focus on the most promising gene-pairs. Second, we explore several functional SNP-to-gene mappings to map SNPs to these gene interactions, including eQTL and chromatin-based, and filter the GWAIS data accordingly. Third, we integrate SNP-pairs statistics into gene-pairs statistics using the adaptive truncated product method (ATPM). ATPM does not require parameters or a known null distribution, and it is fast to compute. We apply different variants of these three steps to a GWAS inflammatory bowel disease (IBD) dataset, and compare the results. Overall, knowledge integration decreases the number of tests performed, reducing computational time and boosting statistical power, and provides interpretability. We validate through a permutation analysis that type I error is under control. The different protocols produce different results, which highlight different mechanisms implicated in IBD while, at the same time, overlapping with the known biology of the disease. Using eQTLs results in more interactions than the standard approach, where all possible SNP pairs are investigated and SNPs are mapped to genes based on their position on the genetic sequence, albeit fewer significant gene-pairs. This pipeline not only detects interactions at the gene level, but also gives a global visualization that identifies neighbors, hubs and highly connected substructures.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}

\\end{chapabstract}")
```

## Introduction

In this chapter we study the detection of gene-level epistasis from
SNP-level epistasis to improve the interpretability of GWAIS. Our
analysis compares ways of converting statistical epistasis at the SNP
level into a gene-based statistical epistasis network. First, we
investigate different functional filters [@ma2015biological] and
SNP-to-gene mapping functions.
Second, as a bijectivity issue arises because a SNP can be mapped on
several genes, we investigate the use of current knowledge on gene-gene
interaction to focus on the most promising gene-pairs. We study whether
the epistatic interactions driving the phenotype are likely to be
currently in existing databases. Third, we used the adaptive truncated
product method to estimate gene-pairs significance.

## Materials and methods

### Dataset and initial quality control {#snp2gene-qc}

We investigated the IIBDGC dataset, described in 
Section \@ref(iibdgc-dataset), whose large sample size helps to overcome the 
issue of reduced statistical power common in GWAIS. 
We performed a quality control as in @ellinghaus2016analysis, reducing the 
number of SNPs from 196\,524 to 130\,071.

The IIBDGC dataset aggregates different cohorts, and contains
confounding population structure (Section \@ref(pop-structure)). PLINK
[@purcell2007plink] cannot take covariates in the logistic regression used to
detect epistasis (Section \@ref(methods:snp-epistasis)). In consequence, we adjusted
phenotypes to account for population structure using the top 7 principal
components as in @ellinghaus2016analysis. Essentially, we derive adjusted 
phenotypes from the logistic regression model by subtracting model-fitted 
values from observed phenotype values (i.e. response residuals).

### Gene interaction detection procedure {#methods:pipeline}

As we describe below, we applied four different functional filters to
the dataset. The functional filter used known interactions between
genes, and the three different ways of mapping SNPs to genes described in Section
\@ref(intro-interpretability), and hence, to
these interactions. In essence, in each of the filtered datasets we only tested
the interactions between SNPs which can be mapped in a particular way to
a pair of interacting genes. The resulting four datasets
(including an unfiltered dataset) were analyzed separately. For
convenience, we will refer to their associated analyses using the terms
*Standard*, *physical*, *eQTL* and *chromatin*. For each dataset, the
entire pipeline described below is applied and the four obtained outputs
are subsequently compared. An overview of the whole pipeline is
available in Figure \@ref(fig:pipeline).

(ref:fig-pipeline-caption) Overview of the gene-gene interaction detection procedure. The whole protocol is described in Section \@ref(methods:pipeline).

```{r pipeline, echo=FALSE, fig.cap='(ref:fig-pipeline-caption)'}
knitr::include_graphics('fig/pipeline2.PNG', dpi = NA)
```

#### Functional SNP pre-filtering {#methods:prefiltering}

The initial step of the protocol is a functional SNP pre-filtering,
which has three stages. First, we mapped the SNPs in the dataset to
genes using FUMA [@watanabe_functional_2017]. FUMA is a post-GWAS
annotation tool. Its SNP2GENE function takes GWAS summary statistics and
maps significant SNPs to genes according to both physical and functional
criteria specified by the user. We created an artificial input where
every SNP is significant in order to perform such mapping on all the
SNPs. We performed three SNP-gene mappings using SNP2GENE: physical,
eQTL and 3D chromatin interaction. In the physical mapping, we mapped a
SNP to a gene when the former was within the boundaries of the latter
$\pm$ 10 kb. The eQTL mapping uses eQTLs obtained
from GTEx [@gtex_consortium_genetic_2017]. We mapped an eQTL SNP to its target gene when the
association P-value was significant in any tissue ($\text{FDR} < 0.05$).
Lastly, in the 3D chromatin interaction mapping, we mapped a SNP to a
gene when a contact had been observed between the former the region
around the latter's promoter (250 bp upstream and 500 bp downstream from
the transcription start site) in any of the Hi-C datasets included in
FUMA ($\text{FDR} < 10^{-6}$). The chromatin mapping might contain new,
undiscovered, regulatory variants which, as eQTL, regulate the expression
of a gene.

Second, after obtaining the SNP-to-gene mappings, we used Biofilter 2.4
[@pendergrass_genomic_2013] to obtain the candidate gene-pairs
subsequently investigated for epistasis evidence. Biofilter generates
pairs of genes with evidence of co-function across multiple publicly
available biological databases. It includes genomic locations of SNPs
and genes, as well as known relationships among genes and proteins such
as interaction pairs, pathways and ontological categories.
Notably, it does not use trait information.
Specifically, we considered only pairs of genes for which both genes
could be mapped, using any of the mappings, to a SNP in our GWAS
dataset. We used only gene pairs supported by evidence in at least 2
databases. When the two SNPs of a pair were located in the HLA region,
we removed the pair, as this complex genomic region is currently not
well understood. Additionally, we removed self-interactions, as
detection of within-gene epistasis requires special considerations and
is beyond the scope of this paper.

Lastly, we filtered the datasets again to explore exclusively
interactions between SNPs mapped to genes known to interact, according
to Biofilter. For that purpose we first converted the Biofilter
gene-pair models into SNP-pair models separately via each of the FUMA
SNP-gene mappings described above. Then, from these SNP pairs sets, we
built the four datasets enumerated above: one without any filter (*Standard*); 
and one for each SNP to gene mapping (*Physical*, *eQTL*,
*Chromatin*). It is worth pointing out that SNP-pair models
were also built based exclusively on the corresponding mapping e.g.
*physical* contains exclusively pairs of SNPs which can be associated to
pairs of genes via a physical mapping. This helps interpretability and
keeps the number of tested interactions under control.

#### Post-filtering quality control

Additional quality controls were performed on each of the four generated
datasets. As motivated in @gusareva_practical_2014, only common variants
($\text{MAF} > 5$%) and in Hardy--Weinberg equilibrium ($\text{P-value} > 0.001$) are
considered. Also, we pruned SNPs that are in linkage equilibrium 
($R^{2} > 0.75$). Lastly all risk SNP described in @liu2015association were 
included.

#### SNP-level epistasis detection and multiple test correction {#methods:snp-epistasis}

We used PLINK 1.9 to detect epistasis through a linear regression on the adjusted phenotypes:

$$Y = \beta_0 + \beta_1 g_A + \beta_2 g_B + \beta_3 g_A g_B,$$

where $g_A$ and $g_B$ are the genotypes under additive encoding for SNPs
A and B respectively; $\beta_0$, $\beta_1$, $\beta_2$, $\beta_3$, are
the regression coefficients. PLINK performs a statistical test to
evaluate if $\beta_3 \neq 0$. Crucially, it only returns SNP-pairs
with a P-value lower than a specified threshold. We used the default
0.0001.

We only considered pairs of SNPs not in strong LD ($R^{2}<0.75$) and
that could be mapped to the corresponding SNP-model obtained from
Biofilter, with self-gene interactions removed (Section
\@ref(methods:prefiltering)). This impacted the total number of
statistical tests. 

To correctly account for multiple testing, the P-value threshold of
significance had to be dataset-dependent as the number of tested SNP
pairs changed from dataset to dataset (Section
\@ref(methods:prefiltering)). We obtained the threshold through a
permutation analysis as in @hemani_detection_2014. In essence,
for each dataset, we permuted the phenotypes 400 times and measured SNP
association as above. This produced a null distribution of the extreme
P-values for this number of tests given the LD structure. For each
dataset, we took the most extreme P-value from each of the 400
permutations and set the threshold for 5% family-wise error rate
(FWER) to be the 95% percentile of these most extreme P-values.

#### From SNP-level to gene-level epistasis {#methods:gene-networks}

Next we converted SNP pairs into gene-pairs using their respective FUMA 
and Biofilter mappings (Section \@ref(methods:prefiltering)). 
The exception was the *Standard*
dataset, where SNPs are physically mapped to genes, and no Biofilter
restriction on which pairs of genes can interact was applied. Still,
self-interactions were removed.

Then, we computed gene-level statistics from the respective SNP-level
statistics of the involved SNPs. In this regard, all $N$ pairs of SNPs
mapped to a gene pair are taken as a set of tests on the same global
null hypothesis $H_{0i}$, where $i=1,2,\dots ,N$. @zaykin2002truncated
developed the truncated product method (TPM) as a method to combine
P-values on a same global hypothesis. It does so by computing the
statistic $W(\tau)=\prod_{i=1}^{N} p_i^{I(p_{i}\le\tau)}$ where
$I(\cdot)$ is the indicator function and $\tau$ is the truncation point.
A P-value $\hat{s}(\tau)$ can be the estimated for a given $W(\tau)$.
TPM was interesting since we did not not have
P-values for every SNP pair but for the most strongly associated ones
only (Section \@ref(methods:snp-epistasis)). However, TPM requires setting the
truncation point $\tau$, a parameter that is arbitrary and might be
gene-pair specific. On top of that, the null distribution of $W(\tau)$
is unknown when P-values are correlated, as is the case. To solve these
problems the adaptive truncated product method (ATPM) was proposed
[@sheng2013adaptive]. ATPM explores different $\tau$, choosing the one
that produces the minimum P-value $\hat{s}(\tau)$. We estimated the
distribution of the ATPM using permutations as in @ge2003resampling.
Specifically, we created $B = 999$ permuted datasets by permuting the
phenotype vector. Based on these $b^{th}$ permuted dataset,
$1 \leq b \leq B$, we perform the N individual tests. We used three
values for $\tau$ (0.001, 0.01, 0.05) and set the significance level
$\alpha=0.05$. The specific procedure goes as follows:

1.  For each gene-pair in the output of the original data analysis,
    based on $p{_1}^{(b)},...,p{_N}^{(b)},0\le b\le B$, calculate the
    truncated product statistics for each candidate truncation threshold
    for the original data and B permuted datasets.

2.  Based on $W(\tau)_b$, use Ge's algorithm to obtain the estimated
    P-value
    $\hat{s}_{k}^{b}=\frac{\sum_{l=0}^B I(W(\tau)_b \geq W(\tau)_l)}{B+1}$,
    $1 \le k \le K$, $0 \le b \le B$

3.  Calculate $M_{b}, 0 \le b \le B$, as
    $M{_b}=min_{1 \le k \le }\hat{s}_{k}^{b}$

4.  The adjusted P-value for the adaptive truncated product statistic
    $M$ is estimated as
    $P_{minP^{(b)}}=\frac{\sum_{b=0}^B I(M \geq M_b)}{B+1}$.

5.  We reject the joint null hypothesis if the adjusted P-value is
    smaller than the significance level $\alpha$.

## Results

### Chromatin contacts map more SNPs per gene than other mappings {#results-power-gain}

We considered three procedures to map SNPs to their gene, as a proxy for
their link to functionality (Section
\@ref(methods:prefiltering)): *physical*, *eQTL* and *chromatin*.
*Chromatin* produced the largest number of mappings (2\,394\,589), an
order of magnitude more than *eQTL* (411\,120) and *physical* (174\,879)
(Table \@ref(tab:datasets-snps)). Similarly, *chromatin* has 
the largest number of SNPs mapped to an individual gene, followed by *eQTL* and
*physical* (Figure \@ref(fig:snps-per-mapping)A). Nonetheless, 
different genes had very unequal contributions from each of the mappings 
(Figure \@ref(fig:snps-per-mapping)B). This is consistent with 
the stark differences between the ranking of genes according to the number of 
SNPs mapped them (Figures \@ref(fig:snps-per-mapping)C, D and E): 
in general, the genes with the most SNPs mapped using the eQTL mapping had 
relatively few SNPs mapped in the chromatin mapping, and viceversa.

(ref:fig-snps-per-mapping-caption) **(A)** Number of SNPs per gene for each of the three mappings described in Section \@ref(methods:prefiltering). Outliers are not displayed to facilitate visualization. **(B)** Ranking of genes with most SNPs mapped using any of the mappings, colored by mapping. Only genes with more than 100 SNPs mapped to it are displayed. **(C,D,E)** Comparison between the rank of each gene according to the number of SNPs mapped to it using each mapping.

```{r snps-per-mapping, echo=FALSE, fig.cap='(ref:fig-snps-per-mapping-caption)'}
knitr::include_graphics('fig/mappings_per_gene.png', dpi = NA)
```

The number of mappings is directly linked the number of SNPs and 
interactions tested per dataset (Table \@ref(tab:datasets-snps)). As it can be
observed, restricting our search exclusively to Biofilter-plausible interactions
leads to an increase in statistical power with respect to the *Standard* 
protocol. Specifically, the number of tests are between 1 and 2 orders of 
magnitude smaller.

```{r datasets-snps, echo=FALSE}
datasets_snps <- rbind(
      c('# SNPs', 38225, 16417, 30146, 16652),
      c('# SNP-gene mappings', format(c(NA, 1.7e5, 2.4e6, 4.1e5), scientific = T)),
      c('# tests', format(c(7.2e8, 4.6e6, 8.9e7, 2.2e7)))
  )
colnames(datasets_snps) <- c('', 'Standard', 'Physical' ,'Chromatin', 'eQTL')

kable(datasets_snps, 
      caption = 'Properties of the different SNP-gene mappings and the filtered datasets.', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = "striped")
```

### The *Physical* protocol does not recover any SNP interaction

We searched SNP epistasis in the four datasets (Section 
\@ref(methods:snp-epistasis)). The different epistatic SNP-SNP networks are 
described on Table \@ref(tab:snp-network) and Figure 
\@ref(fig:snp-networks). Strikingly, while the *Standard* protocol generated the 
largest network (55 nodes), the *Chromatin* was the largest by number of 
interactions (64). The *Physical* protocol produced no significant pairs.

```{r snp-network, echo=FALSE}
snp_networks <- rbind(
      c('# significant pairs', 57, 0, 19, 64),
      c('# nodes', 55, 0, 20, 46),
      c('# connected components', 12, NA, 5, 6),
      c('Size of the largest component', 25, NA, 11, 17),
      c('Average degree', 2.07, NA, 1.90, 2.78)
  )
colnames(snp_networks) <- c('', 'Standard', 'Physical' ,'Chromatin', 'eQTL')

kable(snp_networks, 
      caption = 'Properties of the SNP networks from the different datasets.',
      booktabs = TRUE) %>% 
  kable_styling(latex_options = "striped")
```

(ref:fig-snp-networks-caption) SNP-level epistasis networks for *Standard* (orange), *eQTL* (green), and *Chromatin* (violet) (Sections \@ref(methods:prefiltering) and \@ref(methods:snp-epistasis)). The *Physical* dataset is absent, as no SNP pairs were significant.

```{r snp-networks, echo=FALSE, fig.cap='(ref:fig-snp-networks-caption)'}
knitr::include_graphics('fig/snp_networks.pdf', dpi = NA)
```

### Gene-level network {#results:gene-networks}

We converted the aforementioned SNP-pair networks into gene-pair epistasis
networks, estimating their significance through ATPM (Section 
\@ref(methods:gene-networks)). Most of the SNP-pairs mapped to exclusively one 
gene pair in *eQTL* and *Standard*, removing possible sources of ambivalence 
(Figure \@ref(fig:sign-snps-per-gene-and-viceversa)A). That was 
not the case under the chromatin mapping, where it was more common for the same
SNP pair to map to different gene pairs. We then compared the relationship
between significant gene-pairs and the number of significant SNP pairs that map
to them (Figure \@ref(fig:sign-snps-per-gene-and-viceversa)B). 
Interestingly, we observe that most significant gene pairs are supported by 
relatively small number of SNPs: either few in number, or few with respect to 
the total number of SNP pairs for that gene pair.

(ref:fig-sign-snps-per-gene-and-viceversa-caption) Relationship between the number of significant SNP pairs and of significant gene pairs. **(A)** Histogram of the number of significant gene pairs mapped to the same SNP pair. **(B)** Relationship between the total number of SNP pairs mapped to the same gene pair (y-axis), and the percentage of all significant SNP-pairs between all the SNP-pairs mapped to the same gene (x-axis). Data points are semi-transparent, so multiple points stacked result in a darker shade.

```{r sign-snps-per-gene-and-viceversa, echo=FALSE, fig.cap='(ref:fig-sign-snps-per-gene-and-viceversa-caption)'}
knitr::include_graphics('fig/sign_snps_per_gene_and_viceversa.pdf', dpi = NA)
```

We built an epistatic gene network from the significant gene pairs
(Methods \@ref(methods:gene-networks)), shown in Figure \@ref(fig:gene-networks)
and Table \@ref(tab:gene-networks-stats). Overall, the *Standard* protocol still
produces the largest network, and contains more connected components and
significant gene-pairs. On the other hand, *Chromatin* and *eQTL* mappings
produce similar networks in terms of sizes, number of gene-pairs and
connected components. However, both *Chromatin* and *eQTL*'s networks are
notably smaller than *Standard*'s (11 and 10 nodes versus 29,
respectively). 

*Standard*'s nodes are proportionally more clustered in
connected components, while most *eQTL* and *chromatin*'s connected
components are composed of only a pair of genes. Although this might
reveal the affection of a common mechanism, it is likely a result of the
overlap in the genome of multiple genes, which are mapped to highly
overlapping sets of SNP.

```{r gene-networks-stats, echo=FALSE}
gene_networks <- rbind(
      c('# significant pairs', 26, 0, 5, 7),
      c('# nodes', 29, 0, 10, 11),
      c('# connected components', 8, NA, 5, 5),
      c('Size of the largest component', 6, NA, 2, 3),
      c('Average degree', 1.79, NA, 1, 1.27)
  )
colnames(gene_networks) <- c('', 'Standard', 'Physical' ,'Chromatin', 'eQTL')

kable(gene_networks, 
      caption = 'Properties of the gene networks from the different datasets.', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = "striped")
```

(ref:fig-gene-networks-caption) Gene-level epistasis networks for *Standard* (orange), *eQTL* (green), and *Chromatin* (violet) (Section \@ref(methods:prefiltering)).

```{r gene-networks, echo=FALSE, fig.cap='(ref:fig-gene-networks-caption)'}
knitr::include_graphics('fig/gene_networks.pdf', dpi = NA)
```

A hub is node with a number of links that greatly exceeds the average.
For this application, we define a hub as a node having a degree strictly
superior to three. Only the Standard process contains such hubs: *P4HA2*,
*NKD1*, *RNU4ATAC4P*, *C1orf141*, *IL12RB2*, *IL23R* and *USP4*.

Jointly, 38 significant gene pairs are involved in at least one method,
involving 46 unique genes. Seven chromosome are involved in epistasis.
Notably, 39% of the epistatic genes are located in chromosome 3 and 22%
in chromosome 10.

### *Chromatin* and *Standard* mappings partially replicate previous studies on IBD {#results-ibd-soa}

Several genetic studies studying epistasis on IBD have been conducted
[@lin2017genetic; @vermeire2004genome; @pedros2015epistatic; @mcgovern2009genetic; @glas2009novel; @lin2013nod2].
We compared them to our results at the gene level, the minimal
functional unit at which we expect genetic studies on different
populations to converge. For instance several studies showed epistatic
alterations involving interleukins, like *IL-10* [@lin2017genetic],
*IL-17* and *IL-23* [@mcgovern2009genetic], and *IL-2/IL-21* and
*IL-23R* [@glas2009novel]. Encouragingly, *Standard*'s results include
interactions involving both *IL-10* and *IL-23*, although we do not find
support for the specific interactions described in the aforementioned
studies. In fact, the *Standard* protocol highlights the relevance of
interleukins as hubs (Section \@ref(results:gene-networks)). Out of the five 
gene interactions retrieved in *Chromatin* pipeline, three of them involve at 
least one interleukin. @lin2013nod2 detected interactions involving *NOD2*, 
with both *IL-23R* and other genes. Our *Standard* protocol also detects two
potentially new epistasis interactions involving *NOD2*.

### The type I error of the protocol is controlled

The "multi-stage" nature of the protocol presented in Section
\@ref(methods:pipeline) required controlling the type I error. For
that purpose, we performed 1\,000 permutation analyses for each of the
four datasets, permuting the phenotypes and running the entire protocol
to detect significant gene pairs. When at least one significant
gene-pair is observed in a permutation, that permutation is considered a
false positive (FP). This allowed to compute the type I error rate as
$\frac{ \text{\# FP}}{1000}$. We observed that the type I error was
under control for all four datasets (3.6%, 3.7%, 6.1%, and 3.9% for Standard, 
Physical, Chromatin, and eQTL, respectively).

## Discussion

In this chapter we explore protocols for functional filtering for
epistasis detection on an IBD dataset (Section
\@ref(methods:pipeline)). This is expected to bring two
advantages. The first one is an increase in statistical power.
The high dimensionality of GWAIS data requires a conservative
multiple testing correction and limits the detection of epistasis with low effect
sizes. The proposed protocol tackles this issue, while controlling for
type I error. It does so by limiting the number of tests by filtering
the dataset with functional filters. As we observe in Section 
\@ref(results-power-gain), the reduction is notable. The second advantage is an
improvement of the interpretability of the results, by examining only
statistical interactions that map to a known biological interaction. As shown
in Section \@ref(results-ibd-soa), the proposed eQTL and, specially, chromatin 
mappings provide results which match the biology of IBD, while corresponding 
to known interactions. On the other hand, the Standard protocol detects 
multiple interactions that are hard to interpret. For instance, several 
interactions involve RNA genes of unknown function (e.g. *LOC101927272* or 
*LINC02178*). Hence, our results stress the relevance of regulatory variants
in susceptibility to IBD.

In this chapter, we aim at developing a set of guidelines for the detection of 
gene epistasis, with an application to IBD. Nonetheless, epistasis detection 
at the gene level still requires making many
choices which were out of the scope of this work. One instance is the
choice of encoding for the genotypes. In this work we used the additive
encoding, which can lead to an increased false positive rate
[@van2019increase]. Also we focused on linear regression as epistasis
detection algorithm, as it accepts a continuous outcome variable,
corrects for main effects, and is computationally efficient. However,
other algorithms which share these properties but make different 
assumptions about epistasis (like the ones used in Chapter \@ref(epi-network)), 
would have been suitable as well. 

With regards to the aspects of GWAIS that are within the scope of this work,
it would interesting for the community for us 
to provide set of recommendations based on our experience, and the results
justifying each of them. Before we reach that point, a few extra experiments 
are required.

Our current protocol produces compelling hypotheses,
and shows the benefits of functional filtering with regards to statistical power and 
interpretability. However, the multi-stage nature of the process makes it
impossible to find out what each of the step brings. For instance, if the
detected interactions in any dataset are just a subset of the interactions that could be 
found without filtering out the interactions not in Biofilter; if they are
a subset of the interactions that could be found by a joint *physical+chromatin+eQTL*
mapping; or how often are interactions between genes mapped to SNPs through 
different mechanisms (e.g. a gene regulated by eQTL and a gene physically mapped). 
The answers to such questions are relevant to the community, 
and cannot be answered without isolating their effect from the Biofilter 
interactions. In other words, it would be useful to just map the results of a 
conventional GWAIS result, to observe how they differ from a conventional 
mapping. Related to this point, risk SNPs from GWAS are often located
in chromatin that is active in the tissues involved in the disease 
[@boyle_expanded_2017]. Hence, the presented protocol might lead to the most biologically
plausible epistatic interactions while boosting the power if it focused exclusively
on mappings the eQTLs and chromatin mappings obtained in the tissues relevant 
to IBD (intestines and leukocytes).

It would also be interesting to explore alternative sources of known
interactions. In this chapter we worked
exclusively on interactions from Biofilter, which compiles multiple databases.
The database that Biofilter built contained 37\,266 interactions. This
is notably smaller than other gene interaction databases, like HINT 
(@das_hint_2012, 173\,797 interactions, used in Chapter \@ref(genesis)), or 
STRING (@szklarczyk_string_2019, 11\,759\,455 interactions). Hence changing
databases might result in more, equally interpretable, detected interactions.

Pathway enrichment analyses can inform about the broader framework 
in which the observed gene epistasis occurs. I would like to adapt the "network neighborhood 
search" procedure from @yip2018network to build appropriate gene sets. In 
summary, given reference biological network (e.g. the Biofilter network), a 
gene set for a given pair of genes is obtained in three steps:

1. Remove the edge connecting the two genes in the reference network.
2. Find the shortest path between them in the reference network.
3. Create a gene set including the initial two genes and all the genes in the
shortest path that are part of the epistasis network as well.

Another important question is which null hypothesis
to test in the pathway enrichment analyses. 
In this regard, the literature often distinguishes two kinds of test:
self-contained and competitive [@wang_gene_2011]. As in our study we do not
have gene-wise statistics, we are restricted to the former.
Those tests compare the overlap between a pathway and the gene set to 
the expected overlap from taking equally-sized random sets from the universe
of genes. This is often tested using a hypergeometric test. However, this 
approach requires deciding *a priori* what that gene universe is. Selecting
all the known genes is not an option, as a GWAIS experiment surveys all the 
genome unbiasedly, but not necessarily so all the genes. This is specially true
in an array focused on immunogenomics. Indeed, we observe this bias across all
mappings in the genes mapped to the SNPs with main effects 
(Figure \@ref(fig:platform-bias)).

(ref:fig-platform-bias-caption) Plaform bias in the genes hit by SNPs with significant main effects. Every data point is a gene. The *x*-axis represents the number of significant SNPs mapped to that gene. The *y*-axis represents the expected number, if a random set of SNPs of the same size were selected $\left( \frac {\text{(# significant SNPs)} \times \text{(# SNPs mapped to gene)}} {\text{# SNPS}} \right)$. In red, a line with a slope of 1 representing the case were SNPs are selected at random.

```{r platform-bias, echo=FALSE, fig.cap='(ref:fig-platform-bias-caption)'}
knitr::include_graphics('fig/main_effect_bias.png', dpi = NA)
```

Hence, I propose computing the gene 
background in a dataset specific way. For instance, the *Chromatin* 
results are analyzed in a gene universe where only the genes with a chromatin
mapping to the chip are used. However not all genes are surveyed at the same
resolution, as we observe in Figure \@ref(fig:sign-snps-per-gene-and-viceversa).
If two genes are equally involved in a disease, we are more likely to find an 
association in the gene which we are testing more often. Hence, I would like to
weigh every gene by the number of SNPs that map to it, which should provide 
a conservative null hypothesis. 

Lastly, the protocol presented here is a
complex, multi-stage approach which can be useful to any researcher with any
GWAIS dataset. In consequence, it would be useful to provide a dataset-agnostic
computational pipeline. The user would just need to provide a gene-gene network, 
a SNP-gene mapping, and a GWAIS dataset, and would be given two epistasic
networks, SNP- and gene-based respectively. Generating such a pipeline would 
also allow us to answer the questions outlined above faster.