# High-order epistasis detection through fusion of epistasis networks {#epi-network}

The contents of this chapter are based on *Héctor Climente-González, Diane Duroux, Chloé-Agathe Azencott, Kristel Van Steen*, High-order epistasis detection through fusion of epistasis networks (manuscript in preparation).

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
cat("\\begin{chapabstract}{Abstract:}
In this chapter we explore the detection of high-order epistasis in inflammatory bowel disease (IBD). We model high-order epistasis through a linear regression with interaction terms of orders up to 4. The number of tests increases exponentially with the order of the interactions, resulting in low statistical power and computational challenges. To tackle this, we devised a two-step approach to restrict the number of tests. In a first step, we apply a variety of second-order epistasis detection methods, and build an epistasis network that aggregates all the detected interactions. In a second step, we take all star subnetworks composed of 3 and 4 nodes, and search for high-order epistasis between them. The rationale is that SNPs involved in high-order epistasis are more likely to display second-order as well. The advantage of this approach is that we keep the number of tests under control, performing 124 tests instead of the $5 \\times 10^{17}$ required to exhaustively search order 4 interactions. Indeed, we are able to detect 37 high-order interactions (FDR 10\\%). Mapping these SNP-SNP interactions to gene-gene interactions suggests the involvement of the complement system in susceptibility to IBD.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}
Dans ce chapitre nous nous int\\'eressons \\`a la d\\'etection d'interactions 
\\'epistatiques d'ordre sup\\'erieur dans les maladies chroniques 
inflammatoires de l'intestin (MICI). Nous mod\\'elisions ces interactions 
par une r\\'egression lin\\'eaire comportant des termes d'interaction allant 
jusqu'\\`a l'ordre 4. Le nombre de tests statistiques augmentant de façon 
exponentielle avec l'ordre des interactions, cela r\\'eduit fortement la 
puissance statistiques tout en cr\\'eant des difficult\\'es calculatoires. 
Pour r\\'esoudre ces probl\\`emes, nous proposons une approche en deux \\'etapes 
pour r\\'eduire le nombre de tests. Dans la premi\\`ere \\'etape, nous appliquons 
diverses m\\'ethodes de d\\'etection d'\\'epistasie d'ordre 2, et construisons un 
r\\'eseau d'\\'epistasie qui agr\\`ege toutes les interactions ainsi d\\'ecouvertes. 
Dans un second temps, nous testons tous les sous-r\\'eseaux en \\'etoiles de 3 
ou 4 nœuds. Nous faisons ici l'hypoth\\`ese que les SNPs impliqu\\'es dans des 
interactions d'ordre 3 ou 4 sont plus susceptibles d'apparaître aussi 
dans des interactions d'ordre 2. L'avantage de notre approche est de 
contr\\^oler fortement le nombre de tests statistiques, ici 124 au lieu des 
$5 \\times 10^{17}$ requis pour une recherche exhaustive. Cela nous 
permet de d\\'etecter 37 interaction d'ordre 3 ou 4 (FDR 10\\%). Traduire 
ces interactions SNP-SNP en interactions g\\`ene-g\\`ene sugg\\`ere l'implication 
du syst\\`eme du compl\\'ement dans la susceptibilit\\'e aux MICI.
\\end{chapabstract}")
```

## Introduction

Although in the previous chapter we focused on interactions between two loci, some studies suggest that high-order epistasis, interactions between more than two loci, that is to say, strongly shapes complex traits [@taylor_higher-order_2015]. However, exhaustively searching for high-order epistasis aggravates the challenges of GWAIS (Section \@ref(gwais)): a large number of statistical tests to consider, higher diversity of ways an interaction can occur, and results harder to interpret. Under the rationale that high-order interactions are likely to show also second-order effects, second-order epistasis networks have been leveraged on in the past to detect high-order epistasis [@hu_statistical_2012]. In this chapter we explore high-order epistasis interactions in IBD using a similar approach. We do so in two steps. First, we apply a representative variety of second-order epistasis detection methods, and construct an epistasis network. Then we search for high-order epistasis between the SNPs in the network and their neighbors. As in Chapter \@ref(genesis), the first step also allows us to compare the different methods.

## Materials and methods

### Data, quality control and preprocessing

We studied genetic data from the IIBDGC (Section \@ref(iibdgc-dataset)), and performed a quality control and preprocessing as in Section \@ref(snp2gene-qc). Lastly, we performed LD pruning $R^{2}>0.75$ to remove SNPs in strong LD, taking down the final number of SNPs to 38\,225, which yields $7.3 \times 10^8$ pairs to test. In this chapter we did not apply any functional pre-filter to the data.

### Epistasis detection methods {#methods-methods}

We applied a selection of epistasis detection methods that can process continuous phenotypes and that had a ready-to-use implementation.

#### Linear regression

As in Section \@ref(methods:snp-epistasis), we explored two-way epistasis using a linear regression. In this case, we accounted for multiple testing by setting the family-wise error rate to 0.05.

#### MB-MDR

MB-MDR [@lishout_efficient_2013] is an extension of the classical MDR method [@moore_flexible_2006]. For a pair of SNPs, MDR classifies every pair of alleles as either high risk, or low risk, depending on whether for the examined combination, the cases/controls ratio is positive or negative, respectively. In a cross-validation setting, the SNP-pairs are ranked using the classification accuracy on unseen samples. MB-MDR includes two improvements over the MDR algorithm. First, it accounts for ambiguous pairs of alleles where the number of cases and controls are similar, boosting generalization. Second, MB-MDR extends MDR to continuous phenotypes. This is done by performing a *t*-test that compares the continuous trait between the samples with a specific a pair of alleles with that  of the samples with any other combination. If such difference is significant, the combination of alleles is classified as high risk or low risk, depending on the directionality of the change; else, it is considered ambiguous.

We ran MB-MDR with 999 permutations, and a codominant encoding. We kept interactions with a P-value \< 0.05 as in @lishout_efficient_2013.

#### EpiHSIC

EpiHSIC [@kam-thong_epistasis_2011] is a generalization to continuous phenotypes of EPIBLASTER [@kam-thong_epiblaster-fast_2011], which works on case-control studies. EPIBLASTER detects epistasis in a two stage approach. First all SNP pairs are screened using the difference in Pearson's correlation between two SNPs between cases and controls. Then, a logistic regression is used to detect epistasis among the candidate SNP pairs, similar to our what we do in Section \@ref(methods:snp-epistasis). @kam-thong_epistasis_2011 show that the difference in correlation coefficients is an instance of the Hilbert–Schmidt Independence Criterion (HSIC) which a particular choice of kernels. In EpiHSIC they propose a different selection of kernels that extends the screening step to continuous phenotypes. We used *gpuEpiScan*'s R implementation of this algorithm [@gpuEpiScan].

### High-order epistasis detection {#methods-high-order}

We built a SNP-SNP epistasis network using all the significant interactions from the presented methods. On this network, we selected star networks (one central node connected to all the others) as candidate sets for high-order epistasis detection. For each candidate set we built a linear model that included all possible interaction terms. For instance, for a network with three SNPs, *A*, *B* and *C*, we built the following model:

$$Y = \beta_0 + \beta_1 g_A + \beta_2 g_B + \beta_3 g_A g_B + \beta_4 g_A g_C + \beta_5 g_B g_C + \beta_6 g_A g_B g_C,$$

where the coefficients are the $\beta$ terms, and $g_A$, $g_B$ and $g_C$ are the genotypes under additive encoding for SNPs *A*, *B* and *C* respectively. We consider that there is a 3-way epistatic interaction between these three genes if $\beta_6$ is significantly different from 0. As under the null hypothesis the $\beta$ coefficients follow a standard Normal distribution, we evaluated a departure from it using a *t*-test. We restricted the sets to star subnetworks of size 3 and 4, as the number of tests grows exponentially with the number of SNPs. We accounted for multiple testing by setting the false discovery rate to 10%. For this purpose, the total number of tests was the number of $\beta$ for terms of order 3 and 4 for which a $\beta$ was obtained.

### Code availability

The different epistasis detection methods are implemented in pipelines with a similar interface, available on GitHub (https://github.com/hclimente/gwas-tools). The code necessary to reproduce this chapter's results, altogether with the corresponding analyses, are available on GitHub as well (https://github.com/hclimente/waffles).

## Results

### Epistasis detection methods produce relatively similar results

We applied different methods to search for epistatic interactions in the IIBDGC dataset (Section \@ref(methods-methods)). From these interactions, we built an epistasis network that included all significant pairs from any of the methods. In total they retrieved 823 epistatic interactions involving 459 SNPs (Figure \@ref(fig:overview)). MB-MDR is the method that detects the largest number of interactions (741), followed by logistic regression (68) and EpiHSIC (14). This is not surprising, since we did not apply any multiple testing correction to MB-MDR's output. Despite difference in size, MB-MDR's solutions greatly overlap with logistic regression's (37, 54.4%) and EpiHSIC (3, 21.4%) (Figure \@ref(fig:solutions-overlap)). In the same vein, 110 of the 459 SNPs in epistasis have significant main effects themselves (Fisher's Exact Test $\text{P-value} < 2.2 \times 10^{-16}$ on the overlap), despite both MB-MDR and logistic regression accounting for main effects. This large overlap points to a large agreement on the biological mechanisms of the disease. It also motivates searching for high-order epistasis preferentially among the subset of SNPs involved in second order interactions.

(ref:fig-overview-caption) Overview of the SNP epistasis networks produced by different epistasis detection (Section \@ref(methods-methods)). The network is split into several panels to depict the source of each interaction.

```{r overview, echo=FALSE, fig.cap='(ref:fig-overview-caption)'}
knitr::include_graphics('fig/overview_snp_solutions.png', dpi = NA)
```

(ref:fig-solutions-overlap-caption) Overlap between the solutions from the different epistasis detection methods. Note that the *y*-axis is in logarithmic scale.

```{r solutions-overlap, echo=FALSE, fig.cap='(ref:fig-solutions-overlap-caption)', out.width = '60%', fig.align = 'center'}
knitr::include_graphics('fig/overlap_solutions.png', dpi = NA)
```

(ref:fig-main-effect-overlap-caption) Main effects on IBD, measured using a logistic regression on the case-control phenotype. The 7 largest principal components are included in the model as covariates. SNPs involved in an epistasis interaction are colored in dark grey.

```{r main-effect-overlap, echo=FALSE, fig.cap='(ref:fig-main-effect-overlap-caption)'}
knitr::include_graphics('fig/manhattan_epistasis.png', dpi = NA)
```

### High-order epistasis interactions in IBD

We searched for 3rd and 4th order epistasis in 124 star subnetworks from the aggregated network (Section \@ref(methods-high-order)). Out of them 37 are significant at FDR 10% (Figure \@ref(fig:high-order)). The high proportion of the tested interactions that are significant (29.8%) supports the pertinence of our approach. Only three of the tested star subnetworks involved edges from different sources, probably due to the difference in solution size between MB-MDR and the other methods.

(ref:fig-high-order-caption) Overview of the detected high-order epistasis interactions, shown on the binary epistasis network. The edges involved in the 124 tested 3rd and 4th order interactions are highlighted in blue; the 37 among those that were significant, in red.

```{r high-order, echo=FALSE, fig.cap='(ref:fig-high-order-caption)', out.width = '60%', fig.align = 'center'}
knitr::include_graphics('fig/high_order_significant.png', dpi = NA)
```

### Mapping SNP to genes involves the complement system

We mapped the 823 SNP-SNP interactions to gene-gene interactions. We used any of the maps described in Section \@ref(methods:prefiltering), using exclusively genes with a HUGO symbol. As in Chapter \@ref(epi-snp2gene), we removed genes in the HLA regions and self-interactions. But, in contrast with that chapter, we make use of all three mappings at once, and hence expect gene-gene interactions where the genes were mapped via two different mappings. This resulted in 1\,113 different genes and 20\,826 different gene-gene interactions. When we restricted outselves only to those binary interactions involved in one of the 37 significant high-order interactions, we obtained 162 genes and 5\,273 candidate gene-pairs. This one-to-many mapping illustrates well the problem of linking SNP-SNP interactions to their functional consequences. Restricting the 5\,273 interactions only to those that exist on STRING [@szklarczyk_string_2019], we reduced the number to 892 interactions involving 94 genes. We decided to focus on this latter set of interactions. Although it is doubtful that all these gene-gene interactions are truly linked to the disease, we conjecture that it will be enriched in relationships truly associated to the disease. The high interconnectedness of the network (Figure \@ref(fig:overview)) gives support to this idea, seeming to capture a reduced number of susceptibility mechanisms.

We first studied how many SNP-SNP interactions supported every gene-gene interaction. The gene-gene interactions with higher support involve immunity-related genes. More specifically, they involve multiple genes from the complement system (*C4A*, *C4B* and, to a lesser extent, *C2*). This set of genes is part of the innate system, and is part of the organism's reaction to pathogens. The involvement of the complement system in IBD has received some attention lately [@jain_complement_2014;@heybeli_complement_2016;@sina_intestinal_2018], as mice lacking complement proteins suffer intestinal infammation similar to IBD. In our results, we observe an interaction between the three aforementioned genes of the complement, and in greater numbers with other genes, the one supported by most SNP-SNP interactions being *NOTCH4*.

## Discussion

In this chapter I apply three methods for second order epistasis detection to the IIBDGC dataset. Most of the detected interactions came from the same method, MB-MDR, which exhibited a high sensitivity. Consistently with the observation that SNPs with main effects are involved in such epistatic interactions, we detected 3rd and 4th order interactions between the many SNPs with two-way epistasis. When we mapped the SNP pairs to candidate gene pairs, we observed that many gene-gene interactions involved genes involved in the complement system. This gives a genetic rationale to the recent observations of the involvement of the complement system in the etiology of IBD.

However, our approach has some shortcomings. First, searching for high-order epistasis in hubs from the epistatis network is not possible under the current approach. For instance, the node with the highest degree has 39 neighbors, which implies building a model which includes interaction terms up to order 39. In this regard, a possible way forward would be dividing such star subnetworks into sets of 3 and 4 nodes, and looking exclusively for 3rd and 4th order interactions respectively within them. Another issue is the mapping of SNPs pairs to gene pairs. In this regard, our work in Chapter \@ref(epi-snp2gene) is promising, and answering the questions remaining in that project will pave the way to this one. Some of these questions are whether it is beneficial to use tissue-specific SNP-gene mappings, when should a gene-gene filter be used, or how to appropriately carry out a pathway enrichment analysis on epistasis results. However, this chapter presents new challenges in these fronts as well, like how to use binary interactions when we work on ternary interactions. Although here I decomposed them into binary interactions, it is worth considering higher order gene-interactions obtained from databases, namely co-complexes. Another open question is how the results of the presented approach compare to the results of existing methods to detect high-order interactions [@llinares-lopez_casmap_2018;@suzumura_selective_nodate;@terada_lamplink_2016].

Additionally, in this study we faced a scarcity of methods that work on continuous phenotypes or accept covariates. Such methods are necessary to account for population structure in epistasis detection. In this chapter I have presented three such methods, and we have located a few more: GenEpi [@chang_genepi:_2018], and several approaches based on random forest-based approaches [@yoshida_snpinterforest:_2011; @li_detecting_2016] or mutual information [@pensar_genome-wide_2019]. Adding some of these methods to protocol is likely to increase the number of detected high-order interactions. Additionally, I see this as a good opportunity to compare the methods, as has been done already on methods that work on case-control phenotypes [@Shang2011;@chatelain_performance_2018].