# High-order epistasis detection through fusion of epistasis networks {#epi-network}

The contents of this chapter are based on *Héctor Climente-González, Diane Duroux, Chloé-Agathe Azencott, Kristel Van Steen*, High-order epistasis detection through fusion of epistasis networks (manuscript in preparation).

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
cat("\\begin{chapabstract}{Abstract:}
TODO
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}

\\end{chapabstract}")
```

## Introduction

Although in the previous chapter we focused interactions between two loci, there is evidence that high-order epistasis (interactions between more
than two loci) strongly shape complex traits [@taylor_higher-order_2015]. However, the higher the order of the searched epistasis, the more pressing GWAIS challenges become (Section \@ref(gwais)). Under the rationale that high-order interactions are likely to show also second-order effect, second-order epistasis networks have been leveraged on in the past to detect high-order epistasis [@hu_statistical_2012]. In this chapter we explore high-order epistasis interactions in IBD using a similar approach. We do so in three steps. First, we apply a representative variety of second-order epistasis detection methods, and construct an equal number of epistasis networks. Second, we pool together the results into a consensus network. Last, we search for epistasis between the SNPs in the consensus network. As in Chapter \@ref(genesis), in the process we also compare the different methods.

## Materials and methods

### Data, quality control and preprocessing

We studied genetic data from the IIBDGC (Section \@ref(iibdgc-dataset)), and performed a quality control and preprocessing as in Section \@ref(snp2gene-qc). Lastly, we performed LD pruning $R^{2}>0.75$ to remove SNPs in strong LD, taking down the final number of SNPs to 38\,225.

### Epistasis detection methods {#methods-methods}

We applied a selection of epistasis detection methods that can deal with continuous phenotypes and that had a ready to use implementation.

#### Linear regression

As in Section \@ref(methods:snp-epistasis), we explored two-way epistasis using a linear regression. In this case, we accounted for multiple testing by setting the family-wise error rate to 0.05.

#### MB-MDR

MB-MDR [@lishout_efficient_2013] is an extension of the classical MDR method. For a pair of SNPs, MDR classifies every pair of alleles as either high risk, or low risk, depending on whether for the examined combination, the ratio cases/controls is positive or negative, respectively. In a cross-validation setting, the classification accuracy on unseen samples is used to rank the SNP pairs. MB-MDR includes two improvements over the MDR algorithm. First, it accounts for ambiguous pairs of alleles where the number of cases and controls are similar, boosting the generalization. Second, it extended it to continuous phenotypes. We ran MB-MDR with 999 permutations, and a codominant encoding. We kept interactions with a P-value \< 0.05.

#### EpiHSIC



TODO EpiHSIC [@kam-thong_epistasis_2011]

We used *gpuEpiScan*'s implementation in R of EpiHSIC [@gpuEpiScan].

### Consensus clustering

TODO Which fusion algorithm to use?

### High-order epistasis detection {#methods-high-order}

From the fusion network, we obtained candidate set of SNPs for high-order epistasis. For each of these sets, we built a linear model that included all possible interaction terms. For instance, for a network with three SNPs, *A*, *B* and *C*, we built the following model:

$$Y = \beta_0 + \beta_1 g_A + \beta_2 g_B + \beta_3 g_A g_B + \beta_4 g_A g_C  + \beta_5 g_B g_C + \beta_6 g_A g_B g_C,$$

were the coefficients are the $\beta$ terms, and $g_A$, $g_B$ and $g_C$ are the genotypes under additive encoding for SNPs *A*, *B* and *C* respectively.  Under the null hypothesis, the $\beta$ coefficients follow a standard Normal distribution. As the number of tests grows exponentially with the number of SNPs, we restricted the sets to star subnetworks of size 3 and 4. We accounted for multiple testing by setting the false discovery rate to 10%. For this purposes, the total number of tests was the number of $\beta$ for terms of order 3 and 4 for which we could have a value.

### Code availability

The different epistasis detection methods are implemented in pipelines, available on GitHub (https://github.com/hclimente/gwas-tools). The code necessary to reproduce this chapter's results, altogether with the corresponding analyses, are available in GitHub as well (https://github.com/hclimente/waffles).

## Results

### TODO Main effects

### Epistasis detection methods produce relatively similar results

We applied different methods to search for epistatic interactions in the IIBDGC dataset (Section \@ref(methods-methods)). From these interactions, we built an epistasis network per method (Figure \@ref(fig:overview)). MB-MDR is the method that detects the larger number of interactions (741), followed by logistic regression (68) and EpiHSIC (14). Despite the differences in size, MB-MDR's solutions greatly overlap with logistic regression's (37, 54.4%) and EpiHSIC (3, 21.4%) (Figure \@ref(fig:solutions-overlap)).

(ref:fig-overview-caption) Overview of the SNP epistasis networks produced by the epistasis detection methods described in Section \@ref(methods-methods). Any node participating in an interaction detected by any method is depicted in all panels.

```{r overview, echo=FALSE, fig.cap='(ref:fig-overview-caption)'}
knitr::include_graphics('fig/overview_snp_solutions.png', dpi = NA)
```

(ref:fig-solutions-overlap-caption) Overlap between the solutions from the different methods. Note that the *y*-axis is in logarithmic scale.

```{r solutions-overlap, echo=FALSE, fig.cap='(ref:fig-solutions-overlap-caption)'}
knitr::include_graphics('fig/overlap_solutions.png', dpi = NA)
```

### High-order epistasis interactions in IBD

We searched high-order epistasis in star subnetworks from the aggregated network (Section \@ref(methods-high-order)). In total, we tested 124 interactions 3rd and 4th order interactions, out of which 37 are significant (Figure \@ref(fig:high-order)). The high proportion of the tested interactions that are significant validates this approach. However, most of the edges come from MB-MDR: only three of the tested star subnetworks involve edges from different sources.

(ref:fig-high-order-caption) Overview of the detected high order epistasis interactions. The edges involved in the 124 tested 3rd and 4th order interactions are highlighted in blue; the 37 among those that were significant, in red.

```{r high-order, echo=FALSE, fig.cap='(ref:fig-high-order-caption)'}
knitr::include_graphics('fig/high_order_significant.png', dpi = NA)
```

### TODO Map to genes

## Discussion

In this chapter I apply multiple methods for second order epistasis detection to the IIBDGC dataset. MB-MDR detected more interactions than any other method, exhibiting a high sensitivity to detection.

By combining different methods, we detect epistasis that none of the methods could find by itself.

During this study, we faced a scarcity of methods which work on continuous phenotypes, or accept covariates. Although we found comparisons of methods which study case-control phenotypes [@Shang2011;@chatelain_performance_2018], to the best of our knowledge the equivalent does not exist for continuous phenotypes. As dealing with such outcomes is key to deal with population structure, I would like to extend this work to be such a comparison. In this chapter I have presented three such methods, and we have located three more: GenEpi [@chang_genepi:_2018], a random forest-based approach (e.g. [@yoshida_snpinterforest:_2011; @li_detecting_2016]), or mutual information [@pensar_genome-wide_2019].