# High-order epistasis detection through fusion of epistasis networks {#epi-network}

The contents of this chapter are based on *Héctor Climente-González, Diane Duroux, Chloé-Agathe Azencott, Kristel Van Steen*, High-order epistasis detection through fusion of epistasis networks (manuscript in preparation).

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
cat("\\begin{chapabstract}{Abstract:}
TODO
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}

\\end{chapabstract}")
```

## Introduction

Although in the previous chapter we focused interactions between two loci, there is evidence that high-order epistasis (interactions between more
than two loci) strongly shape complex traits [@taylor_higher-order_2015]. However, the higher the order of the searched epistasis, the more pressing GWAIS challenges become (Section \@ref(gwais)). Under the rationale that high-order interactions are likely to show also second-order effect, second-order epistasis networks have been leveraged on in the past to detect high-order epistasis [@hu_statistical_2012]. In this chapter we explore high-order epistasis interactions in IBD using a similar approach. We do so in three steps. First, we apply a representative variety of second-order epistasis detection methods, and construct an equal number of epistasis networks. Second, we pool together the results into a consensus network. Last, we efficiently search for epistasis between the SNPs in the consensus network. In the process, we evaluate the different methods for epistasis detection.

## Materials and methods

### Data, quality control and preprocessing

We studied genetic data from the IIBDGC (Section \@ref(iibdgc-dataset)), and performed a quality control and preprocessing as in Section \@ref(snp2gene-qc). Lastly, we performed LD pruning $R^{2}>0.75$ to remove SNPs in strong LD, taking down the final number of SNPs to 38\,225.

### Epistasis detection methods {#methods-methods}

We applied the epistasis detection methods that can deal with continuous phenotypes and that had a ready to use implementation.

#### Linear regression

Epistasis between two SNPs can be detected using a linear regression with an interacting term:

$$Y = \beta_0 + \beta_1 g_A + \beta_2 g_B + \beta_3 g_A g_B + \epsilon$$

After fitting the model, it can be evaluated if the interaction coefficient $\beta_3 \neq 0$. Under the null hypothesis, $\beta_3$ follows a standard Normal distribution. For this test, we used PLINK 1.9 [@chang_second-generation_2015]. We accounted for multiple testing by setting the family-wise error rate to 0.05.

#### MB-MDR

MB-MDR [@lishout_efficient_2013] is an extension of the classical MDR method. For a pair of SNPs, MDR classifies every pair of alleles as either high risk, or low risk, depending on whether for the examined combination, the ratio cases/controls is positive or negative, respectively. In a cross-validation setting, the classification accuracy on unseen samples is used to rank the SNP pairs. MB-MDR includes two improvements over the MDR algorithm. First, it accounts for ambiguous pairs of alleles where the number of cases and controls are similar, boosting the generalization. Second, it is extended to continuous phenotypes. We ran MB-MDR with 999 permutations, and codominant TODO. We kept interactions with a P-value \< 0.05.

#### EpiHSIC

TODO EpiHSIC [@kam-thong_epistasis_2011]

We `gpuEpiScan`'s implementation in R [@gpuEpiScan].

### Consensus clustering

TODO Which fusion algorithm to use?

### High-order epistasis detection

TODO Which measure to take?

### Code availability

The different epistasis detection methods are implemented in pipelines, available on GitHub (https://github.com/hclimente/gwas-tools). The code necessary to reproduce this article's results, altogether with the corresponding analyses, are available in GitHub as well (https://github.com/hclimente/waffles).

## Results

### Epistasis detection methods produce relatively similar results

(ref:fig-overview-caption) Overview of the SNP epistasis networks produced by the epistasis detection methods described in Section \@ref(methods-methods). Any node participating in an interaction detected by any method is depicted in all panels

```{r overview, echo=FALSE, fig.cap='(ref:fig-overview-caption)'}
knitr::include_graphics('fig/overview_snp_solutions.pdf', dpi = NA)
```

We applied different methods to search for epistatic interactions in the IIBDGC dataset (Section \@ref(methods-methods)). From these interactions, we built an epistasis network per method (Figure \@ref(fig:overview)). MB-MDR is the method that detects the larger number of interactions (741), followed by logistic regression (68) and EpiHSIC (14). Despite the differences in size, MB-MDR's solutions greatly overlap with logistic regression's (37, 54.4%) and EpiHSIC (3, 21.4%) (Figure \@ref(fig:solutions-overlap)).

(ref:fig-solutions-overlap-caption) Overlap between the solutions from the different methods. Note that the *y*-axis is in logarithmic scale.

```{r solutions-overlap, echo=FALSE, fig.cap='(ref:fig-solutions-overlap-caption)'}
knitr::include_graphics('fig/overlap_solutions.pdf', dpi = NA)
```

### Agreement between different methods


### High-order epistasis

## Discussion

By combining different methods, we detect epistasis that none of the methods could find by itself.

During this study, we faced a scarcity of methods which work on continuous phenotypes, or accept covariates. Although we found comparisons of methods which study case-control phenotypes [@Shang2011;@chatelain_performance_2018], to the best of our knowledge the equivalent does not exist for continuous phenotypes. As dealing with such outcomes is key to deal with population structure, I would like to extend this work to be such a comparison. In this chapter I have presented three such methods, and we have located three more: GenEpi [@chang_genepi:_2018], a random forest-based approach (e.g. [@yoshida_snpinterforest:_2011; @li_detecting_2016]), or mutual information [@pensar_genome-wide_2019].