# Conclusions

In the last 10 years great progress has been made in the understanding of the genetic architecture of complex diseases through GWA(I)S. That success would not have been possible without the effort of multiple international consortia which coordinated research groups to tackle specific diseases. Nonetheless, the GWA(I)S experimental settings present drawbacks that hinder the discovery of susceptibility mechanisms, our understanding of biology and, ultimately, improving diagnosis and clinical care. Some of these challenges are low statistical power, the difficulty to link associated genotypes to functional changes, multiple of ways to detect epistasis, and how to properly account for LD in the statistical treatment of the data. In this work I have explored the usefulness of network methods to tackle these challenges.

In **Chapter \@ref(genesis)** we applied and critically evaluated different high-score subnetwork search methods to the GENESIS breast cancer GWAS dataset. Such methods are particularly relevant in GWAS, as they address some of the drawbacks of the experimental setting: low statistical power and interpretability. Most network methods produced a biologically plausible answer, which by itself deepens our understanding of the susceptibility mechanisms acting out in this specific trait. However this methodological comparison context also highlights how radically different the solutions are from each other. This is the product of different ways of modeling the informal concept of high-score subnetwork mathematically. We combined of the different solutions into a consensus network, which was simpler than the largest individual solutions, but preserved their most important topological and biological properties.

In **Chapter \@ref(martini)** I presented *martini*, an R implementation of SConES that addresses some of its initial shortcomings, extending it to case-control phenotypes, and adding hyperparameter selection options and support for covariates. As I show Chapter \@ref(genesis), and expand in this chapter, SConES is a particularly flexible algorithm among high-score subnetwork search methods. Specifically, it has two hyperparameters that allow to fine-tune the topology and the sparsity of the selected subnetwork. Unfortunately, this flexibility comes at the price of appropriately tuning these parameters, an issue that is exacerbated by the instability of the solutions. In this regard, the feature selection scores I implemented helped finding more realistic solutions in some simulations, although one could still improve upon this proposition. One possibility would be to score the solution using the topology (edge density, centrality betweenness, number of connected components).

In **Chapter \@ref(epi-snp2gene)** we propose a protocol to improve both the statistical power and the interpretability of GWAIS, and apply it to the IIBDGC dataset. In essence, we employ three different ways of mapping SNPs to known gene interactions (physical, eQTL, and chromatin), and only test the corresponding pairs of SNPs. Indeed, we observe that the proposed protocols reduce the number of tests, and detect interpretable interactions which are not found using a conventional approach. For instance, the eQTL filter produces more SNP interactions than any other mapping, while the chromatin filter detected gene interactions are strongly related to interleukins, a immune system family of molecules involved in IBD. Despite the promising results, there are two aspects of the protocol that need to be refined: decoupling SNP-gene mapping from known gene interactions, to study the contribution of each to the positive results; and topping the protocol with an approach to map the detected interactions to the affected pathways.

In **Chapter \@ref(epi-network)** TODO

In this thesis I illustrated several hurdles in our way to identifying and understanding the genetics of complex diseases. I now conclude with a look into the state of the field, and future research directions related to my contribution.

**The future of network methods for GWAS.** The network methods I worked with during my PhD are notably heterogeneous. Although that heterogeneity stems from divergences in what different researchers aim to find, being able to obtain different points of view from a disease is a strength. However, clearer language and more exhaustive comparisons to other methods would be well-received in the methods' publications. With the exception of SConES, all high-score subnetwork search methods I tested deal with gene-gene interaction networks. SConES on the other hand deals directly with SNP networks. The latter kind of networks are potentially very interesting, as they operate at a lower level than the gene and, hence, potentially can handle more information. For instance, it could contain information about the specific protein residues that participate in a protein-protein interface. Or encode LD blocks, via altering the weights of the edges of the network in a proportional way to the correlation between SNPs. This is highlighted by the positive results of SConES even when no protein-protein information was added to the SNP network. On the other hand, the SNP networks that I handled in this thesis were orders of magnitude more complex than the corresponding protein-protein interaction networks for the same dataset. Probably, new methods (or faster versions of the current ones) are required to explore the potential of such, more informative, SNP networks.

**The role of network methods in bioinformatics.** My experience with network methods sometimes clashed with other people's impressions of the field during informal exchanges. In particular, some researchers receive network results and methods with a mix of puzzlement in the face of the overwhelming amount of information they represent, and a subsequent skepticism. In my opinion, this anecdotal evidence speaks to the state of network biology as a field, which lacks clear, agreed on, protocols and goals. Hence, I believe a multi-front effort must be made to close this gap. A first line of attack involves more accessible tools, with similar interfaces and proper documentation. One of my goals in Chapter \@ref(genesis) was to create such  an interface for the tools I used. Another subject to tackle is the development of better tools to visualize and manipulate network results. Innovative visualizations like hive plots [@krzywinski_hive_2012], and the package *tidygraph* to manipulate networks [@pedersen_tidygraph_2019] are moves in the right direction. Interaction databases are also part of the inaccessibility issue: @huang_systematic_2018 evaluated 21 such databases, often collecting overlapping information, and with unclear definitions of what *interaction* means. Efforts must be unified towards a single database, with a user-friendly interface and clarity about its contents -- with the caveat that such efforts may simply result in yet another database that will be used in parallel with the others^[https://xkcd.com/927/]. In this regard, I believe HINT [@das_hint_2012], which aggregates multiple databases, is a step in the right direction. This relates to the issue, also discussed in Chapter \@ref(genesis), of the different types of biological interactions: despite the preponderance of protein-protein interactions, other types of  interactions (like RNA -- RNA or transcription factor -- DNA) need to be better compiled and characterized. Lastly, as much as possible, we should be able to establish meaningful goals in the field, which will allow for descriptors of the networks  and protocols on how to achieve them.

**The future of GWAS.** Despite its issues, GWAS has been a sound success in identifying genetic associations with complex traits, therefore helping understanding their genetic basis. Many past and present initiatives are leveraging on more data to solve some of these issues: massive prospective cohort studies like UK Biobank [@sudlow_uk_2015], more GWAS on understudied populations, and initiatives in functional genomics like ENCODE [@the_encode_project_consortium_integrated_2012]. However, work also remains to be done on statistical side of things. Despite some efforts in the right direction [@wang_statistical_2018], the field is dominated by laboratory- and method-specific practices, which are rarely justified in published articles. For instance, whether to LD prune or not; map SNPs to the exact coordinates of a gene, or include some kilobases upstream and downstream; or what is the minimum MAF of SNPs to analyze. An open conversation is needed which confronts different protocols, and reaches consensuses where possible, and justified guidelines when it is not. On this front, network methods can be powerful allies, as shown in Chapter \@ref(genesis), but work in other fields, both experimental and *in silico*, is also required. For instance a proper incorporation of LD patterns to processing analysis might lead to both an increase in statistical power (by reducing the number of tests to one per independent test) and interpretability (by dealing with the true unit of variation), while accounting for population structure by default (as it is caused by LD). Also, the technology of choice for GWAS is slowly shifting from SNP arrays to the increasingly affordable whole genome-sequencing. This in itself is bringing a substantial change to the GWAS scene, by providing a deeper coverage which includes more rare variants. Nonetheless, this increase in the number of variants studied only increases our need for appropriately treating LD.

**The future of GWAIS.** Epistasis detection is an open and promising field. As in high-score subnetwork search, and as I describe in Chapter \@ref(epi-network), epistasis detection can be achieved through a multiplicity of tools that model it in different ways. As with the aforementioned network methods, we can exploit that multiplicity by collapsing the results into a unified view of disease. Yet, a few challenges remain ahead. The first one, developed above, is the lack of general guidelines for GWA(I)S. The second one, is the inability of most methods to account for population structure, either by accepting covariates, or by accepting an adjusted, continuous phenotype. This last problem also comes up in the study of continuous phenotypes, which were beyond the scope of this thesis. In Chapter \@ref(epi-snp2gene) I explored the possible contribution of functional pre-filtering to epistasis detection. In this regard, and encouraged by my experiences described in Chapters \@ref(genesis) and \@ref(martini), I believe in the potential of epistasis-detection methods that exploit prior knowledge in the form of a network.

**Open science with sensitive data.** I would like to finish reflecting about my experience on *open science* as a GWAS researcher. Understandably, GWAS data from human samples requires a careful treatment, as it contains very sensitive information about both the participants and their families. For that reason, dealing with GWAS data requires compromising on open data, one of the pillars of open science. Hence, at the moment work on genetic data requires trust from all the other researchers who do not have access to a particular dataset. This might change in the future, when data obfuscation solutions that allow to operate on a dataset while hiding the identifiable information [@azencott_machine_2018] are fully developed. Yet, for the moment, the lack of open data should lead to a stronger commitment to open algorithms and results, to facilitate as much as possible that other researchers are able to to apply a publication's tools to their own datasets and study if the conclusions hold.