# Conclusions

In the last 10 years great progress has been made in the understanding of the genetic architecture of complex diseases. That success would not have been possible without the effort of multiple international consortia which coordinated research groups to tackle specific diseases. Nonetheless, the GWAS and GWAIS experimental settings still present drawbacks, which restrain the applicability of their results. In this work I explore the usefulness of network methods to tackle these challenges.

In **Chapter \@ref(martini)** I present *martini*, an R implementation of SConES that addresses some of its initial shortcomings: extending it to case-control phenotypes, adding hyperparameter selection options, and improving user-friendliness. As I show in this chapter, and further elaborate in Chapter \@ref(genesis), SConES is a particularly flexible algorithm among high-score subnetwork search methods. Specifically, it has two hyperparameters that allow to fine-tune the topology and the sparsity of the selected subnetwork. Nonetheless, this flexibility comes at the price of appropriately tuning these parameters, an issue that is exacerbated by the algorithm instability. In this regard, the implemented feature selection scores helped finding more realistic solutions in some simulations, although a final solution to the problem remains to be found. Possibly it will involve scoring the solution using the topology (edge density, centrality betweenness, number of connected components).

In **Chapter \@ref(genesis)** we apply and critically evaluate different high-score subnetwork search methods to the GENESIS breast cancer GWAS dataset. Such methods are particularly relevant in GWAS, as they address some of the drawbacks of the experimental setting: low statistical power and interpretability. All network methods produced a biologically-plausible answer, which by itself deepens our understanding of the susceptibility mechanisms acting out in this specific dataset. However this methodological comparison context also highlights how radically different the solutions are from each other. This is the product of different mathematical models of the optimal susceptibility mechanism. We explored a combination of the different solutions into a consensus network, which was more manageable than the largest of the individual solutions, and preserved the most important topological and biological properties.

In **Chapter \@ref(epi-snp2gene)** TODO

In **Chapter \@ref(epi-network)** TODO

After completing this work, and having extensively dealt with both GWAS and network methods, it is clear that some challenges remain ahead.

**The future of network methods.** The network methods I worked with during my PhD are notably heterogeneous. Although that heterogeneity stems from divergences in what different researchers aim to find, being able to obtain different points of view from a disease is a strength. However, clearer language and more exhaustive comparisons to other methods would be well-received in the methods' publications. With the exception of SConES, all high-score subnetwork search methods tested deal with protein-protein interaction networks. SConES on the other hand deals directly with SNP networks. The latter kind of networks are potentially very interesting, as they operate at a lower level than the gene and, hence, potentially can handle more information. For instance, it could contain information about the specific protein residues that participate in a protein-protein interface. Or encode LD blocks, via altering the weights of the edges of the network in a proportional way to the correlation between SNPs. This is highlighted by the positive results of SConES even when no protein-protein information was added to the SNP network. On the other hand, the SNP networks that I handled in this thesis were orders of magnitude more complex than the corresponding protein-protein interaction networks for the same dataset. Probably, new methods (or faster versions of the current ones) are required to explore the potential of such, more informative, SNP networks.

**The role of network methods in bioinformatics.** My experience with network methods sometimes clashed with other people's impressions of the field during informal exchanges. In particular, some researchers receive network results and methods with a mix puzzlement in face of the overwhelming amount of information they represent, and a subsequent skepticism. In my opinion, this anectodal evidence speak to the state of the field of study of biological networks, which lacks clear, agreed on, protocols and goals. Hence, I believe a multiple-front effort must be made to close this gap. One of them are more accessible tools, with similar interfaces and proper documentation. One of the goals of Chapter \@ref(genesis) was to create such interface at least for the tools I dealt with. Also, better tools to visualizing and manipulating network results. Moves in the right direction are innovative visualizations like hive plots [@krzywinski_hive_2012], and the package *tidygraph* to manipulate networks [@pedersen_tidygraph_2019]. Interaction databases are also part of the inaccessibility: [@huang_systematic_2018] evaluated 21, often collecting overlapping information, and with unclear definitions of what *interaction* means. Efforts must be unified towards a single database, with a user-friendly interface and clarity about its contents^[https://xkcd.com/927/]. In this regard, I believe HINT[@das_hint_2012] is a step in the right direction. This relates to the issue, also discussed in Chapter \@ref(genesis), of the different types of biological interactions: despite the preponderance of protein-protein interactions, other types of biological interactions need to be better compiled and characterized. Lastly, as long as possible, we should be able to establish meaningful goals in the field, which will allow for descriptors of the networks  and protocols on how to achieve them.

**The future of GWAS.** GWAS has been a sound success in identifying genetic associations with complex traits, and in understandind their genetic basis. However, work remains to be done in the field of functional genomics, that is, finding the cellular function associated to a specific genotype. In this front, network methods can be powerful allies, as shown in Chapter \@ref(genesis), but work in other fields, both experimental and *in silico*, is also required. For instance a proper incorporation of LD patterns to processing analysis, might lead to both an increase in statistical power (by reducind the number of tests to one per independent test), the interpretability (by dealing with the true unit of variation), while accounting for population structure by default (as it is caused by LD). Also, the technology of choice for GWAS is slowly shifting from SNP arrays to the increasingly affordable whole genome-sequencing. This in itself will bring a substantial change to the GWAS scene, by providing a deeper coverage which will include rare variants. Nonetheless, this increase in the number of variants studied makes even more necessary an appropriate treatment of LD.

**The future of GWAIS.** Epistasis detection is an open and promising field. As in high-score subnetwork search, and as I describe in Chapter \@ref(epi-network), epistasis detection enjoys a multiplicity of tools that capture different aspects of the problem. And as with those network methods, we can exploit that multiplicity by collapsing the results into a unified view of disease. Yet, bit challenges remain ahead. The first one is the lack of a standardize protocol for epistasis detection. The second one, is the inability of most methods to account for population structure, either by accepting covariates, or by accepting an adjusted, continuous phenotype. Regarding the latter, an identital problem comes up in the study of continuous phenotypes, which were beyond the scope of this thesis. In this thesis we explored the possible contribution of functional pre-filtering to epistasis detection. In this regard, another compelling field of study are epistasis-detection methods that exploit prior knowledge in the form of a network.

**Open science with sensitive data.** I would like to finish reflecting about my experience on *open science* as a GWAS researcher. Understandably, GWAS data from human samples requires a careful treatment, as it contains very sensitive information about the participants and their families. For that reason, dealing with GWAS data requires compromising on open data, one of the pillars of open science. TODO IDEAS?