# Susceptibility genes to breast cancer {#hboc-genes}

Nearly all known HBOC susceptibility genes encode tumor suppressors that participate in genome stability pathways (homologous recombination repair, replication fork stability, transcription-replication collisions, mismatch repair, and DNA damage signaling, checkpoints and cell death). 

## Homologous recombination repair

The homologous recombination repair pathway (HRR) deals with double strand DNA breaks by using the undamaged chromosome as template for error-free repair. After a DSB occurs, the MRN complex (MRE11, RAD50 and NDN) detects and binds the free DNA ends. Then, it promotes DNA damage checkpoint signaling.

HRR involves *BRCA1*, *BRCA2* and, actually, most of the HBOC genes. Because of its ability to interact with a wide range of proteins, BRCA1 is hypothetized to act as a recruitment scaffold. A deficiency of *BRCA1* is linked to the inability to trigger HRR. Mutations in the MRN complex have also been clinically associated to breast cancer, although dubiously so in the case of *RAD50* variants. Reassuringly, some other HBOC genes are interactors of the MRN complex and BRCA1/2.

## Replication fork stability

BRCA1 and BRCA2 protect newly synthesized DNA and promote the restart of stalled forks in an HRR-independent manner. In the absence of these proteins, newly synthesized DNA in a stalled fork would get degraded, leading to genome instability and increasing the risk of cancer.

## Transcription-replication collisions 

Collisions between transcription and replication are emerging as a source of genome instability. In particular, RNA-DNA hybrids called R-loops can form between the nascent transcript and the DNA template. They can lead to double-strand breaks and mutations. Both BRCA1 and BRCA2 participate in the resolution of R-loops, preventing their accumulation. In consequence, BRCA-deficient cells tend to suffer transcriptional stress that leads to genome instability. Nonetheless the relationship between this mechanism and proneness to HBOC is yet to be proven, and the genes involved further investigated.

## Mismatch repair

DNA mismatch repair (MMR) corrects base-base mispairs. When MMR is faulty, accumulations point mutations and genetic changes in repeated nucleotide sequences (microsatellite instability) occur. MMR also plays a role in error-free HRR.

## DNA damage signaling, checkpoints and cell death

Pathways involved in genome maintenance, cell cycle checkpoints and cell death usually eliminate cells with damaged DNA. When proteins involved in them are not active, some processes such as cell cycle arrest, apoptosis and senescence will not occur. In consequence, cells that undergo genomic alterations are allowed to proliferate. The most famous case of HBOC in this pathway is TP53, which coordinates the transcriptional induction of many genome stability factors.

(ref:tab-hboc-genes-caption) Overview of HBOC genes: estimated lifetime risk of breast cancer (age in years) and tumorogenic molecular mechanisms that involves them: homologous recombination repair (HRR), replication fork stability, transcription-replication collisions, mismatch repair (MMR), DNA damage signaling, checkpoints and cell death, and/or others. Adapted from @nielsen_hereditary_2016.

(ref:checkmark) $\checkmark$

```{r hboc-genes, echo=FALSE}
library(magrittr)
library(kableExtra)
library(readr)

hboc_genes <- read_table2(file = "
ATM 60%_by_age_80 (ref:checkmark) - - - (ref:checkmark) -
BARD1 Unknown (ref:checkmark) - - - - -
BLM Unknown - (ref:checkmark) - - - -
BRCA1 57-65%_by_age_70 (ref:checkmark) (ref:checkmark) (ref:checkmark) - (ref:checkmark) -
BRCA2 45-55%_by_age_70 (ref:checkmark) (ref:checkmark) (ref:checkmark) - (ref:checkmark) -
BRIP1 OR:_<_2.0 - - - - - -
CDH1 42%_by_age_80 - - - - - (ref:checkmark)
CHEK2 37%_by_age_70 - - - - (ref:checkmark) -
FAM175A Unknown (ref:checkmark) - - - - -
FANCC Unknown - (ref:checkmark) - - - -
FANCM Unknown - (ref:checkmark) - - - -
MLH1 ~19%_by_age_70 - (ref:checkmark) - (ref:checkmark) - -
MRE11 Unknown - - - (ref:checkmark) - -
MSH2 ~11%_by_age_70 - - - (ref:checkmark) - -
NBN OR:_3.0 (ref:checkmark) - - - - -
NF1 6.5-fold_up_ages_30-39 - - - - - (ref:checkmark)
PALB2 35%_by_age_70 (ref:checkmark) (ref:checkmark) - - - -
PMS2 SIR:_3.8 - - - (ref:checkmark) - -
PTEN 85%_by_age_70 - - - - - (ref:checkmark)
RAD51B Unknown (ref:checkmark) - - - - -
RAD51C Unknown (ref:checkmark) - - - - -
RAD51D Unknown (ref:checkmark) - - - - -
RECQL Unknown - (ref:checkmark) - - - -
RINT1 Unknown (ref:checkmark) - - - - -
STK11 32%_by_age_60 - - - - - (ref:checkmark)
TP53 25%_by_age_70 - - - - (ref:checkmark) -
", col_names = FALSE, col_types = 'cccccccc')

colnames(hboc_genes) <- c('Gene', 'Lifetime risk', 'HRR', 'Rep. fork stab.', 'Tr.-rep. clash', 'MMR', 'DNA Damage, apoptosis', 'Other')

hboc_genes <- data.frame(lapply(hboc_genes, function(x) {
  gsub("^-$", "", x)
  }))

hboc_genes <- data.frame(lapply(hboc_genes, function(x) {
  gsub("_", " ", x)
  }))

kable(hboc_genes, 
      caption = '(ref:tab-hboc-genes-caption)', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = "striped") %>%
  landscape()
```
