--- 
title: "Network-guided genome-wide association studies"
author: "Héctor Climente González"
date: "4 Février 2020"
site: bookdown::bookdown_site
documentclass: env/yjiao
output: 
  bookdown::pdf_book: 
    dev: png
    toc : true
    toc_depth: 4
    keep_tex : yes
bibliography: [references.bib]
link-citations: yes
description: "PhD thesis of Héctor Climente-González"
lof: yes
lot: yes        
fontsize: 11pt
biblio-style: "env/yjiao"
geometry: "a4paper,left=1.5in,right=1.3in,top=1.1in,bottom=1.1in,includefoot,includehead,headheight=13.6pt"
linestretch: 1.05
---

# Acknowledgements {-}

This thesis articulates the last 3 years of my research into a moderately thick book, and I do take some pride in having written it. However, this work would not exist in the first place without all the people who believed in me, guided me when I was lost, and plucked a smile out of me when I most needed it. For that reason, I understand now the convention of preceding theses with a shoutout to everyone who actually made them possible. The following lines are a poor attempt at that.

First I want to acknowledge the role of my supervisor, **Chloé**. Thank you for the freedom to pursue my own interests, and thank you for the guidance to get there. For finding time for me when you did not have any, and for the cheerful random chats. For teaching me how to be a meticulous researcher, to think better, and to communicate my science to all sorts of people. Lastly, I will forever thank you for the trust you put in me and for giving me the opportunity to learn from you and from the rest of CBIO.

During my doctorate I was fortunate to collaborate with three excellent laboratories. I not only learned a lot from them, but also met wonderful colleagues and mentors. I want to thank **Nadine Andrieu**, **Makoto Yamada** and **Kristel Van Steen** for such opportunities. Thank you as well to the members of their teams, who welcomed me with open arms and made me feel at home when I was so far away. Most especially, thank you to **Christine Lonjou**, **Diane Duroux**, and **Fabienne Lesueur**, with whom it has really been a pleasure to work.

I would like to thank Institute Curie's Training Unit for organizing the IC3i program, for always supporting me and helping me out when needed. Thank you to as well to all the recipients of the IC3i scholarship for creating such a friendly environment. Thank you, people from ADIC, for organizing Happy Fridays, and most specially thank you to all the amazing people that joined me there after a long day of work. Thank you to Institut Curie in general, and to U900 in particular for the amazing scientific support. Most specially, thank you, **Christine Lonjou**, for all the times you sat with me to discuss my results, for our little conversations in Japanese, and for all your advice. I really appreciate the pinch of joy that you bring to our floor. And thank you, **Om Kulkarni**, for, as you would put it, *being an awesome human being*. Thank you for your enthusiasm, your attention to detail, your humanity, and for the wave of positivity you bring along.

Thank you to the current and former members of CBIO. All of you created a fun and constructive environment, and gave me many moments to treasure. My warmest thanks to **Joe**, for always having something to share: a joke, a gaze, an ear, a helping hand, or your latest idea; **Peter**, for all the hours sharing animes, technology and games, and for all those moments of helping each other out; **Beubeu-chan**, for always talking with passion about the big picture, the small picture, and the pictures we didn't even know were there when we started; **Judith**, for being such a positive force of change; **Maguette**, for enduring my awful jokes; **Lotfi**, for our shared fascinations and frustrations (mostly for the latter though); **Asma**, for your sense of humor and our little jokes; and **Thomas**, for dissecting my problems with surgical precision.

I would not be half the scientist and the person I am today if not for the exceptional concurrence of twisted minds that I met during my time at GRIB. It was an absolute privilege to share so many nerdy days and great nights with you all. Thank you to **Eduardo**, **Babita**, **JC**, **Caterina**, **Emma**, **Will**, **Pau**, **Jorge**, **Juanlu**, **José Luis**, and **Isaac**.

Mi Lobe **Yunlong**, coming to Paris was worth it since you blinded me with your inner light.^[You are a funny guy.] Which is ironic, since you worked restlessly to make me hate this city so much.^[:)] Thanks for showing me the ways of the eigens, as well as the ways of the alitas. For knowing that I can always count on you, and for our trips together. Most specially, thank you for that heartwarming feeling of deep syntony when we are together: the laughs, the long conversations, the gossiping and the shared fascinations.

Oiee **Shona**, querida, why are you mi parti only? Thank you for all the sacrifice and the compromise, and thank you as well for all the hopes and shared dreams. Thank you for all our little things, the words, the dances, the dishes, the glances, the cards, the songs, and the flags with which we spice up our lives. Thanks for completing me, for taking care of me and for letting me take care of you. Thanks for our deep talks, and for sharing your brain with me. Thanks for listening when I do not see a way forward, and for taking me by the hand to look for that way together.

**Peke**, ni este párrafo, ni un libro serían suficientes para agradecerte el haber entrado en mi vida. Mereces un premito por todas las buenas decisiones que he tomado, y otro por todas las malas, por saber que mientras puediésemos reirnos juntos de ellas, serían soportables. Gracias por permitirme ser cómo soy desde el momento en que nos conocimos, amando mis virtudes y perdonando cuando te he herido. Y gracias por ser el pilar de mi vida, pues todo es impermanente menos nosotros.

Lastly, I would like to thank my family, to whom I ultimately owe everything I am today. Gracias a mis **tíos**, Esther y Manolo, y a mi primo **Alexis**, por hacer de cualquier celebración un placer. Gracias **Pili** por la alegría que irradias. Gracias **Paquita** por haberme cuidado, educado (y soportado) como a un nieto más. Gracias a mi **Yaya** Gloria, que con su cariño, su humor, y sus patatas rebozadas, me sostuvo durante años en Barcelona. Gracias a mi **Abuelo** Manolo, quien me enseñó la paciencia, la dedicación y la bondad. Y gracias a mi **Abuela** Esther, de quién aprendí a querer incondicionalmente. Gracias **Alicia** por haber sido tanto mi hermana pequeña como un ejemplo a seguir. Gracias **Mamá** por tu amor; por todas nuestras conversaciones; por haber recorrido lo que hiciese falta para estar a mi lado; por tus consejos y tu soporte; por enseñarme con rigidez y con flexbilidad a partes iguales; por inculcarme el respeto y la consideración. Por esto y por mil cosas más, esta tesis es posible, sobre todo, gracias a ti.

<!-- 
Cover dimensions wrong because
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
-->

```{r env, echo=FALSE}
# options(knitr.graphics.auto_pdf = TRUE)
library(magrittr)
library(kableExtra)
```

<!--chapter:end:index.Rmd-->

\mainmatter

# Context

```{r echo=FALSE, results="asis", eval='is_latex_output'}
cat("\\begin{chapabstract}{Abstract:}
This thesis treats the identification of the genetic factors underlying complex diseases. Usually this is done via genome-wide association studies (GWAS), when univariate association is studied, and via genome-wide association interaction studies (GWAIS), when interactions between genetic factors are considered. Both experimental settings involve the genotyping of hundreds of thousands of single-nucleotide polymorphism (SNPs) over thousands of individuals, both affected by the disease under study and unaffected. Despite their success in identifying many variants behind common diseases, GWA(I)S still present challenges, namely low statistical power, difficult interpretation, and arbitrary choices at multiple points of the study. Approaching the problem from a network perspective might help tackling these issues. In essence, this involves considering both the individual association with the disease of each gene and their biological context, modeled as a network built from known gene-gene interactions. In this thesis I focus on how network-based methods can boost biomarker discovery and interpretability on GWA(I)S. I apply these methods to two complex diseases: familial breast cancer and inflammatory bowel disease.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}
Cette th\\`ese porte sur l'identification des facteurs g\\'en\\'etiques qui 
sous-tendent des maladies complexes. Les \\'etudes d'association g\\'enome 
entier (GWAS), sont g\\'en\\'eralement utilis\\'ees pour \\'etudier des associations 
univari\\'ees, tandis que les \\'etudes d'association d'interactions g\\'enome 
entier (GWAIS) prennent en consid\\'eration des interactions entre facteurs 
g\\'en\\'etiques (ou \\'epistasie). Ces deux approches n\\'ecessitent de g\\'enotyper 
des centaines de milliers de polymorphismes d'un seul nucl\\'eotide (SNPs) 
pour des milliers d'individus, affect\\'es ou non par la maladie \\'etudi\\'ee. 
Bien qu'elles aient permis d'identifier de nombreux variants associ\\'es \\`a 
des maladies courantes, les GWA(I)S pr\\'esentent toujours de nombreux 
d\\'efis, parmi lesquels leur faible puissance statistique, la difficult\\'e 
de leur interpr\\'etation, ainsi que les choix arbitraires qui doivent \\^etre 
faits \\`a diff\\'erentes \\'etapes de ces \\'etudes. Consid\\'erer ces \\'etudes sous 
l'angle des r\\'eseaux biologiques peut permettre de r\\'epondre \\`a ces d\\'efis. 
Il s'agit de consid\\'erer \\`a la fois l'association individuelle entre 
chaque g\\`ene et la maladie, et le contexte biologique de ces g\\`enes, 
mod\\'elis\\'e par un r\\'eseau contruit \\`a partir d'interactions g\\`ene-g\\`ene 
connues. Dans cette th\\`ese, j'\\'etudie comment l'utilisation de r\\'eseaux 
biologiques facilite la d\\'ecouverte de nouveaux biomarqueurs ainsi que 
l'interpr\\'etation des GWAS(I)S. J'applique ces m\\'ethodes \\`a deux maladies 
complexes : le cancer du sein familial, et les maladies chroniques 
inflammatoires de l'intestin.
\\end{chapabstract}")
```

## The common disease/common variant framework

Complex diseases are those caused by a mixture of genetic, environmental and lifestyle factors. The object of study of this thesis are the methodologies to identify such genetic factors. This is of paramount importance for disease prevention, understanding the etiology of diseases, and providing better treatments. 

The genetic architecture of a trait includes the variants that contribute to the risk, as well as their allelic frequencies, effect sizes, and their genetic mode of action (e.g. dominant or recessive). From this point of view, complex diseases are easier to understand in contrast with Mendelian ones. The latter are caused by a single locus with a strong effect, and hence follow the Mendelian rules of inheritance. In essence, and barring considerations on reduced penetrance, whether an individual will develop a Mendelian disease or not depends exclusively on the two alleles at that particular locus, and their genetic mode of action. By contrast, the genetic architecture of complex is modeled by the *liability-threshold model*, an extension to binary traits of the infinitesimal model used to describe the genetics of continuous phenotypes like height. Under the infinitesimal model, a continuous trait is shaped by many Mendelian alleles, each of them with a small contribution to the trait [@barton_infinitesimal_2017]. Specifically, it computes a score for an individual based on the contribution of each of the alleles in the genetic architecture, plus the contribution of the environment. The liability-threshold model computes an equivalent latent score and, if it takes a value above a given threshold, disease will ensue.

(ref:fig-size-freq-caption) Relationship between the minor allele frequency of an allele, and its effect size. As a consequence of evolutionary forces, highly deleterious alleles which cause Mendelian diseases are infrequent, while alleles with small effects that increase slightly the susceptibility to a complex disease are more widespread. Figure adapted from @bush_chapter_2012, licensed under CC BY.

```{r size-freq, echo=FALSE, fig.cap='(ref:fig-size-freq-caption)'}
knitr::include_graphics('fig/dl_frequency_vs_effect_size.png', dpi = NA)
```

Because the risk alleles have such small effect sizes, they are not under strong purifying selection (Figure \@ref(fig:size-freq)). This allows them to be common (>1-5% of the population), unlike the mutations causing Mendelian diseases, which are rare, as they strongly decrease the fitness of the individual [@manolio_finding_2009]. In consequence, the study of the genetics of complex diseases relies on the *common disease, common variant* hypothesis: common diseases are partly attributable to allelic variants present in more than 1-5% of the population, which cause, by themselves or in combinations, small increments in risk (1.1 - 1.5-fold). However, another consequence of this limited effect size is that only weak associations between causal variants and phenotypes can be expected. Again, this notion is radically different from Mendelian diseases, where every carrier of the risk allele develops the disease under complete penetrance. Therefore, the study of the genetics of complex diseases requires the identification of many risk variants among the 88 millions known ones [@the_1000_genomes_project_consortium_global_2015]. 

The most common source of genetic variation in humans are single base-pair changes in the DNA sequence, called single-nucleotide polymorphisms or SNPs [@the_1000_genomes_project_consortium_global_2015]. They usually involve two alleles, meaning that in a population there are two possible base-pairs for a genetic position. SNPs are characterized by their minor-allele frequency, that is, the frequency of the least common allele in the population. Although in this thesis I focused my work on SNPs, other forms of genetic variation exist. These are the structural variants, which involve variation in the structure and the quantitative arrangement of the chromosomes [@spielmann_structural_2018]. Copy number variants (CNVs), a type of structural variants, consist on a repeated segment of the genome, where the specific number of repeats changes from person to person. CNVs are relevant for human health. For instance a neurological disorder known as Huntington's disease ensues when a specific tri-nucleotide in the huntingtin gene is repeated more than 36 times [@macdonald_novel_1993].

## Genome-wide association studies {#intro-gwas}

Genome-wide association studies (GWAS) are experiments that explore large cohorts, systematically surveying both a high number of genetic variants and the individuals' phenotype(s) [@bush_chapter_2012]. Their goal is to find associations between genotypes and the studied phenotype. These associations might lead to earlier diagnoses, choices of treatment appropriate for a patient's genetic background, and better understanding of the etiology of the disease. For that purpose, a classical GWAS involves a statistical test of association between each variant and the phenotype. Often that statistical test is a logistic regression, which allows using additional variables that might act as confounders. For instance, for a SNP $i$:

\begin{equation}
logit \left(p_i\right) = \alpha + \beta_i g_i + \gamma X
(\#eq:gwas-logreg)
\end{equation} 

where $g_i$ is the vector of genotypes at SNP $i$, $\beta_i$ is the coefficient, and $X$ and $\gamma$ are respectively the matrix of covariates and the vector with the covariates' coefficients. A statistical test can be conducted on the value of $\beta$ by transforming it into a Z-score. In order to evaluate the significance of the associated P-value, an appropriate threshold which accounts for multiple testing is needed. Often, that P-value is chosen by setting the family-wise error rate to 0.05 i.e. $0.05/\text{\# SNPs}$ (e.g. 10^-7^ if 500\,000 SNPs are tested). Then, genome-wide significant SNPs undergo a follow-up study on an independent cohort.

The 1000 Genome Projects catalogued 84.7 million SNPs across multiple human populations [@the_1000_genomes_project_consortium_global_2015]. However GWAS do not need to survey all of them to obtain a representative view of the genome. Instead, they exploit the correlations between the variants consequence of the genetic history, a phenomenon known as linkage disequilibrium (LD). Thanks to LD, GWAS survey the whole genome in an inexpensive fashion using SNP arrays that contain only a small fraction of the known variants [@visscher_10_2017]. In consequence, the SNPs associated with a disease are likely not the causal ones, but SNPs close to the causal variants, and hence in LD with them.

Since the first GWAS in the late 2000s, more than 5\,600 studies have shed light into the genetics of complex traits, identifying more than 70\,000 variant-trait associations [@buniello_nhgri-ebi_2019]. From this community effort we took a few lessons about the architecture of complex traits. First, GWAS confirmed the infinitesimal model in all studied complex traits, whose variance can only be explained but by many loci with small effect sizes [@visscher_10_2017]. Such explanatory variants tend to be located in chromatin that is open and expressed in the tissues relevant for the disease [@boyle_expanded_2017]. Also, even genes functionally implicated in the disease explain a small fraction of the trait variance [@boyle_expanded_2017]. In fact GWAS has revealed widespread pleiotropy, as the same genomic regions are often found in association to multiple traits [@visscher_10_2017]. This has profound implications, showing how interrelated different biomolecules are. In fact, holistic models that add nuances to the infinitesimal model have been suggested, like the *omnigenic model* [@boyle_expanded_2017]. This model postulates that only a few core genes are directly implicated in the disease, and alterations on them have a strong effect. But alterations in many other, unrelated, genes can also lead to disease as they propagate through biological networks to affect the functionality of those core genes. A more complete view of biological networks and disease is available in Chapter \@ref(genesis).

### Challenges {#intro-gwas-challenges}

The discovery of the genetic basis of complex diseases is hindered by several intrincacies in both the GWAS setting and the underlying biology. That is why, despite its success in identifying associated variants, all of them together do not explain the whole genetic variance of the disease. Possible explanations include the involvement of a large number of common variants with small effects; of rare variants with large effects not covered in GWAS; of epistasis; and of unaccounted gene-environment interactions [@manolio_finding_2009]. My thesis involved developing methodologies that tackle these challenges, which I present below.

#### Low statistical power

Due to the low effects sizes implied by the common disease / common variant hypothesis, GWAS would require very large sample sizes. Yet, for practical limitations, they have traditionally remained in the thousands, with the latest studies raising them up to hundreds of thousands [@sudlow_uk_2015]. However, the number of biomarkers required to scan the whole genome is even larger, from hundreds of thousands to millions. In consequence, GWAS is conducted in an ultradimensional setting which, jointly with the small effect sizes, leads to low statistical power [@wray_pitfalls_2013;@visscher_heritability_2008;@button_power_2013]. Statistical power also takes a hit due to the partial correlations between the tests, as multiple test correction procedures like Bonferroni consider the statistical tests independent, and hence overcorrect [@wang_statistical_2018]. Statistical power can be further reduced when the causal SNPs are in weak LD with the closest genotyped variants, for instance if they are rare, and when the phenotype is heterogeneous [@visscher_10_2017], as is common in complex diseases. This has several implications. First, that by definition we have a small chance of discovering effects that are true. Second, it raises the probability of a discovery to be a false positive. Third, when an underpowered study discovers a true effect it is more likely to overestimate its effect size. One practical consequence of this low power in GWAS is the difficulty to reproduce results [@visscher_heritability_2008].

#### Choice of encoding {#encoding}

The most commonly used association tests in GWAS require making assumptions on the mode of action of the SNPs (dominance, codominance, etc.). For instance, the logistic regression presented in Equation \@ref(eq:gwas-logreg) needs a single number that represents the two alleles of each individual. Converting the genotype into such a number is known as *encoding a genotype*. There are several such encodings, and its choice have implications on the study: choosing an encoding which diverges from how the SNP truly acts will reduce the statistical power [@romagnoni_comparative_2019]. A common one is the additive encoding, which assumes that the minor allele is responsible for the phenotype, proportionally to the number of copies; hence, the major allele in homozygosis is represented by a 0, the heterozygous genotype by a 1, and the minor allele in homozygosis by a 2. In Appendix \@ref(block-hsic-lasso) we explore a feature selection algorithm which, when applied to GWAS, makes no assumptions on the mode of action of the SNP.

#### Estimating individual risk

Once the genetic architecture is fully understood, it will be possible to estimate the full genetic component of a patient. This involves moving moving from population-level associations, to an individual assessment [@wray_pitfalls_2013]. Conventionally these are done through polygenic risk scores, which consist on a linear combination of genotypes from the associated loci in a patient, weighted by their effect sizes. However, their utility is far from clinical applications [@visscher_10_2017]. In this thesis, I apply machine learning algorithms for sample classification in Chapter \@ref(genesis) (Section \@ref(results-benchmark)) and in Appendix \@ref(block-hsic-lasso). Also, it is worth stressing that even when the whole genetic architecture of a disease is uncovered, the $R^2$ of a linear predictor would be upper bounded by the heritability ($h^2$). A complete prediction will require, in addition, fully understanding epistasis (Section \@ref(gwais)), environmental effects, and the interactions between environment and genetics.

#### Population structure {#pop-structure}

As explained before, the GWAS exploits LD to avoid genotyping all known variants. However, these correlations between SNPs depend on the evolutionary history of each sample, and hence are population specific. Therefore, GWAS designs must account for samples with different ancestries or, in other words, capture the population structure in the data. Failure to do so might lead to overestimating allelic and genotypic frequencies, reducing statistical power and producing spurious associations [@wang_statistical_2018]. Population structure can be captured by the principal components of the genotype matrix, and hence they are often used to account for it [@price_principal_2006]. For instance, a logistic regression using the main principal components as covariates is often used to obtain measures of association at the SNP level correcting by potentially confounding population structure [@michailidou_genome-wide_2015;@michailidou_association_2017;@ellinghaus_analysis_2016].

Another consideration is that most GWAS have been carried out on populations of European ancestry, which have a reduced variability in comparison to other human populations [@visscher_heritability_2008]. In fact, studies on non-European populations have yielded a big number of new, intriguing variants.

#### Interpretability {#intro-interpretability}

By design, a genotyped SNP acts as a tag for the region in the genome in LD with it. Hence, even if that "tag SNP" shows statistical association with a disease, fine-mapping studies are needed to pinpoint the specific SNP that gives the susceptibility to it. From a computational perspective, a frequent strategy to link tag SNPs to a function is identifying the genes under the influence of their genomic region [@lehne2011snps;@Jorgenson], as those are considered the functional unit of inheritance. For instance, a SNP might affect a gene by provoking an amino acid change in the protein product, or altering its gene expression or its splicing. Yet, mapping genomic regions to the genes they might influence is not trivial. In the literature, we find three ways of of doing so. The first one is the physical mapping, which maps that region to the genes whose genomic coordinates overlap with it. Often the gene boundaries are expanded by a fixed number of kilobases, as SNPs in promoters or nearby enhancers can affect gene expression [@segre_common_2010]. Nonetheless, physical mapping can be ambiguous due to the overlap between genes (as I illustrate in Section \@ref(results:consensus)). The second SNP-gene mapping is through gene expression regulation, when SNPs in the associated genomic region are significantly associated to the expression of a gene. Such loci are known as eQTL. In this regard, the GTEx project [@gtex_consortium_genetic_2017], a project that genotypes multiple human subjects and extracts expression information from their different tissues, is a useful source of tissue-specific eQTL. Gene expression mapping is not exempt of the overlap problem that occurs in the physical mapping. Solutions which consider LD patterns and the association across the whole genomic region have been proposed [@liu_abundant_2019]. The third SNP-gene mapping is based on the 3D structure of the genome, which causes distant genomic regions to be close in space [@spielmann_structural_2018]. In this situation, SNPs in a genomic region are associated to neighboring genes in the tridimensional space.

Once a set of SNPs is reliably mapped to a gene, multiple tests extist to aggregate their results into a gene statistic and measure the association of the gene as a whole with the disease. Aggregation of such SNP statistics into gene statistics is likely to increase the statistical power when dealing with complex diseases [@wu2010novel]. Multiple approaches exist, like SKAT [@wu_rare-variant_2011], VEGAS2 [@mishra_vegas2:_2015], or COMBAT [@wang_combat:_2017].

## Epistasis {#intro-epistasis}

Epistasis is the phenomenon where the effect of one locus on the phenotype depends on the state of one or more additional loci. It has two variants: biological and statistical [@moore_traversing_2005]. Biological epistasis refers to the physical interaction occurring between the loci, for instance via their protein products. It is partly a consequence of the redundancy of biological mechanisms, which requires alterating multiple sites in order to alter their broader biological function [@niel_survey_2015]. Multiple cases of biological epistasis contributing to phenotypes in model organisms have been reported in the literature [@mackay_epistasis_2014]. Statistical epistasis, by contrast, is the observation that the association between one locus and the phenotype changes across the level of the other locus. In essence, statistical epistasis refers to the biological epistasis detectable at the population level. In this regard, links between epistasis and complex diseases like Alzheimer's disease [@combarros_epistasis_2009], inflammatory bowel disease [@cho_identification_1998] and hypertension [@kimura_multilocus_2012] have been found. As in this thesis I worked exclusively in the detection of the latter kind, for brevity's sake I will refer to it simply as *epistasis*.

Despite the links between epistasis and complex disease mentioned above, estimating the magnitude of its contribution to complex diseases in humans is hard [@gusareva_practical_2014]. Nonetheless, @zuk_mystery_2012 proposed that incorrectly accounting for epistasis might be behind the so-called *missing heritability* of complex traits. Additionally, studies of traits in model organisms suggest that epistasis plays a key role of their genetic architecture [@mackay_epistasis_2014]. This motivates further studying the involvement of epistasis in complex diseases.

## Genome-wide association interaction studies {#gwais}

Genome-wide association interaction studies (GWAIS) share the experimental design with GWAS, but focus on the detection of epistasic associations. The identification of gene-gene interactions is crucial to understand the functional basis of the association: if two interacting SNPs alter the same gene, and both alterations are required to modify its function, if they affect different genes from the same pathway, from two different pathways that cross-talk, etc. As opposed to GWAS, no standard GWAIS protocol exists yet, although some general recommendations have been issued [@gusareva_practical_2014; @ritchie_search_2018].

Due to their similar experimental design, GWAIS and GWAS share the challenges outlined in Chapter \@ref(intro-gwas-challenges). Nonetheless, such problems are often aggravated in GWAIS. For instance, a larger number of tests implies a further reduction of the statistical power. Interpretation is also more complicated, as two or more SNPs need to be mapped to their respective genes [@gusareva_practical_2014]. However it requires two additional considerations, related to the multiple genetic scenarios in which epistasis can occur. As illustration, @li_complete_2000 estimated that there are 50 different fully penetrant disease models involving two loci in epistasis. The first consideration is that a number of arbitrary choices must be made, like the order of the explored interactions, whether to pre-filter the data according to function or detectable main effects, or the genetic encoding (see Section \@ref(encoding)) [@romagnoni_comparative_2019]. For instance, fourth order epistasis involves four different loci jointly contributing to a phenotype. Nonetheless, as the number of interactions grows exponentially with the epistasis order, most methods and studies focus on second order epistasis. The second consideration is that epistasis introduces the challenge of quantifying and assessing the significance of an statistical interaction. Multiple strategies to detect epistasis have been proposed, from logistic regression with an interaction term to deep learning. (@niel_survey_2015 is a comprehensive review of the main epistasis detection strategies.) In general, this diversity comes from tackling different aspects of the computational and statistical issues that arise from the large number of potential interactions. For instance, logistic regression with an interaction term is an exhaustive method with a parallelized implementation [@chang_second-generation_2015], but makes strong assumptions on the underlying relationship between the genotype and the phenotype. Hence, statistical power is compromised when that model is inaccurate. On the other hand, MDR [@moore_flexible_2006] is model-free and exhaustive, but is limited to case-control phenotypes and its inability to compute P-values analytically makes it slower. In Chapter \@ref(epi-network) we examine different epistasis detection methods.

## Diseases studied in this thesis

The bulk of my work in this thesis revolved around two complex diseases: familial breast cancer and inflammatory bowel disease.

### Breast cancer

Cancer is the name of a collection of related diseases. Specifically, all cancers undergo an uncontrolled proliferation of the patient cells, which spread into surrounding tissues. In a normal organism, cells grow and divide to maintain the tissue. As cells grow old, or accumulate too much damage, they undergo cell death and new cells will take their place. However, in cancer, this orderly process breaks down. Cells refuse to die when they get old, or accumulate damage. New cells are formed even if they are not needed. In consequence they form purpose-less growths called tumors.

This abnormal behavior occurs as consequence of the alteration of crucial genes. These alterations can be inherited from our parents, or acquired during our lifetime, due to replication errors or exposure to DNA-damaging substances. As with any other phenotypic trait, the likelihood of developing cancer is determined by the interplay between the genetic background and the environment: genetic backgrounds may favor or hinder the acquisition of mutations, and so do environmental factors.

Breast cancer occurs when breast cells undergo this uncontrolled proliferation. In most of the cases they begin in the ducts that carry the milk to the nipple. However the tumor can originate in other tissues, mainly the milk-producing gland.

Breast cancer is the second most commonly diagnosed cancer among women, after non-melanoma skin cancer. It is also the second leading cause of cancer deaths after lung cancer. It is mostly a women's disease: only about 1% of the diagnosed cases are in men. Among the most important risk factors for breast cancer we can highlight age, family history, reproductive history, usage of oral contraceptives and exposure to radiation. Most breast cancers occur after age 50.

Breast cancer is a very heterogeneous disease: while all the tumors appear in the same organ, the tissue where they originate, the molecular mechanism involved, the response to therapy, etc. vastly differ. In general, clinical decisions are based on the expression of 3 molecular markers: the expression of the endocrine receptors for estrogen and progesterone (ER and PgR, respectively) and the expression of the HER2 gene. The proteins these three genes code for are targets for chemotherapy. Based on the results, we distinguish three main breast cancer subtypes: hormone receptor positive, HER2 positive and triple negative.

- Hormone receptor positive (HR+): HR+ tumors include the tumors expressing ER and/or PR, which respectively depend on estrogen and/or progesterone to grow. They happen mostly in postmenopausal women. HR+/HER2- also known as LuminalA are the majority of breast cancers (60-75%) and they present the best prognosis.
- HER2 positive (HER2+): HER2+ tumors depend on the protein HER2/neu (human epidermal growth factor receptor 2) to proliferate, which they over-express. HR+/HER2+ (also known as LuminalB) constitute 10% of the cases, while HR-/HER2 (also known as HER2-enriched) involves 5% of them. There are a couple of very effective drugs against it.
- Triple-negative: Triple-negative tumors lack the expression of all three of ER, PgR and HER2. These patients present a worse prognosis than the rest, due to the aggressiveness of the tumor and the lack of a clear molecular target. Still, the main treatment is chemotherapy.

In the mid-19th century a French medical doctor, Pierre Paul Broca, reported for the first time a case of familial breast cancer [@nielsen_hereditary_2016]. Indeed, his wife acquired breast cancer, as many women in her family had for, at least, 4 generations. Cases of familiar breast cancer usually occur in women younger than 50 years, and bilateral primary breast tumors are frequent. Epidemiological studies later quantified the relative risk conferred by a the presence of multiple breast cancers in the family at 2.7. Moreover, they exhibit a higher likelihood of acquiring triple-negative breast cancer.

It was not until the late 20th century that two genes involved in DNA repair, *BRCA1* and *BRCA2*, were associated with hereditary breast and ovarian cancer (HBOC). Some mutations in these genes increase the risk of developing breast cancer, giving respectively a 57–65% or 45–55% risk of developing breast cancer by age 70 among women. For that reason, *BRCA1* and *BRCA2* mutations are rare in most populations (1 of 400). 

HBOC follows an autosomal dominant inheritance pattern. While approximately 5–10% of all patients with breast cancer exhibit a monogenic predisposition to breast and ovarian cancer, only about 25% of them harbor *BRCA1/2* mutations. An additional 23 genes have been associated with familial breast and/or ovarian cancer (Table \@ref(tab:hboc-genes)).

Nearly all known HBOC susceptibility genes encode tumor suppressors that participate in genome stability pathways (homologous recombination repair, replication fork stability, transcription–replication collisions, mismatch repair, and DNA damage signaling, checkpoints and cell death; see Appendix \@ref(hboc-genes) for more information).

#### The GENESIS dataset {#genesis-dataset}

In Chapter \@ref(genesis), I search for risk factors for familial breast cancer 
in the French population on the he GENE Sisters (GENESIS) study
[@sinilnikova_genesis:_2016]. Index cases are patients with infiltrating
mammary or ductal adenocarcinoma, who had a sister with breast cancer,
and who have been tested negative for *BRCA1* and *BRCA2* pathogenic
variants. Controls are unaffected colleagues and/or friends of the
cases, born around the year of birth of the corresponding case ($\pm 3$
years). We focused on the 2\,577 samples of European ancestry, of which
1\,279 are controls and 1\,298 are cases. The genotyping was performed
using the iCOGS array, a custom Illumina array designed to study genetic
susceptibility of hormone-related cancers [@sakoda_turning_2013]. It
contains 211\,155 SNPs, including SNPs putatively associated with breast,
ovarian, and prostate cancers, SNPs associated with survival after
diagnosis, and SNPs associated to other cancer-related traits, as well
as functional candidate variants in selected genes and pathways.

### Inflammatory bowel disease

Inflammatory bowel disease (IBD) is a group of complex diseases that, as the name indicates, share a common theme of inflammation of the intestines. The two main subtypes are ulcerative colitis and Chron's disease. Clinically, these two share a lot of the symptoms, mainly intermittent abdominal pain and diarrhea [@liu_genetics_2016]. However, they differ in the specific regions of the digestive tract that get affected, as well as the specific lesions. IBD's incidence worldwide has been continually growing, specially in newly industrialized countries, and after decades of growth it has stabilized in North America, Oceania and Europe [@ng_worldwide_2017]. In these latter countries the prevalence is slightly above 0.3\%. 

The genetic component of IBD was recognized more than a century ago [@ek_history_2014]. However, it was not until 2001 that the first gene, *NOD2*, was linked to IBD susceptibility. Ever since, hundreds of loci have been associated to IBD as well, in positions related to immune system genes, both innate and adaptive [@loddo_inflammatory_2015;@ellinghaus_analysis_2016;@liu_genetics_2016]. For instance *NOD2*, and other susceptibility genes like *IL23R*, and *PTPN2*, are related to cell signalling in immune cells. However, most of the loci associated to genes are rare variants (< 0.5\%) with large effect sizes. By contrast, several GWAS have identified very common SNPs (20-50\%) with small effect sizes (odds ratio < 1.1) [@liu_genetics_2016], but which do not encode any coding change [@jostins_hostmicrobe_2012]. This raises questions about the underlying biology (see Section \@ref(intro-interpretability)). For a comprehensive view of the genetics of IBD, interested readers can read @liu_genetics_2016 and @loddo_inflammatory_2015.

#### The IIBDGC dataset {#iibdgc-dataset}

In Chapters \@ref(epi-snp2gene) and \@ref(epi-network), I search for epistatic interactions giving susceptibility to IBD. Specifically, I examine the largest case-control GWAS on ulcerative colitis and Chron's disease to date [@jostins_hostmicrobe_2012], produced by the International Inflammatory Bowel Disease Genetics Consortium (IIBDGC). The dataset contain 66\,280 samples, out of which 32\,622 are cases and 33\,658 are controls. The Immunochip SNP array was used for the genotyping [@cortes_promise_2010], which contains 196\,524 polymorphisms, with a special focus on immunogenetics.

## Network view of complex diseases

Human biology is notoriously complicated, as sheer numbers demonstrate: to form a 70 kg man, 3.0 \texttimes 10^13^ cells [@sender_revised_2016] assemble and interact to produce and maintain 79 organs. To achieve that level of complexity, human cells depend on their genetic material, carefully tuned by epigenetics and enabled by a favorable environment. In terms of genetics, the object of my work, a human diploid genome is 6.4 billion base pairs long, and harbors 44\,393 genes, of which 20\,444 encode for a protein and 23\,949 are RNA genes. DNA, proteins and RNA are in constant interplay with each other, with the metabolites, and with the environment. Proteins physically interact with each other in highly specific ways (protein-protein interactions or PPIs). If such interactions are stable enough, proteins can assemble into large complexes to carry out particular functions. But proteins also interact with DNA to regulate gene expression (transcription factor - DNA interactions). And so on: enzymes interact with their metabolites, hormones with their receptors, the individual with their environment, etc. At a fundamental level, a person and their traits are just the emerging pattern born from the interaction of all these factors. Hence, biology, from ecosystems to molecular biology, cannot be understood if not as an interplay. Mindful of this, and enabled by the omics technologies of the 21st century, researchers have striven to capture and understand these relationships. The goal of *systems biology* is achieving a global understanding of the complex interplay that drives biology. Among all the possible biological relationships, in this thesis I focus on protein-protein interactions, including protein complexes, as their coverage is larger and their properties better understood. Such interactions are available in databases like HINT [@das_hint_2012], The BioGRID [@oughtred_biogrid_2019] or IntAct [@hermjakob_intact_2004].

Relationships between pairs of entities can be mathematically formalized as a network, which makes them analytically approachable. In such networks, often proteins or genes are the nodes, which are connected by edges in a pairwise fashion when they are functionally related. Although the edges might have directionality, often PPIN are undirected, as the direction of the edges is often unknown (transitory interactions), or inexistent (co-complexes) [@barabasi_network_2011]. The field of systems biology relies on the assumption that the network accurately captures the context a biomolecule requires to carry out its biological function. We can distinguish three levels of network properties: properties of individual nodes (local), the joint properties of groups of nodes (mesoscale), and the properties over the whole network (global). Indeed, at all three levels biological networks are structured, very differently from random networks [@barabasi_network_2011;@Chaiboonchoe2013].

At the global level, for instance, the degree, which represents the number of edges per node, follows a power-law distribution [@barabasi_network_2011]. This implies that, at the local level, a few genes participate in the majority of the edges. Such nodes are called *hubs*. Importantly, this node property is informative of the gene function: *in utero* essential genes, like knots preventing the network from falling apart, tend to be hubs. An important consequence of the structured degree distribution at the mesoscale level is the emergence of *modules*, subsets of nodes densely interconnected to each other, and sparsely to the rest. A mesoscale property of biological networks is that they have a modular structure i.e. a strong division into modules. Such modules often constitute functional units within the network, where the nodes jointly contribute to a specific function [@mitra_integrative_2013].

Another global property of a graph is the distribution of distances between pairs of nodes, that is, how many edges must be passed to travel from one node to another. Such is the notion of *path* between nodes. Often we are interested in the *shortest* path, for they reveal the fastest way information can flow from one node to another. An examination of the distribution of shortest paths in biological networks shows that all nodes are close to each other (shortest paths are, in fact, pretty short on average) [@barabasi_network_2011]. Such networks are called *small world networks*. This structure makes the flow of information resilient to the removal of nodes or edges [@Chaiboonchoe2013], a cause of biological epistasis [@niel_survey_2015].

### Networks in disease

Examining how biological networks relate to disease, and the topology around disease genes produces a nuanced approach to disease: cut the knot in the center of the web that keeps it all together, and it all will fall off; cut a bunch of peripheral, less important nodes around it, the outcome might be the same. In other words, there are many ways of producing the same disease [@leiserson_network_2013]. 

The properties of biological networks enumerated above lay down the foundations of the use of biological networks to study disease genes. For instance, they justify the *local hypothesis*, which expects genes involved in a disease to interact with each other [@barabasi_network_2011]. They also justify the *disease module hypothesis*, which expects genes involved in the same disease to share a module. Experimentally, network propagation highlights the differential topological properties of disease genes and biological networks. *Network propagation* refers to methods that use all the possible paths in the network to re-rank the genes on it [@cowen_network_2017]. They include heat diffusion, random walk, graph kernels, and even Google's search algorithm. In essence, for these methods, association of a node with a phenotype can be thought of as a volume of water: the more strongly associated, the more voluminous. Generically, each node starts with that initial volume which, iteratively, gets distributed among its neighbors. Equivalently, every node will receive a share from their neighbors'. The expectation is that truly associated genes will be densely interconnected to each other, forming cycles and modules that will keep the water from diffusing to other, uninvolved genes. After a number of steps the volume of water in each node is re-evaluated, and used to re-prioritize the genes. Using network propagation @huang_systematic_2018 recently showcased the differential topological properties of disease genes. Across different biological networks, they were able to retrieve disease-related genes with varying success using only a subset of them through a random walk with restart procedure.

Indeed, disease genes exhibit different properties than non-disease genes [@pinero_uncovering_2016; @cai_broker_2010; @furlong_human_2013; @barabasi_network_2011]. For starters, disease genes are not hubs [@cai_broker_2010; @das_hint_2012], but tend to be non-essential genes located towards the periphery. This is coherent with an evolutionary framework, where mutations in essential genes would be highly deleterious, even resulting in embryonic lethality. However, this enrichment is driven mainly by cancer genes [@pinero_uncovering_2016]. Additionally, disease genes tend to be *bottlenecks* i.e. the sole link between many peripheral genes and the rest of the network [@cai_broker_2010], suggesting that disease arises when these vulnerable regions of the network break down. 

These properties, however, do not affect equally all types of disease genes. Many of the aforementioned properties do not extend to genes identified through GWAS [@cai_broker_2010]. Differences arise even when comparing genes involved in complex diseases with those involved in Mendelian ones.

### Network-guided approaches to disease study 

As exemplified in the previous section, networks can be leveraged on to gain insight of the biology of the disease. Below I summarize several ways of doing that, on which I have worked on during my PhD.

#### High-score subnetwork search {#intro-hs3}

One of the focuses of my thesis was the study of networks where the nodes are scored by their association to the disease. The scores might come from omics experiments, or from *a priori* knowledge of disease genes. In essence, such methods look for connected subnetworks made of nodes with high scores. Lacking a term broadly agreed upon in the community, I refer to such algorithms as high-score subnetwork search. Although heterogeneous, to some extent all existing approaches are based on the *guilt-by-association principle* (Figure \@ref(fig:network-gwas)): nodes nearby other nodes associated to the disease are suspect of being associated themselves, even if their association is non-significant by conventional standards. However, taking only the genes associated with those associated would be prone to false positives, as often networks include edges that are not relevant for the biological problem at hand. Several high-score subnetwork search methods are available in the literature. Essentially, they differ in the considerations they make on what the solution looks like. In Chapter \@ref(genesis) we describe a representative set of these methods, critically discuss their performance at biomarker discovery on the GENESIS dataset, and compare their strengths and weaknesses.

(ref:fig-network-gwas-caption) Overview of the high-score subnetwork search applied to GWAS. **(A)** From a set of GWAS results, first **(B)** SNP P-values are assigned to genes (e.g. by physical mapping). **(C)** The significance of the phenotype-association for each gene is based on the assigned P-value, which is overlaid on a co-functional network. **(D)** The significance of each gene is boosted by guilt-by-association, resulting in additional candidate genes (e.g., Gene D) for the phenotype. Figure and caption adapted from @lee_aragwab_2018, licensed under CC BY 4.0.

```{r network-gwas, echo=FALSE, fig.cap='(ref:fig-network-gwas-caption)'}
knitr::include_graphics('fig/dl_network_gwas.jpg', dpi = NA)
```

<!--
TODO
MOVE TO THE GENESIS PAPER
Nonetheless, these methods are not exempt of problems. For instance, these methods are heavily affected by the *small world* nature of biological networks, which make it relatively easy to find that two associated nodes are close [@pinero_uncovering_2016]. Additionally, when the biological network contains edges that are irrelevant for a particular disease the complexity of the problem is increased, and the signal decreased. Also [@mitra_integrative_2013].
-->

#### Module detection

In the context of disease, densely interconnected subnetworks (modules) of consistently altered nodes might represent the mechanisms that lead to it. Hence, their identification can provide insights into its etiology. In Appendix \@ref(smartas) we apply module detection techniques to a subnetwork of genes with abnormal splicing in cancer.

#### Aggregation of networks

During my research, often I obtained multiple high-scoring subnetworks using different approaches, which provided complementary perspectives of the same disease. In consequence, I was interested in integrating them into a single subnetwork, which I would analyze. In Chapter \@ref(genesis) we discuss a naive way of aggregating subnetworks from different high-score subnetwork search algorithms, were the edges are unweighted. In Chapter \@ref(epi-network) we create an epistasis network from interactions from several epistasis detection methods, whose edges are weighed by the confidence they exist.

## Contributions

The object of my thesis was the methodological study and application of network methods to GWAS data. In essence, networks contain prior information, which can be traded for statistical astringency in our analysis. In other words, if genes strongly associated to a disease, albeit non-significantly so, are interconnected in an underlying biological network, we are more likely to believe they represent a consistently altered biological mechanism.

In collaboration with Nadine Andrieu and Fabienne Lesueur (Institut Curie), and working closely with Christine Lonjou (Institut Curie), I applied six high-score subnetwork search to the GENESIS dataset (Section \@ref(genesis-dataset)). Our goal was to find new biomarkers for breast cancer susceptibility. My work on GENESIS is explained in Chapter \@ref(genesis). In summary, I applied six different network-based, biomarker discovery methods to the GENESIS dataset. The methods provide a representative view of the high-score subnetwork search field. We performed a methodological comparison and a benchmark of the methods, highlighting their strengths and weaknesses. Finally, we conclude that combining the methods provide a more complete answer than any of the individual solutions. Our network analysis recovers both genes and genomic regions previously found in association with breast cancer susceptibility, as well as new genes. Importantly, all of those genes are tied in a subnetwork, providing a rationale on how alterations on those genes might lead to disease.

My first approach to the problem involved working on SConES [@azencott_efficient_2013], one of the high-score subnetwork search methods examined in Chapter \@ref(genesis). I give a more detailed explanation of this algorithm, together with my work on it in Chapter \@ref(martini). In summary, I worked on applying SConES to case-control datasets, and strategies to parametrize it. Additionally, in *martini* SConES can take user-provided covariates. Regarding the former, as SConES implemened exclusively the regression version of SKAT [@wu_rare-variant_2011;@ionita-laza_sequence_2013], I implemented two ways of measuring association between a SNP and a binary phenotype. Regarding the latter, SConES has two parameters, $\eta$ and $\lambda$, which control the sparsity and the inter-connectedness of the selected SNPs, respectively. Originally the parameters that produced the most stable solution were selected. I explored the impact of penalized-likelihood measures like BIC, AIC, and AICc, which score a set of features based on both their sparsity and the accuracy of a linear classifier built on them. The product of my work is published as *martini* [@martini], an R package published in the peer-reviewed R repository Bioconductor [@bioconductor].

In 2019, when I started working on epistasis detection, we established a collaboration with Kristel Van Steen (University of Liège, Belgium). Specifically, I worked closely with Diane Duroux, a PhD student in her research group. Our goal was to build an epistasis gene network of inflammatory bowel disease. I discuss our efforts in this regard in Chapter \@ref(epi-network). For that purpose, we surveyed suitable epistasis detection methods, applied them to the IIBDGC dataset (Section \@ref(iibdgc-dataset)), and integrated the solution. However, as a previous step we needed to appropriately map epistatic SNP networks to epistatic gene networks. I describe this work in Chapter \@ref(epi-snp2gene). In summary, we examined and evaluated four different mappings (physical, eQTL, chromatin, and the three together).

In collaboration with Makoto Yamada (RIKEN AIP, Japan) and Samuel Kaski (Aalto University, Finland), I developed block HSIC Lasso, a general-purpose non-linear feature selector. The work involved modifying an existing algorithm, HSIC Lasso, to reduce its memory consumption. On top of that, we worked on both improving its performance and solving numerical issues in edge cases. The algorithm is implemented as the Python package *pyHSICLasso*, available on both PyPI and GitHub. Then, I characterized the algorithm and applied it to several biological datasets. Crucially, three of the datasets were GWAS, which was a milestone in terms of the number of features block HSIC Lasso can handle. Thanks to this work we analyzed the impact of considering non-redundancy and non-linear models when selecting SNPs for patient classification. We describe the algorithm and our conclusions in the article *Block HSIC Lasso: model-free biomarker detection for ultra-high dimensional data* published in the proceedings of the ISMB/ECCB 2019 [@climente-gonzalez_block_2019]. The full manuscript is available in Appendix \@ref(block-hsic-lasso).

Additionally, I worked on the involvement of alternative splicing in cancer with Eduardo Eyras (Australian National University), a continuation of my previous research. During 2016 and early 2017, we prepared the answer to reviewers for the article *The functional impact of alternative splicing in cancer*, which was published in Cell Reports in August 2017 [@climente-gonzalez_functional_2017]. The full manuscript is available in Appendix \@ref(smartas). I also explored evidence of epistasis in cancer, looking for mutual exclusivity between alterations in alternative splicing and somatic mutations. We searched for evidence of mutual exclusivity at both the gene and the pathway level. Among others, we found a compelling pattern in *GATA3*. This finding was published in Cell Reports in March 2018, in the broader article *Systematic analysis of splice-site-creating mutations in cancer* [@jayasinghe_systematic_2018]. The full manuscript is available in Appendix \@ref(scm). In addition, I compiled the code required for the analyses in @climente-gonzalez_functional_2017, and created a Python package with a clean interface. The package, *spada*, is available in both PyPI and GitHub. This package searches alterations of alternative splicing in a user-provided dataset, and maps them to functional consequences at the protein level. Specifically, such consequences are the removal/addition of functional modules to the protein (e.g. domains) and the loss or gain of protein-protein interactions. Lastly, I applied *spada* to two leukemia datasets from TARGET, an NIH program that aims to understand the molecular basis of several childhood cancers. This last analysis is still on-going.

Lastly, I carried out my research committed to open, reproducible science. As such, all my projects have an associated, version-controlled, laboratory notebook, which includes as much data as I am allowed to share. All such laboratory notebooks are made out public when the paper is. All the scripts I developed are open source, under permissive MIT license. Specifically, I made an effort to develop project-independent tools, which are useful to anyone which needs similar to the ones I had. These tools are available on GitHub (https://github.com/hclimente/gwas-tools).

<!--chapter:end:01-intro.Rmd-->

# Combining network-guided GWAS to discover susceptibility mechanisms for breast cancer {#genesis}

The contents of this chapter are based on *Héctor Climente-González, Christine Lonjou, Fabienne Lesueur, GENESIS investigators, Dominique Stoppa-Lyonnet, Nadine Andrieu and Chloé-Agathe Azencott*, Combining network-guided GWAS to discover susceptibility mechanisms for breast cancer (manuscript in preparation). Parts of this work were presented in ASHG 2018 in the poster *Héctor Climente-González, Christine Lonjou, Fabienne Lesueur, GENESIS investigators, Dominique Stoppa-Lyonnet, Nadine Andrieu and Chloé-Agathe Azencott*, Judging genetic loci by the company they keep: comparing network-based methods for biomarker discovery in familial breast cancer. 

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
library(magrittr)
library(kableExtra)

cat("\\begin{chapabstract}{Abstract:}
Systems biology provides a comprehensive approach to biomarker discovery and biological hypothesis building. It does so by jointly considering the statistical association between a gene and a phenotype, and the biological context of each gene, represented as a network. In this chapter we study how six network methods can be used to discover new biomarkers for breast cancer susceptibility by searching subnetworks highly associated to this  phenotype. We interrogate a familial breast cancer genome-wide association study (GWAS) focused on BRCA1/2 negative French women. By trading statistical astringency for biological meaningfulness, most network methods get more compelling results than standard SNP- and gene-level analyses, recovering causal subnetworks tightly related to cancer susceptibility. We perform an in-depth benchmarking of the methods with regards to size of the solution subnetwork, their utility as biomarkers, and the stability and the runtime of the methods. Interestingly, a combination of solution subnetworks provided a concise subnetwork of 93 genes, enriched in known breast cancer susceptibility genes (BABAM1, BLM, CASP8, FGFR2, and TOX3, Fisher's exact test P-value = $7.8 \\times 10^{-5}$) and more central than average. Additionally, it includes subnetworks of mechanisms related to cancer, like protein folding (HSPA1A, HSPA1B, and HSPA1L) or mitochondrial ribosomes (MRPS30, MRPS31, MRPS18B). We also observed a general dysregulation in the neighborhood of COPS5, a gene related to multiple hallmarks of cancer.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}
La biologie des syst\\`emes permet de d\\'evelopper une approche globale de la 
recherche de biomarqueur et de la construction d'hypoth\\`eses biologiques. 
Elle rend en effet possible de replacer l'association statistique entre 
un g\\`ene et un ph\\'enotype dans le contexte biologique de ce g\\`ene, gr\\^ace \\`a 
des r\\'eseaux biologiques. Dans ce chapitre, nous \\'etudions l'applicabilit\\'e 
de six m\\'ethodes bas\\'ees sur des r\\'eseaux pour la d\\'ecouverte de nouveaux 
biomarqueurs de susceptibilit\\'e dans le cancer du sein, en cherchant des 
sous-r\\'eseaux fortement associ\\'es \\`a un ph\\'enotype. Nous \\'etudions une GWAS 
sur le cancer du sein familial, concernant des femmes françaises BRCA1/2 
n\\'egatives. La plupart de ces m\\'ethodes permettent d'obtenir des r\\'esultats 
plus convaincants que des analyses standard, et retrouvent des 
sous-r\\'eseaux fortement li\\'es \\`a la susceptibilit\\'e dans le cancer. Nous 
proposons une comparaison d\\'etaill\\'ee des m\\'ethodes en terme de taille de 
la solution, de son utilisabilit\\'e comme biomarqueur, de stabilit\\'e ainsi 
que de temps de calcul. Nous combinons les sous-r\\'eseaux retourn\\'es par 
ces m\\'ethodes en un r\\'eseau restreint \\`a 93 g\\`enes, enrichi en g\\`enes de 
susceptibilit\\'e au cancer du sein connus, (BABAM1, BLM, CASP8, FGFR2, and 
TOX3, p-valeur du test exact de Fisher = $7.8 \\times 10^{-5}$), et plus 
central que la moyenne. De plus, ce r\\'eseau contient des sous-r\\'eseaux de 
m\\'ecanismes li\\'es au cancer, tels que le repliement des prot\\'eines (HSPA1A, 
HSPA1B, and HSPA1L) ou les ribosomoes mitochondriaux (MRPS30, MRPS31, 
MRPS18B). Nous observons aussi une dysr\\'egulation g\\'en\\'eralis\\'ee au 
voisinage de COPS5, un g\\`ene li\\'e \\`a plusieurs caract\\'eristiques 
fondamentales du cancer.
\\end{chapabstract}")
```

## Introduction

Network-based biomarker discovery methods provide a comprehensive
statistical framework which includes the structure of biological data
tackle the the challenges in the GWAS field
(Section \@ref(intro-gwas-challenges)). In essence, each SNP has a measure of
association with the disease, given by the experiment, and functionally
biological relationships, given by a network built on prior knowledge.
Then, the problem becomes finding a functionally-related set of genes
that is highly associated with the disease. Such set is easier to 
interpret, as it is related to known biological mechanisms, and trade 
statistical significance for biological relevance, potentially improving
the power of the study. This problem is addressed by the high-scoring 
subnetwork search methods (Section \@ref(intro-hs3)) which use a 
guilt-by-association strategy to identify disease 
susceptibility genes [@azencott_network-guided_2016].
Different solutions have been proposed, often stemming from divergent different
mathematical frameworks and considerations of what the optimal solution
looks like. Some methods strongly constrain the problem to certain kinds
of subnetworks. Such is the extreme case of LEAN
[@gwinner_network-based_2016], which focuses on star subnetworks, i.e.
instances were both a gene and its direct interactors are associated
with the disease. Other algorithms, like dmGWAS [@jia_dmgwas:_2011] and
heinz [@dittrich_identifying_2008], focus on interconnected genes with
high association with the disease. However, they differ in their
tolerance to the inclusion of lowly associated nodes, and the possible
number of disconnected subnetworks in the solution. Lastly, other
methods also consider the topology of the network, favoring solutions
that are densely interconnected; such is the case of HotNet2
[@leiserson_pan-cancer_2015], SConES [@azencott_efficient_2013], and
SigMod [@liu_sigmod:_2017].

In this chapter, we analyze the effectiveness of these six methods to
discover new biomarkers on GWAS data. We focus on the GENESIS dataset
[@sinilnikova_genesis:_2016], a study of familial breast cancer
conducted in the French population (Section \@ref(genesis-dataset)). 
After following a classical GWAS
approach, we use these network-based methods to recover additional
familial breast cancer biomarkers. Some of them are known, while others
are specific to this dataset. Lastly, we carry out a comparison of the
solutions obtained by the different methods, and aggregate them to
obtain a consensus network of predisposition to familial breast cancer.

## Methods

### Data preprocessing and quality control

In this study we used the GENESIS dataset, described in Section \@ref(genesis-dataset).
We discarded SNPs with a minor allele frequency lower than 0.1%, those
not in Hardy - Weinberg equilibrium in controls (P-value \<0.001), and
those missing on more than 10% of the samples. A subset of 20 duplicated
SNPs in *FGFR2* were also removed. In addition, we removed the samples
with more than 10% missing genotypes. After control for relatedness, 17 
additional samples were removed (6 for sample identity error, 6 false 
“friend/control” having family link with other samples, 3 “friend/control” 
having a high relatedness score). Lastly, based on study selection criteria, 
11 other samples were removed (1 control having cancer, 4 index cases with no 
affected sister, 3 half-sisters, 1 sister with CLIS, 1 with BRCA mutation 
detected in the family, 1 with molecular diagnosis not received). The final 
dataset included 1\,271 controls and 1\,280 cases, genotyped over 197\,083 SNPs.

We looked for population structure that could create confounding
associations (Section \@ref(pop-structure)). A PCA revealed no differential 
population structure between cases and controls (Figure \@ref(fig:pcs)).
Independently, we did not find evidence of genomic inflation ($\lambda$
= 1.05) either, further confirming the absence of confounding population
structure.

(ref:fig-pcs-caption) GENESIS shows no differential population structure between cases and controls. **(A,B,C,D)** Eight main principal components computed on the genotypes of GENESIS. Cases are colored in green, controls in orange.

```{r pcs, echo=FALSE, fig.cap='(ref:fig-pcs-caption)'}
knitr::include_graphics('fig/sfigure_1.pdf', dpi = NA)
```

### High-score subnetwork search algorithms {#sec:org6ced6c9}

#### SNP and gene association {#methods:node-score}

To measure association between a genotype and the phenotype, we
performed a per-SNP 1 d.f. $\chi$^2^ allelic test using PLINK v1.90
[@chang_second-generation_2015]. Then, we used VEGAS2v2 to compute the
gene-level association score from the SNP P-values [@mishra_vegas2:_2015]. 
In order to map SNPs to genes we used their overlap on the
genome: all SNPs located within the boundaries of a gene, $\pm 50$ kb, were
mapped to that gene. To compute the gene association we used the 10% of
SNPs linked to the gene with lowest P-values. We used the 62\,193 genes
described in GENCODE 31 [@frankish_gencode_2019], although only 54\,612
could be mapped to at least one SNP. Out of those, we focused
exclusively on the 32\,767 that could be mapped to an HGNC symbol. Out of
the SNPs 197\,083 remaining after quality control, 164\,037 mapped to at
least one of these genes.

We use such mapping to compare the outputs of methods that produce SNP-
to those that produce gene-lists, and vice versa. In the former case, we
consider any gene that can be mapped to any of the selected SNPs as
selected as well. In the latter, we consider all the SNPs that can be
mapped to that gene as selected by the method.

#### Mathematical notation {#methods:notation}

In this chapter, we use undirected, vertex-weighted networks, or graphs,
$G = (V,E,w)$. $V = \{v_{1}, \dots{}, v_{n}\}$ refers to the vertices,
with weights $w: V \rightarrow \mathbb{R}$. Equivalently,
$E \subseteq \{\{x,y\} | x,y \in V \wedge x \neq y\}$ refers to the
edges. When referring to a subnetwork S, $V_{S}$ is the set of nodes in
S and $E_{S}$ is the set of edges in S. A special case of subgraphs are
*connected* subgraphs, which occur when every node in the subgraph can
be reached from any other node.

On top of a weight, nodes have other properties provided by the topology
of the graph. In this chapter we focus on two: degree centrality, and
betweenness centrality. The degree centrality, or degree, is the number
of edges that a node has. The betweenness centrality, or betweenness, is
the number of times a node participates in the shortest paths between
two other nodes.

In addition, we use several matrices that describe different properties
of a graph. The described matrices are square, and have as many rows and
columns as nodes are in the network. The element $(i,j)$ represents a
selected relationship between $v_i$ and $v_j$. The *adjacency matrix*
$W_G$ contains a 1 when the corresponding nodes are connected through an
edge, and 0 otherwise; the diagonal is zero. The *degree matrix* $D_G$
is a diagonal matrix which contains the degree of the different nodes.
Lastly, the *Laplacian matrix* $L_G$ is defined as $L_G = D_G - W_G$.

#### Methods used {#methods:methods}

Beyond the assumption that genes that contribute to the same function
will be nearby in the protein-protein interaction network (PPIN), they
might be topologically related to each other in diverse ways (densely
interconnected modules, nodes around a hub, a path, etc.). That is not
the only choice to make: how to score the nodes, whether the affected
mechanisms form a single connected component or several, how to frame
the problem in a computationally efficient fashion, what is the best
network to use, etc. In consequence, multiple solutions have been
proposed. In this chapter, we examine six of them: five that explore the
protein-protein interaction network, and one which explores SNP
networks. We selected methods that were open source, had an
implementation available, and an accessible documentation. Their main
differences are summarized in Table \@ref(tab:method-comparison).

(ref:tab-method-comparison-footnote) *Field*: field in which the algorithm was developed. *Nodes*, the type of network, either gene (protein-protein interaction network usually) or a SNP network. *Exhaustive*: whether all the possible solutions given the selected hyperparameters are explored. *Solution*: additional properties are enforced on the solution subnetwork, other than being dense in high scores and connected. *Components*: number of connected subnetworks in the solution. *Input*: genotype data or GWAS summary statistics. *Scoring*: how SNP/gene P-values are transformed into node scores.

(ref:dmgwas) @jia_dmgwas:_2011

(ref:heinz) @dittrich_identifying_2008

(ref:hotnet) @leiserson_pan-cancer_2015

(ref:lean) @gwinner_network-based_2016

(ref:scones) @azencott_efficient_2013

(ref:sigmod) @liu_sigmod:_2017

(ref:logten) -log~10~(P)

(ref:chisq) $\chi^2$

```{r method-comparison, echo=FALSE}
method_summary <- rbind(
  c('dmGWAS', 'GWAS', 'Genes', 'No', '', '1', 'Summary', '(ref:logten)', '(ref:dmgwas)'),
  c('heinz', 'Omics', 'Genes', 'Yes', '', '1', 'Summary', 'BUM', '(ref:heinz)'),
  c('HotNet2', 'Omics', 'Genes', 'Yes', 'Modular', '> 1', 'Summary', 'Local FDR', '(ref:hotnet)'),
  c('LEAN', 'Omics', 'Genes', 'Yes', 'Star', '> 1', 'Summary', '(ref:logten)', '(ref:lean)'),
  c('SConES', 'GWAS', 'SNPs', 'Yes', 'Modular', '> 1', 'Genotypes', '(ref:chisq)', '(ref:scones)'),
  c('SigMod', 'GWAS', 'Genes', 'Yes', 'Modular', '1', 'Summary', '(ref:logten)', '(ref:sigmod)')
)

colnames(method_summary) <- c('Method', 'Field', 'Nodes', 'Exhaustive', 'Solution', 'Components', 'Input', 'Scoring', 'Reference')

kable(method_summary, 
      caption = 'Summary statistics on the results of multiple network methods on the gene-gene interaction network. The first row contains the summary statistics on the whole network.', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = c("striped", "scale_down")) %>%
  footnote(general = '(ref:tab-method-comparison-footnote)',
           threeparttable = TRUE, fixed_small_size = TRUE)
```

dmGWAS

:   dmGWAS searches the subgraph with the highest local density in low
    P-values [@jia_dmgwas:_2011]. To that end it searches candidate
    subnetwork solutions using a greedy, "seed and extend", heuristic:

    1.  Select a seed node.

    2.  Compute Stouffer's Z-score Z~m~ for the current subgraph as

        $$Z_m = \frac{\sum z_i}{\sqrt{k}}$$

        where *k* is the number of genes in the subgraph, z~i~ =
        $\phi$^-1^(1 - P-value~i~), and $\phi$^-1^ is the inverse normal
        distribution function.

    3.  Identify neighboring nodes i.e. nodes at distance $\le$ *d*. We
        set d = 2.

    4.  Add the neighboring nodes whose inclusion increases the Z~m+1~
        more than Z~m~ (1 + r). In our experiments, we set r = 0.1.

    5.  Repeat 2-4 until no increment Z~m~ (1 + r) is possible.

    Lastly, the module's Z-score is normalized as

    $$Z_{N}=\frac{Z_{m}-\operatorname{mean}\left(Z_{m}(\pi)\right)}{\operatorname{SD}\left(Z_{m}(\pi)\right)}$$

    where Z~m~($\pi$) represent a vector containing 100000 random
    subsets of the same number of genes.

    We used the implementation of dmGWAS in the dmGWAS 3.0 R package
    [@dmgwas]. We used the function *simpleChoose* to select the
    solution subnetwork, which aggregates the top 1% modules into the
    solution subnetwork.

heinz

:   The goal of heinz is to identify the highest-scored connected
    subgraph on the network [@dittrich_identifying_2008]. The authors
    propose a transformation of the genes' P-value into a score that is
    negative under no association with the phenotype, and positive value
    when there is. This transformation is achieved by modelling the
    distribution of P-values by a beta-uniform model (BUM) parameterized
    by the desired FDR. Thus formulated, the problem is NP-complete. To
    solve it efficiently it is re-casted as the Prize-Collecting Steiner
    Tree Problem (PCST), which seeks to select the connected subnetwork
    S that maximizes the *profit* p(S):

    $$p(S) = \sum_{v \in V_S} p(v) - \sum_{e \in E_S} c(e).$$

    were p(v) = w(v) - w' is the *profit* of adding a node, c(e) = w' is
    the *cost* of adding an edge, and $w' = min_{v \in V_{G}} w(v)$. All
    three are positive quantities. heinz implements the algorithm from
    @ljubic_algorithmic_2006, which in practice is often fast and
    optimal, neither is guaranteed. We used BioNet's implementation of
    heinz, available on Bioconductor [@beisser_bionet:_2010; @heinz].

HotNet2

:   HotNet2 was developed to find connected subgraphs of genes
    frequently mutated in cancer [@leiserson_pan-cancer_2015]. To that
    end, it considers both the local topology of the network and the
    scores of the nodes. The former is captured by an insulated heat
    diffusion process: at the beginning, the score of the node
    determines its initial heat; iteratively each node yields heat to
    its "colder" neighbors, and receives heat from its "hotter"
    neighbors, while retaining part of its own (hence, *insulated*).
    This process continues until equilibrium is reached, and results in
    a similarity matrix F. F is used to compute the similarity matrix E
    that accounts also for similarities in node scores as

    $$E = F \operatorname{diag}(w(V)),$$

    where $\operatorname{diag}(w(V))$ is a diagonal matrix with the node
    scores in its diagonal. We scored the nodes as in
    @nakka_gene_2016, assigning a score of 0 for the genes with low
    probability of being associated to the disease, and
    -log~10~(P-value) to those likely to be. In this dataset, the
    threshold separating both was a P-value of 0.125, which was obtained
    using a local FDR approach [@scheid_twilight_2005]. To obtain
    densely connected subnetworks, HotNet2 prunes E, only preserving
    edges such that w(E) \> $\delta$. Lastly, HotNet2 evaluates the
    statistical significance of the subnetworks by comparing their size
    to the size of networks obtained by permuting the node scores.
    HotNet2 has two parameters: the restart probability $\beta$, and the
    threshold heat $\delta$. Both parameters are set automatically by
    the algorithm, and are robust [@leiserson_pan-cancer_2015]. HotNet2
    is implemented in Python [@hotnet2].

LEAN

:   LEAN searches disregulated "star" gene subnetworks, that is,
    subnetworks composed by one central node and all its interactors
    [@gwinner_network-based_2016]. By imposing this restriction, LEAN is
    able to exhaustively test all such subnetworks (one per node). For a
    particular subnetwork of size *m*, the P-values corresponding to the
    involved nodes are ranked as p~1~ $\le$ ... $\le$ p~m~. Then, *k*
    binomial tests are conducted, to compute the probability of having
    *k* out of *m* P-values lower or equal to p~k~ under the null
    hypothesis. The minimum of these *k* P-values is the score of the
    subnetwork. This score is transformed into a P-value through an
    empirical distribution obtained via a subsampling scheme, where sets
    of *m* genes are selected randomly, and their score computed.
    Lastly, P-values are corrected for multiple testing through a
    Benjamini-Hochberg correction. We used the implementation of LEAN
    from the LEANR R package [@leanr].

SConES

:   SConES searches the minimal, modular, and maximally associated
    subnetwork in a SNP graph [@azencott_efficient_2013]. Specifically,
    it solves the problem
    
    \begin{equation}
    \underset{S \subseteq G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} + \underbrace{\lambda \sum_{v \in V_S} \sum_{u \not\in V_S} L_{vu} }_{\text { connectivity }}-\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}
    (\#eq:scones)
    \end{equation} 

    where $\lambda$ and $\eta$ are hyperparameters that control the
    sparsity and the connectivity of the model. Given two
    hyperparameters, the aforementioned problem has a unique solution,
    that SConES finds using a graph min-cut procedure. We used the
    version on SConES implemented in the R package martini [@martini].
    As in @azencott_efficient_2013, we selected $\lambda$ and $\eta$
    by cross-validation, choosing the values that produce the most
    stable solution across folds. Note that the solution to the above
    problem can consist of several connected subnetworks which are
    disconnected from each other. In this case, the selected
    hyperparameters were $\eta$ = 3.51, $\lambda$ = 210.29 for SConES
    GS; $\eta$ = 3.51, $\lambda$ = 97.61 for SConES GM; and $\eta$ =
    3.51, $\lambda$ = 45.31 for SConES GI.

SigMod

:   SigMod aims at identifying the most densely connected gene
    subnetwork that is most strongly associated to the phenotype
    [@liu_sigmod:_2017]. It addresses an optimization problem similar to
    that of SConES (Equation \@ref(eq:scones)), but using the adjacency
    matrix rather than the Laplacian matrix (Section \@ref(methods:notation)), 
    to quantify solutions containing many edges.

    $$\underset{S \in G}{\arg \max } \underbrace{\sum_{v \in V_S} w(v)}_{\text { association }} + \underbrace{\lambda \sum_{v \in V_S} \sum_{u \in V_S} W_{vu} }_{\text { connectivity }} -\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }}.$$

    As SConES, this optimization problem can also be solved by a graph
    min-cut approach.

    SigMod presents three important differences with SConES. First it is
    designed for gene-gene networks. Second, by replacing the adjacency
    by the Laplacian matrix, it favors subnetworks containing many
    edges. SConES, instead, penalizes connections between the
    selected and unselected nodes. Third, it returns a single connected
    subnetwork, which it achieves by exploring a grid of hyperparameters
    and processing their respective solutions. Specifically, for the
    range of $\lambda$ = $\lambda$~min~, ..., $\lambda$~max~ for the
    same $\eta$, it prioritizes the solution with the largest change in
    size from $\lambda$~n~ to $\lambda$~n+1~. Such a large change
    implies that the network is strongly interconnected. This results in
    one candidate solution for each $\eta$, which are processed by
    removing any node not connected to any other. A score is assigned to
    each candidate solution by summing their node scores and normalizing
    by size. The candidate solution with the highest standardized score
    is the chosen solution. SigMod is implemented in an R package
    [@sigmod].

#### Gene-gene network {#methods:gene-network}

Out of the six methods tested, five use a gene-gene interaction network
(Section \@ref(methods:methods)). Although their respective statistical
frameworks are compatible with any type of network (protein
interactions, gene coexpression, regulatory, etc.), for practical
reasons we focused on a PPIN, as they are interpretable, well
characterized, and most of the methods were designed to scale
appropriately to it. We built our PPIN from both binary and co-complex
interactions stored in the HINT database (release April 2019)
[@das_hint_2012]. Unless specified otherwise, we used only interactions
coming from high-throughput experiments to avoid biasing the topology of
the network by well-studied genes with more known interactions on
average. Out of the 146\,722 interactions from high-throughput
experiments that HINT stores, we were able to map 142\,541 to a pair of
HGNC symbols. The scoring function for the nodes changed from method to
method (Section \@ref(methods:methods)).

Additionally, we compared the results of the aforementioned PPIN with
those obtained on another PPIN built using interactions coming from both
high-throughput and targeted studies. In that case, out of the 179\,332
interactions in HINT, we mapped 173\,797 to a pair of HGNC symbols.

#### SNP networks {#methods:snp-network}

SConES [@azencott_efficient_2013] is the only of the studied methods
designed to handle SNP networks. As in gene networks, two SNPs are
linked in a SNP network when there is evidence of shared functionality
between two SNPs. The authors suggested three ways of building these
networks: connecting the SNPs consecutive in the genomic sequence ("GS
network"); interconnecting all the SNPs mapped to the same gene, on top
of GS ("GM network"); and interconnecting all SNPs mapped to two genes
for which a protein-protein interaction exists ("GI network"). We
focused on the GI network, as it is the network that fits better the
scope of this chapter. However, at different stages of the chapter we
also used GS and GM. For the GM network, we used the mapping described
in Section \@ref(methods:node-score). For the GI network, we used the PPI as
described in Section \@ref(methods:gene-network). For all three networks the 
node score used is the association of the individual SNPs with the phenotype;
specifically, we used the 1 d.f. $\chi$^2^.

#### Consensus network {#methods:consensus}

The different high-weight subnetwork discovery algorithms make different
assumptions on the properties of the solutions, and employ different
strategies to find them. Hence, combining the outcome of the different
approaches might provide a more complete outlook on the specific
alterations on the GENESIS dataset. We built such consensus network by
retaining the nodes that were selected by at least two of the methods.
We combined the results of 6 methods: dmGWAS, heinz, HotNet2, LEAN,
SConES GI, and SigMod.

### Evaluation of methods {#sec:orgc968928}

#### Classification accuracy of selected biomarkers {#methods:comparison}

A desirable solution is one that is sparse, while offering a good
predictive power on unseen samples. We evaluated the predicting power of
the SNPs selected by the different methods through the performance of an
L1-penalized logistic regression trained exclusively on those SNPs to
predict the outcome (case/control). The L1 penalty helps to account for
LD to reduce the number of SNPs included in the model (size of the
active set), while improving the generalization of the classifier. The
value of the regularization parameter, which controls both the magnitude
and the sparsity of the coefficients, was set by cross-validation. To
that end, we used the different network-methods on a random subset of
80% of the samples. On this same subset we trained our classifier
exclusively on the SNPs selected by a particular method. When the method
retrieved a list of genes (all of them except SConES), all the SNPs
mapped to any of those genes were used. Then we
evaluated performance of the classifier on the remaining 20% of the
dataset. We repeated this procedure 5 times to estimate the average and
the deviation of the different performance measures. The 
performance measures we used were: size of the solution, size of the
active set, specificity, and sensitivity. The size of the active set
provides an estimate of a plausible, more sparse solution with a
comparable predictive power to the original solution.

Additionally, for each of the methods, we evaluated their stability and
their runtime. The stability of an algorithm is its sensitivity to small
changes of the input, and is measured using the Pearson's correlation
between different runs as suggested in @nogueira_measuring_2016. To
obtain a baseline, we also performed the procedure using all the SNPs.
Lastly, another desirable property is that the method retrieves a good
candidate causal subnetwork. In consequence, we compared the outcome of
each of the methods to the consensus subnetwork of all the solutions
(Section \@ref(methods:consensus)).

#### Biological relevance of the genes {#methods:bcac}

An alternative way to validate the results is comparing our results to
an external dataset. For that purpose, we recovered a list of 153 genes
associated to familial breast cancer from DisGeNET
[@pinero_disgenet:_2017]. Across this chapter we refer to these genes as
*breast cancer susceptibility genes*.

Additionally, we used the summary statistics from the Breast Cancer
Association Consortium (BCAC) [@michailidou_genome-wide_2015]. BCAC has
conducted one of the largest efforts in GWAS, involving over 120\,000
women of European ancestry. As opposed to GENESIS, samples were not
selected based on family history, and hence the study is enriched in sporadic
breast cancers. Another difference is that BCAC is a relatively
heterogeneous study on a pan-European sample, while GENESIS is a
homogeneous dataset focused on the French population. Despite these
differences, there should be shared genetic architecture. On top of
that, that overlap should become larger when the results are aggregated
at the gene level. For that purpose, we computed the gene association as
in Section \@ref(methods:node-score). The iCOGS array was used for genotyping in
BCAC [@sakoda_turning_2013], the same array as for GENESIS
[@sinilnikova_genesis:_2016]. Although imputed data are available, we
used exclusively the SNPs available on GENESIS after quality control to
make the results comparable.

### Code availability {#sec:org1d907c0}

This work required developing computational pipelines for several GWAS
analyses, such physically mapping SNPs to genes, computing gene scores,
and performing six different network analyses. For each of those
processes, a streamlined, project-agnostic pipeline with a clear
interface was created. They are compiled in the following GitHub
repository: <https://github.com/hclimente/gwas-tools>. The code that
applies these pipelines to the GENESIS project, as well as the code that
reproduces all the analyses in this chapter are available at
<https://github.com/hclimente/genewa>. Although the GENESIS dataset is
not publicly available, the published code should work on any other GWAS
dataset. All the produced gene subnetworks were deposited on NDEx
(<http://www.ndexbio.org>), under the UUID
e9b0e22a-e9b0-11e9-bb65-0ac135e8bacf.

## Results {#sec:orgc718f04}

### A conventional GWAS shows that FGFR2 is strongly associated with familial breast cancer {#results:conventional}

(ref:fig-snp-gene-manhattan-caption) Association in GENESIS. The red line represents the Bonferroni threshold. **(A)** SNP association, measured from the outcome of a 1df $\chi^2$ allelic test. Significant SNPs that are within a coding gene, or within 50 kilobases of its boundaries, are annotated. The Bonferroni threshold is $2.54 \times 10^{-7}$. **(B)** Gene association, measured by P-value of VEGAS2v2 [@mishra_vegas2:_2015] using the 10% of SNPs with the lowest P-values. The Bonferroni threshold is $1.53 \times 10^{-6}$.

```{r snp-gene-manhattan, echo=FALSE, fig.cap='(ref:fig-snp-gene-manhattan-caption)'}
knitr::include_graphics('fig/figure_1.png', dpi = NA)
```

We conducted association analyses in the GENESIS dataset at both the SNP
and the gene levels (Section \@ref(methods:node-score)). Two genomic regions 
have a P-value lower than the Bonferroni threshold in chromosomes 10 and 16 
(Figure \@ref(fig:snp-gene-manhattan) A). The former overlaps with gene
*FGFR2*; the latter with *CASC16*, and it is located near the
protein-coding gene *TOX3*. Variants in both *FGFR2* and *TOX3* were
related to breast cancer susceptibility in other cohorts negative for
*BRCA1/2* [@rinella_genetic_2013]. Only the peak in chromosome 10
replicated in the gene-level analysis, with *FGFR2* just above the
threshold of significance (Figure \@ref(fig:snp-gene-manhattan)B).

These results show the overlap in the genetic architecture of the
disease between the studied French population sample and other
populations, especially at the gene level. In addition, there are other
SNPs whose P-values, although higher than the conventional threshold of
significance, show a strong association with familial breast cancer. The
most prominent of such regions are 3p24 and 8q24, both of which have
been associated to breast cancer susceptibility in the past
[@brisbin_meta-analysis_2011; @search_newly_2009]. This motivates
exploring network methods, which trade statistical significance for
biological relevance.

### Network methods successfully identify genes associated with breast cancer {#results:separate-networks}

(ref:tab-gene-solutions-footnote) *Num genes*: number of genes selected out of those that are part of the PPIN; for SConES GI the total number of genes, including RNA genes, was added in parentheses. $\overline{\mbox{Betweenness}}$: mean betweenness of the selected genes in the PPIN. $\hat{\mbox{P}}$~gene~: median P-value of the selected genes. $\rho$~consensus~: Pearson\\'s correlation between the subnetwork and the consensus network.

(ref:betweenness) $\overline{\mbox{Betweenness}}$

(ref:p-gene) $\hat{\mbox{P}}_{gene}$

(ref:rho) $\rho_{consensus}$

```{r gene-solutions, echo=FALSE}
solution_summary <- rbind(
      c('HINT HT', 13619, 142541, 16706, 0.46,0.066),
      c('Consensus', 55, 117, 74062, 0.0051, 1),
      c('dmGWAS', 194, 450, 49115, 0.19, 0.41),
      c('heinz', 4, 3, 113633, 0.0012, 0.21),
      c('HotNet2', 440, 374, 7739, 0.048, 0.31),
      c('LEAN', 0, 0, NA, NA, NA),
      c('SConES GI', '0 (1)', 0, NA, NA, NA),
      c('SigMod', 142, 249, 92603, 0.0083, 0.73)
  )
colnames(solution_summary) <- c('Network','Num genes','Num edges', 
                                '(ref:betweenness)',
                                '(ref:p-gene)', '(ref:rho)')

footnote(kable(solution_summary, 
               caption = 'Summary statistics on the results of multiple network methods on the gene-gene interaction network. The first row contains the summary statistics on the whole network.', 
               booktabs = TRUE) %>% 
           kable_styling(latex_options = c("striped", 'scale_down')),
         general = '(ref:tab-gene-solutions-footnote)', 
         threeparttable = TRUE, escape = FALSE, fixed_small_size = TRUE)
```

(ref:tab-snp-solutions-footnote) $\overline{\mbox{Betweenness}}$: mean betweenness of the selected SNPs in the corresponding full network; $\hat{\mbox{P}}$~SNP~: median P-value of the selected SNPs.

(ref:p-snp) $\hat{\mbox{P}}$~SNP~

```{r snp-solutions, echo=FALSE}
solution_summary <- read.table(text = "
                  Network SNPs Edges Subnetworks Betweenness P
                  GS 197083 197060 NA 20300000 0.49
                  'SConES GS' 1590 1585 5 25200000 0.023
                  GM 197083 6442446 NA 3990000 0.49
                  'SConES GM' 1692 177611 5 4400000 0.055
                  GI 197083 28733720 NA 1460000 0.49
                  'SConES GI' 408 539 5 9330000 0.076
                  ", header = TRUE, stringsAsFactors = FALSE)
solution_summary$Edges <- formatC(solution_summary$Edges, digits = 2, format = 'e')
solution_summary$Betweenness <- formatC(solution_summary$Betweenness, digits = 2, format = 'e')

colnames(solution_summary) <- c('Network', 'SNPs', 'Edges', 'Subnetworks',
                                '(ref:betweenness)', '(ref:p-snp)')

kable(solution_summary, 
      caption = 'Summary statistics on the results of SConES on the three SNP-SNP interaction networks. The first row within each block contains the summary statistics on the whole network.', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = c("striped", 'scale_down')) %>%
  footnote(general = '(ref:tab-snp-solutions-footnote)', 
           threeparttable = TRUE, escape = FALSE, fixed_small_size = TRUE)
```

We applied six network methods to the GENESIS dataset (Section
\@ref(methods:methods)), obtaining six solutions (Figure 
\@ref(fig:overview-solutions)): one
for each of the five gene-based methods (Section
\@ref(methods:gene-network)), and one for SConES GI (Section
\@ref(methods:snp-network)). The solutions are very heterogeneous
(Table \@ref(tab:gene-solutions) and Table
\@ref(tab:snp-solutions)): none of the subnetworks examined by
LEAN are significant (adjusted P-value \< 0.05), while HotNet2 produced
the largest solution subnetwork with 440 genes. SConES GI failed to
recover genes in the PPIN, but it recovered one genomic region mapped to
RNA gene *RNU6-420P*. All solution subnetworks except LEAN's are, on
average, more strongly associated to breast cancer than the whole PPIN
(median P-values $\ll 0.46$), despite containing genes with higher
P-values (Figure \@ref(fig:manhattan-solutions)). This exemplifies the 
trade-off between statistical significance and biological relevance. However,
there are nuances between solutions: heinz strongly favored highly
associated genes, while dmGWAS is less conservative (median gene
P-values 0.0012 and 0.19, respectively); SConES tended to select whole
LD-blocks; and HotNet2 and SigMod were less likely to select lowly
associated genes.

(ref:fig-overview-solutions-caption) Overview of the subnetworks produced by the different network methods. **(dmGWAS, heinz, HotNet2, LEAN, and SigMod)** contain gene subnetworks; **(SConES GI)**, SNP subnetworks.

```{r overview-solutions, echo=FALSE, fig.cap='(ref:fig-overview-solutions-caption)'}
knitr::include_graphics('fig/sfigure_2.png', dpi = NA)
```

(ref:fig-manhattan-solutions-caption) Manhattan plots showing the biomolecules selected by each method. In **(Consensus, dmGWAS, heinz, HotNet2, and SigMod)** datapoints are genes; in **(SConES GS, GM, and GI)**, SNPs. LEAN was excluded, as it did not select any gene.

```{r manhattan-solutions, echo=FALSE, fig.cap='(ref:fig-manhattan-solutions-caption)', out.height='80%', fig.align='center'}
knitr::include_graphics('fig/sfigure_3.png', dpi = NA)
```

The solution subnetworks present other desirable properties. First, four
of the methods succeeded at recovering genes involved in the disease
(Figure \@ref(fig:disgenet-comparison)), as their subnetworks were
enriched in breast cancer susceptibility genes (dmGWAS, heinz, HotNet2,
and SigMod, Fisher's exact test one-sided P-value \< 0.03). We also
compared the outcome of the network methods to the association tests
conducted on the population of European ancestry from the Breast Cancer
Association Consortium (BCAC) [@michailidou_genome-wide_2015]
(Figure \@ref(fig:bcac-comparison)). Encouragingly, every solution
subnetwork is enriched in genes or SNPs that are Bonferroni-significant
in BCAC. This confirms the capability of network methods to find the
same signal as in more powered studies by leveraging on prior knowledge.
Second, the genes in four solution subnetworks display on average a
higher betweenness centrality than the rest of the genes, a difference
that is significant in three solutions (dmGWAS, and SigMod, Wilcoxon
rank-sum test P-value \< 1.4 10^-21^). This agrees with the notion that
disease genes are more central than other, non-essential genes
[@pinero_uncovering_2016]. We observe that this conclusion holds in this
disease, as known breast cancer susceptibility genes have higher
betweenness centrality than others (one-tailed Wilcoxon rank-sum test
P-value = 2.64 10^-5^, Figure \@ref(fig:consensus-stats)C). Interestingly, 
SConES' selected SNPs are also more central than the average SNP (Table
\@ref(tab:snp-solutions)), suggesting that causal SNPs are 
also more central than unrelated SNPs. However, very central nodes are also
more likely to be connecting a random pair of nodes, making then more
likely to be selected by the examined methods. Hence, further work is
needed draw conclusions.

(ref:fig-disgenet-comparison-caption) Proportion of the selected genes by each of the methods on the GENESIS data that is a known breast cancer susceptibility gene (Section \@ref(methods:bcac)). Only genes present in the protein-protein interaction network were considered. LEAN is not displayed as it did not select any gene. The presented network methods recover a higher proportion of breast cancer susceptibility genes than of other genes, despite their lack of significance in GENESIS.

```{r disgenet-comparison, echo=FALSE, fig.cap='(ref:fig-disgenet-comparison-caption)', out.height='80%', fig.align='center'}
knitr::include_graphics('fig/sfigure_5.pdf', dpi = NA)
```

(ref:fig-bcac-comparison-caption) Proportion of the Bonferroni significant biomolecules (in either the GENESIS or the BCAC datasets) selected by each of the methods on the GENESIS data. **(Consensus, dmGWAS, heinz, HotNet2, and SigMod)** involve significant genes, only among those present in the protein-protein interaction network. **(SConES GS, GM and GI)** involve significant SNPs. LEAN is not displayed as it did not select any gene. The presented network methods recover a higher proportion of significant genes than of non-significant genes in both datasets, despite their lack of significance in GENESIS.

```{r bcac-comparison, echo=FALSE, fig.cap="(ref:fig-bcac-comparison-caption)", out.height='80%', fig.align='center'}
knitr::include_graphics('fig/sfigure_4.pdf', dpi = NA)
```

(ref:fig-consensus-stats-caption) Genes on the consensus network. Breast cancer susceptibility genes are colored in pink; the rest are colored in grey. **(A)** Number of methods selecting every gene in the subnetwork. **(B)** VEGAS P-values of association of the genes, with regards to the number of methods that selected them. **(C)** Comparison of betweenness centrality of the genes in the consensus network and the other genes in the PPIN and not in the consensus network. To improve visualization, we removed outliers. **(D)** Relationship between the log~10~ of the betweenness centrality and the -log~10~ of the VEGAS P-value of the genes in the consensus network. The blue line represents a fitted generalized linear model.

```{r consensus-stats, echo=FALSE, fig.cap='(ref:fig-consensus-stats-caption)', out.height='75%', fig.align='center'}
knitr::include_graphics('fig/sfigure_8.pdf', dpi = NA)
```

As the solutions were quite different from each other it is hard to draw
joint conclusions. The 4-gene solution selected by heinz includes the
breast cancer susceptibility gene *TOX3*, in region 16q12. By dealing
with SNP networks, SConES studies the association of non-coding regions,
as well as SNPs in any gene, coding or not. In fact, SConES GI, which
adds to GM the interactions between genes, retrieves 4 subnetworks in
intergenic regions, and 1 overlapping an RNA gene (*RNU6-420P*). SigMod,
despite being related to SConES, produces a vastly different, large
solution. On top of recovering three breast cancer susceptibility genes,
a keratin-based region of its subnetwork affects the cytoskeleton
(*structural constituent of cytoskeleton*, GO enrichment's adjusted
P-value = 9.10 10^-4^), a potentially novel susceptibility mechanism for
cancer susceptibility. Interestingly, dmGWAS solution is also related to
cytoskeleton (*tubulin binding*, GO enrichment's adjusted P-value =
0.031). But, additionally, it includes a submodule of proteins related
to *unfolded protein binding* (GO enrichment's adjusted P-value =
0.045), which has been previously related to cancer susceptibility
[@calderwood_heat_2016]. Lastly, HotNet2 produced 135 subnetworks, 115
of which have less than five genes. The second largest subnetwork (13
nodes), contains the two breast cancer susceptibility genes *CASP8* and
*BLM*.

### heinz retrieves a small, highly informative set of biomarkers in a fast and stable fashion {#results-benchmark}

(ref:fig-benchmark-caption) Comparison of network-based GWAS methods on GENESIS. Each method was run 5 times of a random subset of the samples, and tested on the remaining samples (Section \@ref(methods:comparison)). **(A)** Number of SNPs selected by each method and number of SNPs on the active set used by the Lasso classifier. Points are the average over the 5 runs; lines represent the standard error of the mean. A grey diagonal line with slope 1 is added for comparison. For reference, the active set of Lasso using all the SNPs included, on average, 154\,117.4 SNPs. **(B)** Sensitivity and specificity on test set of the L1-penalized logistic regression trained on the features selected by each of the methods. In addition, the performance of the classifier trained on all SNPs is displayed. Points are the average over the 5 runs; lines represent the standard error of the mean. **(C)** Pairwise Pearson\\'s correlations of the solutions used by different methods. A Pearson's correlation of 1 means the two solutions are the same. A Pearson's correlation of 0 means that there is no SNP in common between the two solutions. **(D)** Runtime of the evaluated methods, by type of network used (gene or SNP). For gene network-based methods, inverted triangles represent the runtime of the algorithm itself, and circles the total time, which includes the algorithm themselves and the additional 119\,980 seconds (1 day and 9.33 hours) which took VEGAS2v2 on average to compute the gene scores from SNP summary statistics.

```{r benchmark, echo=FALSE, fig.cap='(ref:fig-benchmark-caption)', out.width='80%', fig.align='center'}
knitr::include_graphics('fig/figure_4.pdf', dpi = NA)
```

As the employed methods produced such different results, we compared
their solutions in a 5-fold subsampling setting (Section
\@ref(methods:comparison)). Specifically, we measured 
four properties (Figure \@ref(fig:benchmark)): 
size of the solution subnetwork,
sensitivity and specificity of an L1-penalized logistic regression on
the selected SNPs, stability, and computational runtime.

Both solution size and active set of SNPs selected by Lasso varies
greatly between the different methods (Figure \@ref(fig:benchmark)A). 
heinz has the smallest solutions, with an
average of 182 selected SNPs are selected by
Lasso. The largest solutions come from SConES GI (6256.6 SNPs), and
dmGWAS (4255.0 SNPs). Interestingly, heinz has the highest proportion of
the selected SNPs that go into the active set (99.9%), although it is
high for all the methods (\> 86%). This suggests methods are selecting
informative SNPs on average.

To determine whether the selected SNPs could be used for patient
classification we computed the sensitivity and the specificity of the
classifier on the testing data (Figure \@ref(fig:benchmark)B). 
All classifiers' sensitivities were in the
0.42 -- 0.51 range; the specificities, between 0.54 and 0.62. On
average, SigMod had the highest sensitivity (0.51); dmGWAS, the highest
specificity (0.52). Both heinz and SigMod had on average better
sensitivity than the classifier trained on all the SNPs, but none had
superior specificity. However, the differences are negligible, well
within the 95% confidence interval.

Another desirable quality of an algorithm is stability (Section
\@ref(methods:comparison)). Both heinz and LEAN displayed a high
stability in our benchmark, consistently selecting the same genes and no
genes over the 5 subsamples, respectively (Figure \@ref(fig:benchmark)C). 
Conversely, the other methods displayed similarly low stabilities.

In terms of computational runtime, the fastest method was heinz (Figure
\@ref(fig:benchmark)D), which leverages on its ability to find
efficiently the solution in a few seconds. The slowest method was dmGWAS
(1 day and 17 hours on average) followed by SConES GI (1 day and 4.32
hours on average). However, including the time required to compute the gene
scores slows down considerably gene-based methods; on this
benchmark, that step took on average 1 day and 9.33 hours. Considering
that, it took 3 days and 2.4 hours on average for dmGWAS' to produce
results.

### No solution is perfect {#genesis-issues}

(ref:fig-issues-caption) Drawbacks confronted when using network guided methods. **(A)** dmGWAS solution subnetwork. Genes with a P-value \< 0.1 are highlighted in red. **(B)** Centrality degree and -log~10~ of the VEGAS P-value for the nodes in SigMod solution subnetwork. **(C)** Genomic regions where either SConES GS, GM or GI select SNPs.

```{r issues, echo=FALSE, fig.cap='(ref:fig-issues-caption)'}
knitr::include_graphics('fig/figure_2.png', dpi = NA)
```

In practice, and despite their similarities and their involvement in
cancer mechanisms, the solutions are remarkably different 
(Figure \@ref(fig:pearson-methods)A). That is due to the particularities
of the methods which directly or indirectly provide information about
the dataset. For instance, the fact that LEAN did not provide any
biomarkers implies that there is no gene such that both itself and its
environment are on average strongly associated with the disease.

(ref:fig-pearson-methods-caption) Pearson's correlation between the different solution subnetworks. **(A)** Correlation between selected SNPs. **(B)** Correlation between selected genes. In general, the solutions display a very low overlap.

```{r pearson-methods, echo=FALSE, fig.cap='(ref:fig-pearson-methods-caption)'}
knitr::include_graphics('fig/sfigure_6.pdf', dpi = NA)
```

In this dataset, heinz's solution is very conservative, providing a
small solution with the lowest median P-value for the subnetwork (Table
\@ref(tab:gene-solutions)). Due to this parsimonious and highly
associated solution, it was the best method to select a set of good
biomarkers for classification. (Figure \@ref(fig:benchmark)B). 
Its conservativeness stems from its
preprocessing step, which models the gene P-values as a mixture model of
a beta and a uniform distribution, controlled by an FDR parameter. Due
to the limited signal at the gene level in this dataset (Figure
\@ref(fig:snp-gene-manhattan)B), only 36 of them are retain a
positive score after applying the BUM model (Section
\@ref(methods:methods)). Hence, heinz's solution subnetwork
consists only of 4 genes, which does not provide much insight of the
biology of cancer. Importantly, it ignores genes that are strongly
associated to cancer in this dataset like *FGFR2*.

On the other end of the spectrum, we have large solutions provided by
dmGWAS, HotNet2, and SigMod. dmGWAS' subnetwork is the least associated
subnetwork on average. This is due to the greedy framework it uses,
which considers all nodes at distance 2 of the examined, and accepts
weakly associated genes if they are linked to another, strongly
associated one. This is exacerbated when the results of successive
greedy searches are aggregated, leading to a large, tightly connected
cluster of unassociated genes (Figure \@ref(fig:issues)A). 
SigMod displays the same tendency, as the
most central genes are the least associated to the disease (Figure
\@ref(fig:issues)B). This relatively low signal-to-noise ratio
combined with the large solution requires additional analyses to draw
conclusions, such as enrichment analyses. In the same line, HotNet2's
subnetwork is even harder to interpret, being composed of 440 genes
divided into 135 subnetworks. Lastly, SigMod misses some of the most
strongly associated, breast cancer susceptibility genes in the dataset,
like *FGFR2* and *TOX3*.

By virtue of using a SNP subnetwork, SConES analyzes each SNP in their
context. It therefore selects SNPs in genes none of whose interactors
are associated to the disease, as well as SNPs in non-coding regions or
in non-interacting genes. In fact, due to linkage disequilibrium, such
genes are favored by SConES, as selecting SNPs in an LD block which
overlaps with a gene favors selecting the rest of the gene. This might
explain why the GS and GM networks, heavily affected by linkage
disequilibrium, produce similar results (Figure
\@ref(fig:pearson-methods)B). On the other hand, SConES penalizes
selecting SNPs and not their neighbors. This makes it conservative
regarding SNPs with many interactions, for instance those mapped to hubs
in the PPIN. For this reason, SConES GI did not select any protein
coding gene, despite selecting similar regions as SConES GS (Figure
\@ref(fig:issues)C). In fact SConES GS and SConES GM select
regions related to breast cancer, like 16q12 (*TOX3*, Section
\@ref(results:conventional)), 3p24 (*SLC4A7*/*NEK10*,
@search_newly_2009), 5p12 (*FGF10*, *MRPS30*, @quigley_5p12_2014), and
10q26 (*FGFR2*, Section \@ref(results:conventional)). On top of that only SConES
GS selects region 8q24 (*POU5F1B*, @breyer_expressed_2014). We hypothesize
that the lack of results on the PPIN network of SConES GI and LEAN due
to the same cause: the absence of joint association of a module.
Although in the case of SConES other hyperparameters could lead to a
more informative solution (e.g. lower $\lambda$, Section\@ref(methods:methods)),
it is unclear what is the best strategy
to find them. In addition, due to the iCOGS SNP array design, the genome
of GENESIS participants has not been unbiasedly surveyed: some regions
are fine-mapped --- which might distort gene structure in GM and GI
networks --- while others are under studied --- hurting the accuracy
with which the GS network captures the genome structure.

### Aggregating solutions provides insights into the biology of cancer {#results:consensus}

To leverage on the strengths of each of the methods and compensate their
respective weaknesses, we built a consensus subnetwork that captures the
mechanisms most shared among the solution subnetworks (Section
\@ref(methods:consensus)). The consensus subnetwork (Figure
\@ref(fig:consensus-names)) contains 93 genes and is enriched in breast
cancer susceptibility genes (Fisher's exact test P-value = 7.8 10^-5^).
Due to the limited overlap between methods, only 20 genes were common to
more than two of them (Figure \@ref(fig:consensus-stats)A). Encouragingly, 
the more methods selected a gene, the higher its association was (Figure
\@ref(fig:consensus-stats)B). Globally, a GO enrichment shows the
involvement of two cellular processes: unfolded protein binding, and
structural constituent of cytoskeleton (adjusted P-values of 0.001,
0.001, respectively), which were already observed in different solutions
(Section \@ref(results:separate-networks)). Remarkably, many of the
selected genes are related to mitochondrial translation. For instance,
MRPS30 (VEGAS P-value = 0.001), encodes a mitochondrial ribosomal
protein and was also linked to breast cancer susceptibility
[@quigley_5p12_2014]. Albeit disconnected from MRPS30, the consensus
network includes a 2-node subnetwork composed of two mitochondrial
ribosomal protein (MRPS31 - VEGAS P-value = 7.67 10^-3^ - and MRPS18B -
VEGAS P-value = 7.92 10^-3^), which suggests an involvement of
mitochondrial ribosomes in carcinogenesis.

(ref:fig-consensus-names-caption) Consensus subnetwork on GENESIS (Section \@ref(methods:consensus)). **(A)** Each node is represented by a pie chart, which shows the methods that selected it. The labeled genes have a VEGAS2v2 P-value \< 0.001 and/or are known breast cancer susceptibility genes (colored in pink). **(B)** The name of every gene is indicated.

```{r consensus-names, echo=FALSE, fig.cap='(ref:fig-consensus-names-caption)', out.height='80%', fig.align='center'}
knitr::include_graphics('fig/sfigure_7.pdf', dpi = NA)
```

We also examined the topological properties of the nodes. The genes in
the consensus network have higher betweenness centrality than the rest
of the genes (Wilcoxon rank-sum test P-value = 4.29 10^-18^).
Interestingly, within genes in the consensus network, cancer genes are
as central as non-cancer genes (Wilcoxon rank-sum test P-value = 0.57).
Centrality, however, is weakly anti-correlated with association to the
disease (Pearson correlation coefficient = -0.26, Figure
\@ref(fig:consensus-stats)D), which suggests that some highly
central genes were selected because they were on the shortest path
between two highly associated genes. In view of this, we hypothesize
that highly central genes might contribute to the heritability through
consistent alterations of their neighborhood, consistent with the
omnigenic model of disease [@boyle_expanded_2017]. For instance, the
most central node in the consensus network is *COPS5* 
(Figure \@ref(fig:consensus-names)), a gene related to multiple hallmarks
of cancer and which is overexpressed in multiple tumors, including
breast and ovarian cancer [@liu_jab1_cops5_2018]. Despite its lack of
association in GENESIS (VEGAS P-value = 0.22), its neighbors in the
consensus subnetwork have consistently low P-values (median VEGAS
P-value = 0.006).

The consensus subnetwork is not completely connected: out of the 93
genes, the largest connected subnetwork includes only 49. A GO
enrichment analysis showed that this component is related to three major
cellular processes: unfolded protein binding, structural constituent of
cytoskeleton, and poly(U) RNA binding (adjusted P-values of 0.01, 0.04,
and 0.04, respectively). We found support in the literature of the
involvement of each of these functions in the development of cancer, as
discussed next. The consensus network also contains a protein directly
involved in caspase-mediated apoptosis, *CASP8* (VEGAS P-value = 1.95
10^-4^). This is related to the enriched activity, *unfolded protein
binding*, which inhibits caspase-dependent apoptosis, raising the
chances of developing cancer [@calderwood_heat_2016]. It involves three
Hsp70 chaperones of the consensus subnetwork: HSPA1A, HSPA1B, and
HSPA1L. They genes encoding these proteins are all near each other at
6p21. In fact, out of the 22 SNPs that map to any of these three genes,
9 map to all of them, and 4 to two, making hard to disentangle their
association. *HSPA1A* was the most strongly associated one (VEGAS P-value
= 8.37 10^-4^). Remarkably, 14 of the 93 genes are in subnetworks of
size 1 (isolated) or 2, as they do not have a consistently altered
neighborhood. One of them is the well-known breast cancer susceptibility
gene *FGFR2* (Section \@ref(results:conventional)
). Another one is the also well-known
*SLC4A7* gene (VEGAS P-value = 2.70 10^5^), which encodes a sodium
bicarbonate cotransporter. The genomic region containing both *SLC4A7*
and nearby gene *NEK10* (VEGAS P-value = 1.56 10^-5^) have been
consistently associated with breast cancer susceptibility
[@search_newly_2009]. *NEK10* is a gene that might be involved in
cell-cycle control, but it is absent from the PPIN and hence it could
not be studied by gene methods. Despite that, the fact that both dmGWAS,
HotNet2 and SigMod link *SLC4A7* in their different subnetwork supports
the notion that this gene is the responsible for breast cancer
susceptibility.

## Discussion {#genewa-discussion}

In this chapter we evaluate the viability of systems biology approaches
to GWAS, and examine a GWAS dataset on familial breast cancer focused on
*BRCA1/2* negative French women. Systems biology addresses two of the
largest GWAS issues: interpretability and an overly conservative
statistical framework that hinders discovery. This is achieved by
considering the biological context of each of the genes and SNPs. Based
on divergent considerations of what the desired set of biomarkers is,
several methods for network-guided biomarker discovery have been
proposed. We reviewed the performance of six of them on GWAS. Despite
their differences, most of them produced a relevant subset of
biomarkers, recovering known familial breast cancer genes. We also
discuss the limitations of such analyses, related to the lack of known
interactions around some genes. A crucial step for the gene based
methods is the computation of the gene score. In this chapter we used
VEGAS2v2 [@mishra_vegas2:_2015] due to the flexibility it offers to use
user-specified gene annotations. However, it presents known problems
(selection of an appropriate percentage of top SNPs, long runtimes and
P-value precision limited to the number of permutations
[@nakka_gene_2016]), other algorithms might have more statistical power.

The network methods we studied differ in what the optimal solution
subnetwork looks like. On the one hand, SConES and heinz prefer small
highly associated solutions. On the other hand, SigMod and dmGWAS
gravitate towards larger, less associated solutions which provide a wide
overview of the biological context. While the former provide a reduced
set of biomarkers, the latter deepen our understanding of the disease
and provide biological hypotheses. Yet, they are not exempt of limitations.
dmGWAS and SigMod's solution's size require further analyses, which risk
oversimplifying their richness. Also, incautious practitioners might be
misled by some genes, which are very central in the solution
subnetworks, while being weakly associated. Nonetheless, they are pushed
into the solution by their privileged topological properties. On the
other end, conservative solutions, like SConES GI and heinz might not
shed much light on the etiology of the disease.

To overcome the problems posed by the individual methods while
exploiting their strengths, we propose combining them into a consensus
subnetwork. We use a straightforward aggregation to generate it,
including any node that was recovered by at least two methods. The
resulting network is a synthesis of the altered mechanism: it is smaller
than the largest solutions (SigMod and dmGWAS), which makes it more
manageable, and includes the majority of the strongly associated smaller
solutions (SConES and heinz). The consensus subnetwork captures
mechanisms and genes known to be related to cancer, recovering known
breast cancer susceptibility genes as well as genome regions associated
to breast cancer susceptibility. However, thanks to its smaller size and
its network structure, it provides compelling hypotheses of
non-canonical mechanisms involved in carcinogenesis, like mitochondrial
translation and chaperone activity.

The strength of network-based analyses comes from leveraging prior
knowledge to boost discovery. In consequence, they show their
shortcomings in front of understudied genes, especially those not in the
network. Out of the 32\,767 genes that we can map the genotyped SNPs to,
60.7% (19\,887) are not in the protein-protein interaction network. The
majority of those (14\,660) are non-coding genes, mainly lncRNA, miRNA,
and snRNA (Figure \@ref(fig:biotypes-excluded)). The importance of these genes,
like *CASC16*, is highlighted in Section \@ref(results:conventional). 
Among the excluded protein-coding
genes we find genes like *NEK10* (P-value 1.6 10^-5^) or *POU5F1B*, both
linked to breast cancer susceptibility [@search_newly_2009]. However, on
average protein-coding genes absent from the PPIN are less associated
with this phenotype (Wilcoxon rank-sum P-value = 2.79 10^-8^, median
P-values of 0.43 and 0.47). As we are using interactions from
high-throughput experiments, such difference cannot be due to well-known
genes having more known interactions. As disease genes tend to be more
central [@pinero_uncovering_2016], we hypothesize that it is due to
interactions between central genes being more likely. It is worth noting
that network approaches that do not use PPIs, like SConES GS and GM, did
recover SNPs in *NEK10* and *CASC16*. Lastly, all the methods rely
heavily on how SNPs are mapped to genes. In Section
\@ref(results:conventional) we highlight ambiguities that appear
when genes overlap or are in linkage disequilibrium.

```{r biotypes-excluded, echo=FALSE, fig.cap='Biotypes of genes from the annotation that are not present in the HINT protein-protein interaction network.'}
knitr::include_graphics('fig/sfigure_9.pdf', dpi = NA)
```

As not all databases compile the same interactions, the choice of the
PPIN determines the final output. In this work we used exclusively
interactions from HINT from high-throughput experiments. This responds
to concerns of some authors about biases introduced by adding
interactions coming from targeted studies in the literature
[@cai_broker_2010; @das_hint_2012] where a "rich getting richer"
phenomenon is observed: popular genes have a higher proportion of their
interactions described. On the other hand, @huang_systematic_2018 found that the best
predictor of the performance of a network for disease gene discovery is
the size of the network. This also supports
using the largest amount of interactions. To clarify their impact on
this study, we compared the impact of using only physical interactions
from high-throughput experiment versus interactions from both
high-throughput and the literature (Section \@ref(methods:gene-network)). We 
conclude that for most of the methods a larger network did not greatly impact 
the size or the stability of the solution, the classification accuracy, or the 
runtime (Figure \@ref(fig:lc-ht-comparison)).

(ref:fig-lc-ht-comparison-caption) Comparison of benchmark on high-throughput interactions to benchmark on both high-throughput and literature curated interactions. Grey lines represent no change between the benchmarks (1 for ratios, 0 for differences). **(A)** Ratios of the selected features between both benchmarks and of the active set. **(B)** Shifts in sensitivity and specificity. **(C)** Shift in Pearson's correlation between benchmarks. **(D)** Ratio between the runtimes of the benchmarks. For gene network-based methods, inverted triangles represent the ratio of runtimes of the algorithms themselves, and circles the total time, which includes the algorithm themselves and the additional 119\,980 seconds (1 day and 9.33 hours) which took VEGAS2v2 on average to compute the gene scores from SNP summary statistics. In general, adding additional interactions slightly improves the stability of the solution, but increases the solution size, has mixed effects on the sensitivity and specificity, and impacts negatively the required runtime of the algorithms.

```{r lc-ht-comparison, echo=FALSE, fig.cap='(ref:fig-lc-ht-comparison-caption)'}
knitr::include_graphics('fig/sfigure_10.pdf', dpi = NA)
```

In order to produce the consensus network, we had to face the different
interfaces, preprocessing steps, and unexpected behaviors of the various
methods. To facilitate that other authors apply them to new datasets and
aggregate their solutions, we built six nextflow pipelines
[@di_tommaso_nextflow_2017] with a consistent interface and, whenever
possible, parallelized computation. They are available on GitHub:
<https://github.com/hclimente/gwas-tools>.

<!--chapter:end:02-genewa.Rmd-->

# The *martini* R package {#martini}

Parts of this work were presented in ISMB/ECCB 2017 in the poster *Héctor Climente-González and Chloé-Agathe Azencott*, R package for network-guided Genome-Wide Association Studies.

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
cat("\\begin{chapabstract}{Abstract:}
In this chapter I introduce martini and \\verb|scones.nf|. martini is an implementation of SConES in R with three improvements over previous ones. First, it includes measures of association for binary traits ($\\chi^2$ and logistic regression). Second, it accepts user-specified covariates, included as additional terms in the logistic regression. Third, it includes four ways of scoring the cross-validation folds: consistency (as other implementations), and three penalized log-likelihood measures (BIC, AIC, and AICc). martini is available in Bioconductor 3.7. \\verb|scones.nf| is a computational package that further facilitates the usage of martini. It does so by accepting common file formats for GWAS and networks, and by performing a two-staged cross-validation that achieves better hyperparameters. \\verb|scones.nf| is available on \\url{https://github.com/hclimente/gwas-tools}.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}
Dans ce chapitre, je pr\\'esente martini et \\verb|scones.nf|. martini est 
une impl\\'ementation en R de SConES, qui pr\\'esente trois am\\'eliorations sur 
les impl\\'ementations pr\\'ec\\'edentes. Tout d'abord, martini inclut des 
mesures d'association pour des traits binaires ($\\chi^2$ et r\\'egression 
logistique). Deuxi\\`emement, martini permet de sp\\'ecifier des covariables, 
incorpor\\'ees comme des termes suppl\\'ementaires dans la r\\'egression 
logistique. Troisi\\`emement, martini inclut quatre façons de mesurer la 
performance en validation crois\\'ee : la consistance (comme pr\\'ec\\'edemment), 
et trois mesures de log-vraisemblance p\\'enalis\\'ee (BIC, AIC and AICc). 
martini est disponible dans Bioconductor 3.7. \\verb|scones.nf| est un paquet 
logiciel qui facilite encore plus l'utilisation de martini, en acceptant 
des formats de fichiers usuels pour les donn\\'ees GWAS ainsi que les 
r\\'eseaux, et gr\\^ace \\`a une validation crois\\'ee en deux \\'etapes qui permet 
d'obtenir de meilleurs hyperparam\\`etres. \\verb|scones.nf| est disponible 
sur \\url{https://github.com/hclimente/gwas-tools}.
\\end{chapabstract}")
```

## Introduction

In Chapter \@ref(genesis) I presented six high-score subnetwork search methods, and their application to GWAS. In this chapter I focus on my work on one of them, SConES [@azencott_efficient_2013], which was presented with the other methods in Section \@ref(methods:methods). As a reminder, SConES finds a small set of highly interconnected SNPs associated to the disease by solving the following problem:

\begin{equation}
\underset{S \subseteq G}{\arg \max } \underbrace{\sum_{v \in V_S} s_v}_{\text { association }}-\underbrace{\lambda \sum_{v \in V_S} \sum_{u \not\in V_S} W_{vu} }_{\text { connectivity }}-\underbrace{\eta \lvert V_S \rvert }_{\text { sparsity }},
\end{equation}

where $\lambda$ and $\eta$ are hyperparameters, $s_v$ is the association score of node $v$, and $W$ is the Laplacian matrix of the network. The mathematical notation is described in Section \@ref(methods:notation). SConES works on SNP networks in which the SNPs are linked to each other if there is evidence of shared function, for instance, if they are mapped to the same gene (Section \@ref(intro-interpretability)).

With the goal of applying SConES to the GENESIS dataset (Section \@ref(genesis-dataset)), I developed a user-friendly version that solved some of SConES' initial shortcomings (detailed in Section \@ref(martini-improvements)). The result was an R package, *martini* [@martini], which was published in Bioconductor 3.7 [@bioconductor], a peer-reviewed R repository. *martini* was the R version ("user friendly") of *gin* (GWAS Incorporating Networks), which is my \verb!C++! re-implementation of SConES based, in turn, on EasyGWAS [@grimm_easygwas_2017]. Hence, *martini* combines the accessibility of R and of the Bioconductor environment with the computational efficiency of \verb!C++!.

## Improvements over SConES {#martini-improvements}

### Covariates and additional measures of association

SConES scored the relevance of each SNP to the phenotype using the linear SKAT test of association [@wu_rare-variant_2011;@ionita-laza_sequence_2013]. Yet, no version of SConES implemented a way to measure association with a binary phenotype, like logistic regression-based SKAT. Hence, we decided to implement two additional measures of association in *martini*, $\chi^2$ and logistic regression, to apply SConES on case-control studies. Additionally, the latter allowed *martini* to handle covariates and hence correct for population structure (Section \@ref(pop-structure)).

### Hyperparameter optimization

By examining the results of other implementations of SConES on simulated examples, we observed that it produced suboptimal solutions. Those (unshown) results were either trivial ones (all SNPs or no SNP retrieved), or cases where SNPs strongly associated with a phenotype were not selected. To address those issues, I experimented with altering how SConES chooses its hyperparameters ($\lambda$ and $\eta$), in the hope that other values would produce better solutions.

SConES chooses the best $\lambda$ and $\eta$ based on consistency across 10 cross-validation folds. In other words, for each specific combination of $\lambda$ and $\eta$ to evaluate, the data is divided into 10 equal parts, or folds, each containing 90\% of the samples. In each fold, a selection vector $v$ is produced with the respective hyperparameters. The length of $v$ is equal to the number of SNPs $N$, and each element is set to 0, if the corresponding SNP is not selected, and to 1, if it is. The consistency $C$ between the selection vectors $v_i$ and $v_j$ of two folds for the same $\lambda$ and $\eta$ can be calculated as

$$ C = N\|v_i \cdot v_j \|_0 - \|v_i\| _0\|v_j\|_0 .$$

Then a normalized consistency $C'$ is computed by dividing $C$ by the maximum possible consistency $C^*$:

$$C^* = N \min(\|v_i\| _0, \|v_j\|_0) - \|v_i\| _0\|v_j\|_0. $$

Each combination of $\lambda$ and $\eta$ is scored using the mean of all such pairwise normalized consistencies. The hyperparameters that produce the most consistent result are used on the whole dataset to produce the algorithm's output. Consistency was the metric of choice in the original SConES to favor stability in the solution [@azencott_efficient_2013]. 

We explored alternatives to consistency as selection criterion. Measures based on the accuracy of a linear classifier trained on the selected SNPs had been tested before, but exhibited proneness to overfitting. Hence we turned to penalized log-likelihood measures [@Dziak2005], developed in the field of information theory. These scoring functions had the potential to overcome the overfitting of a linear classifier by adding a regularization term to improve generalization. They take the form

$$L(X,y,\hat{\theta})-c(\hat{\theta}),$$

where $L(X,y,\hat{\theta})$ is the log-likelihood of the model, which depends on the feature matrix $X$, the outcome vector $y$, and the parameters $\hat{\theta}$; and $c(\hat{\theta})$ is a measurement of the model's complexity. Particularly, we explored three measures [@Dziak2005]: Akaike information criterion (AIC), Bayesian information criterion (BIC), and corrected Akaike information criterion (AICc). All three take the form

$$L(X,y,\hat{\theta})-\lambda p_{in},$$

where $\lambda$ is a factor that controls the penalty for complexity; and $p_{in}$ is the number of features included in the model. Specifically they are defined as:

$$ AIC=2L(X,y,\hat{\theta})-2p_{in} ,$$

$$ BIC = -2L(X,y,\hat{\theta})-\ln(n)(p_{in}+ 2) ,$$

and

$$ AIC_c=AIC+\frac{2p_{in}(p_{in}+1)}{n-p_{in}-1}=-2L(X,y,\hat{\theta})+2\left(\dfrac{n}{n-p_{in}-1}\right)p_{in} .$$

AICc is a modification of AIC that penalizes complex models (many features included) in high dimensional settings, that is, where the number of features is much larger than the number of samples, as in GWAS.

*martini* implements these three measures to score each combination of hyperparameters. As with consistency, every considered $\lambda$ and $\eta$ is tested in a 10-fold split of the data. Then, for each fold, a linear model is built, which aims to predict the phenotype using the SNPs selected in that fold. These scores relate the likelihood of these linear models to its complexity. Each combination of hyperparameters is scored by averaging the 10 folds, and the one that produces the lowest score on average is chosen. In some simulation studies (not shown), we observed that they solved the problems of the consistency score outlined above.

### Network-based simulations

We conjecture that SConES will detect biomarker with increased sensibility with respect to non-network frameworks. In other words, it should be able to detect causal SNPs for less heritable phenotypes. In order to test this hypothesis, *martini* facilitates the simulation of phenotypes and interconnected causal networks on real GWAS datasets. This simulation tool is broken down into two functions.

The first of them is `simulate_causal_snps()`, which takes a SNP network in which each SNP is annotated with the genes it maps to (as can be obtained by `get_GI_network()`, see Section \@ref(martini-ui)). It takes two additional parameters: the number of genes involved in the disease (`n`), and the proportion of the SNPs mapped to a causal gene that are causal themselves (`p`). Then, it randomly scans the network until it finds a maximum of two connected SNP subnetworks that are mapped to `n` different genes. A fraction `p` of such subnetwork are selected as causal.

The second function is `simulate_phenotype()`, which re-implements the `--simu-cc` phenotype simulation function of the GCTA suite [@Yang2011]. `simulate_phenotype()` requires an existing GWAS experiment (`gwas`), and a set of causal SNPs (`snps`). When the causal SNPs are the output of `simulate_causal_snps()`, it inherits the additional constraint that the causal SNPs are connected in an underlying network. It also accepts other optional parameters which I describe below. Then, it simulates the quantitative phenotype $y_j$ for patient $j$ using the following additive model:

$$ y_j = \sum\nolimits_i w_{ij} u_i + e_j ,$$

where the weight $w_{ij}$ is the inclination of the genotype $i$ of patient $j$ over the phenotype; the allelic effect of the $i$-th causal variant $u_i$ in arbitrary units; and the residual effect $e_j$ is the the proportion of the trait not attributable to the genotype. The vector of effect sizes $u$ can be specified by the user via the `effectSize` parameter. If it is not, by default it is sampled from a standard Normal distribution.

The weight $w_{ij}$ is calculated as

$$ w_{ij} = \frac{x_{ij} - 2p_i}{\sqrt{2p_i(1 - p_i)}} $$

where $x_{ij}$ is the number of reference alleles for the $i$-th causal variant of the $j$-th individual; and $p_i$ is the frequency of the $i$-th causal variant. $w_{ij}$ follows a sigmoid like behavior for different $p$ (Figure \@ref(fig:p-vs-w)): the rarer an allele is, the stronger its impact on the phenotype.

(ref:fig-p-vs-w-caption) Allelic effect $i$ as function of causal allele frequency $p$ for different counts of causal allele in a patient ($x = 0, 1, 2$).

```{r p-vs-w, echo=FALSE, fig.cap='(ref:fig-p-vs-w-caption)'}
knitr::include_graphics('fig/2_p_vs_w.png', dpi = NA)
```

An interesting bit of this simulation is the residual effect $e_j$. It depends directly on the heritability of the trait, which must be given by the user via the `h2` parameter. Then $e_j$ is generated from a Normal distribution with mean of 0 and variance 

$$ \frac{1}{h^2 - 1} \text{var} \left(\sum\nolimits_i w_{ij} u_i\right),$$

where $w$ and $u$ are the weight and effect sizes specified above. When all variance is due to genetics ($h^2 = 1$), $e_j = 0$ for all the patients $j$.

Lastly, a user can request a binary phenotype setting the `qualitative` option to `TRUE`. In this case, the user must also specify three additional parameters: the number of cases (`ncases`), the number of controls (`ncontrols`), and the prevalence of the trait (`prevalence`). With these parameters, `simulate_phenotype()` takes the `ncontrols` samples with the lowest $y$ as controls, and the `ncases` samples with the highest $y$ as cases. However, `ncases` needs to be lower or equal than `prevalence` $\times |y|$, where $|y|$ is the total number of samples in the GWAS experiment. This ensures that only the most extreme samples (as defined by the prevalence and the qualitative simulation) are cases.

### Interface, documentation and quality assurance {#martini-ui}

Last, but not least, *martini* includes the two main groups of functions required to run SConES. The first group involves the creation of the SNP networks, which were described in detail in Section \@ref(methods:snp-network). They are the `get_GS_network()`, to obtain a network that relates the SNPs based on genomic structure; `get_GM_network()` for a network that, on top of the previous one, relates SNPs mapped to the same gene; and `get_GI_network()` which, on top of the latter, relates SNPs mapped to genes that interact in a user provided list. The second important function is `scones()` which takes a GWAS dataset and a SNP network and runs SConES.

All functions exported by *martini* have a `man` page, and hence information of the functions arguments, behavior and return value can be obtained via `help(fun)`. Accompanying examples and toy datasets are provided. Additionally, I wrote two vignettes to explain its basic behavior: one to run SConES ([*Running SConES*](https://www.bioconductor.org/packages/release/bioc/vignettes/martini/inst/doc/scones_usage.html)), and another to simulate network-based phenotypes ([*Simulating SConES-based phenotypes*](https://www.bioconductor.org/packages/release/bioc/vignettes/martini/inst/doc/simulate_phenotype.html)).

*martini* was thoroughly subjected to unit tests via the *testthat* package [@wickham_testthat:_2011]. At the moment of writing this text, *martini* had a code coverage of 96%.

## The `scones.nf` pipeline

In addition to the changes implemented in *martini*, I developed a ready-to-use computational pipeline that simplifies its usage: `scones.nf`. This pipeline just requires genotype data in PLINK binary file format and, when needed for the creation of a GM or GI network, a gene annotation file and a protein-protein interaction file. `scones.nf` is available on GitHub (https://github.com/hclimente/gwas-tools). In terms of function, the difference with vanilla *martini* is that it performs an exhaustive grid-search to optimize both $\lambda$ and $\eta$, as opposed to *martini*'s single grid search step. In the latter, both parameters explore the same range of values, which is calculated from the association scores $c$ (e.g. SKAT score). Specifically, it creates a linearly spaced n-component vector (n = 10 by default) between $\lfloor \log_{10} \min(c) \rfloor$ and $\lceil \log_{10} \max(c) \rceil$, then explores its powers of 10. On the other hand `scones.nf` makes the grid search finer, because it explores the grid in an iterative way. After the first exploration, identical to the one just described, the best $\lambda$ and $\eta$ according to some selection criterion are picked. Then it creates a new hyperparameter space, ranging from $\log_{10}(\text{best } \lambda) - \Delta$ to $\log_{10}(\text{best } \lambda) + \Delta$, where $\Delta = 0.2(\log_{10}\max{(\text{explored }\lambda)}- \log_{10} \min{(\text{explored }\lambda)})$. The final $\lambda$ and $\eta$ are chosen using the same procedure on this new grid. Prior to these improvements, `scones.nf` was not able, in some instances, to recover the best solution, returning a trivial solution instead.

## Conclusions

In this chapter I introduced *martini* and the `scones.nf` pipeline. Jointly, they make SConES easily applicable to any GWAS dataset. Specially, they provide a wider range of options to the user in terms of how to measure the association between the genotypes and the phenotype, and how to select SConES hyper-parameters. Thanks to these improvements, we were able to obtain the SConES results presented in Chapter \@ref(genesis). Additionally, we provide a network-based phenotype simulation framework. *martini* [@martini] is available in Bioconductor (https://www.bioconductor.org/packages/martini); `scones.nf` is available on GitHub (https://github.com/hclimente/gwas-tools).

However, *martini* and SConES still present shortcomings with regards to hyper-parameter selection (Sections \@ref(results-benchmark) and \@ref(genesis-issues)). As we note, SConES solutions were unstable despite using *consistency* for model selection. This requires further examination. Additionally, when applied to GENESIS the selected solutions did not use the protein-protein interaction network, but other methods did. Although this is, to some extent, expected in that dataset, it does not inform much about the biology of the disease. In other words, in this case, SConES should be more tolerant to including unassociated SNPs in order to interconnect subnetworks of strongly associated SNPs. Hence, different parameters (lower values of $\lambda$) might relax the connectivity constraints enough to capture biological mechanisms more broadly. In this regard, it would be promising to use topological measures as scores for hyperparameter selection, for instance favoring settings that lead to densely interconnected networks. In this vein, it also is worth further exploring SigMod [@liu_sigmod:_2017] a modification of SConES described in Section \@ref(methods:methods). While SConES penalizes selecting SNPs with many outbound edges and does not account for inbound edges, SigMod favours SNPs with many inbound edges, while disregarding outbound edges.

Lastly, I would like to explore SNP networks beyond the ones proposed in the original manuscript (Section \@ref(methods:snp-network)). As I described in Section \@ref(intro-interpretability), and will deepen in Chapter \@ref(epi-snp2gene), there are multiple ways of relating SNPs to a function, and hence to build SNP networks. One compelling idea is to build a SNP network that reflects the 3D structure of the genome. Another opportunity for development involves encoding the LD relationship between the SNPs in the network, by tuning the edge weights. This would allow to either favor the selection of LD blocks (large LD implies large edge weight), or favor the selection of independent, associated SNPs (large LD implies small edge weight). These networks are very complex, involving hundreds of thousands of nodes and tens of millions of edges. In this regard, the ability of SConES to efficiently handle such networks enables this research and motivates working on solving the aforementioned issues. 

<!--chapter:end:03-martini.Rmd-->

# Boosting interpretability and statistical power in epistasis detection by using prior biological knowledge {#epi-snp2gene}

The contents of this chapter are based on *Diane Duroux, Héctor Climente-González, Lars Wienbrandt, David Ellinghaus, Chloé-Agathe Azencott and Kristel Van Steen*, Boosting interpretability and statistical power in epistasis detection by using prior biological knowledge (manuscript in preparation). Parts of this work were presented in IGES 2019 in the poster *Diane Duroux, Héctor Climente-González, Lars Wienbrandt, David Ellinghaus, Chloé-Agathe Azencott and Kristel Van Steen*, Improving efficiency in epistasis detection with a gene-based analysis using functional filters.

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
library(magrittr)
library(kableExtra)

cat("\\begin{chapabstract}{Abstract:}
Detecting epistatic interactions at the gene level is essential to understanding the biological mechanisms of complex diseases. Unfortunately, genome-wide interaction association studies (GWAIS) involve many statistical challenges that make such detection hard. This chapter considers three steps to analyze the conversion of SNP-SNP interactions into a gene-gene network through the integration of biological knowledge. First, we investigate the use of known gene-gene interactions to focus on the most promising gene-pairs. Second, we explore several functional SNP-to-gene mappings to map SNPs to these gene interactions, including expression quantitative trait loci (eQTL) and chromatin-based, and filter the GWAIS data accordingly. Third, we integrate SNP-pairs statistics into gene-pairs statistics using the adaptive truncated product method (ATPM). ATPM does not require parameters or a known null distribution, and it is fast to compute. We apply different variants of these three steps to a GWAS inflammatory bowel disease (IBD) dataset, and compare the results. Overall, knowledge integration decreases the number of tests performed, reducing computational time and boosting statistical power, and provides interpretability. We validate through a permutation analysis that type I error is under control. The different protocols produce different results, which highlight different mechanisms implicated in IBD while, at the same time, overlapping with the known biology of the disease. Using eQTLs results in more interactions than the standard approach, where all possible SNP pairs are investigated and SNPs are mapped to genes based on their position on the genetic sequence, albeit fewer significant gene-pairs. This pipeline not only detects interactions at the gene level, but also gives a global visualization that identifies neighbors, hubs and highly connected substructures.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}
La d\\'etection d'interactions \\'epistatiques entre g\\`enes est essentielle \\`a 
la compr\\'ehension des m\\'ecanismes des maladies complexes. Malheureusement, 
les \\'etudes d'association d'interactions g\\'enome entier (GWAIS) posent de 
nombreux d\\'efis statistiques qui rendent cette d\\'etection difficile. Ce 
chapitre consid\\`ere trois \\'etapes pour analyser la conversion 
d'interactions SNP-SNP en un r\\'eseaux de g\\`enes, \\`a travers l'int\\'egration 
de connaissances biologiques. Tout d'abord, nous nous int\\'eressons \\`a 
l'utilisation d'interactions g\\`ene-g\\`ene connues afin de nous limiter aux 
paires de g\\`enes les plus prometteuses. Deuxi\\`emement, nous \\'etudions 
plusieurs correspondances fonctionnelles entre SNPs et g\\`enes, bas\\'ees 
entre autre sur les loci r\\'egulant l'expression (eQTL) ou les contacts de 
chromatine, que nous utilisons aussi pour filtrer les donn\\'ees. 
Troisi\\`emement, nous proposons de d\\'eriver, des statistiques obtenues pour 
des paires de SNPs, une statistique pour une paire de g\\`enes, par une 
m\\'ethode de produit tronqu\\'e adaptative (ATPM). L'ATPM ne requiert ni de 
fixer des param\\`etres, ni de connaître la distribution nulle, et peut 
\\^etre calcul\\'ee efficacement. Nous appliquons diff\\'erentes variantes de ces 
trois \\'etapes \\`a un jeu de donn\\'ees GWAS des maladies chroniques 
inflammatoires de l'intestin (MICI), et comparons les r\\'esultats obtenus. 
L'int\\'egration de connaissances \textit{a priori} permet de r\\'eduire le 
nombre de tests statistiques \\`a effectuer, ce qui r\\'eduit les temps de 
calculs et augmente la puissance statistique. Elle facilite aussi 
l'interpr\\'etation des r\\'esultats. Une analyse par permutations nous permet 
de valider que l'erreur de type I est bien contr\\^ol\\'ee. Diff\\'erents 
protocoles produisent des r\\'esultats diff\\'erents, qui mettent en lumi\\`ere 
diff\\'erents m\\'ecanismes impliqu\\'es dans les MICI, tout en coïncidant avec 
des m\\'ecanismes d\\'ej\\`a connus de ces maladies. L'utilisation de 
correspondances SNPs identifi\\'es dans des eQTLs permet d'obtenir plus 
d'interactions SNP-SNP que l'approche standard, mais moins de paires de 
g\\`enes significatives. Notre pipeline permet non seulement de d\\'etecter 
des interactions entre g\\`enes, mais aussi de les visualiser dans un 
r\\'eseau global, et d'identifier voisinages, hubs, et structures fortement 
interconnect\\'ees.
\\end{chapabstract}")
```

## Introduction

In this chapter we study the detection of gene-level epistasis from
SNP-level epistasis to improve the interpretability of GWAIS. Our
analysis compares ways of converting statistical epistasis at the SNP
level into a gene-based statistical epistasis network. First, we
investigate different functional filters [@ma2015biological] and
SNP-to-gene mapping functions.
Second, as a bijectivity issue arises because a SNP can be mapped on
several genes, we investigate the use of current knowledge on gene-gene
interaction to focus on the most promising gene-pairs. We study whether
the epistatic interactions driving the phenotype are likely to be
currently in existing databases. Third, we used the adaptive truncated
product method to estimate gene-pairs significance.

## Materials and methods

### Dataset and initial quality control {#snp2gene-qc}

We investigated the IIBDGC dataset, described in 
Section \@ref(iibdgc-dataset), whose large sample size helps to overcome the 
issue of reduced statistical power common in GWAIS. 
We performed a quality control as in @ellinghaus_analysis_2016, reducing the 
number of SNPs from 196\,524 to 130\,071.

The IIBDGC dataset aggregates different cohorts, and contains
confounding population structure (Section \@ref(pop-structure)). PLINK
[@purcell2007plink] cannot take covariates in the logistic regression used to
detect epistasis (Section \@ref(methods:snp-epistasis)). In consequence, we adjusted
phenotypes to account for population structure using the top 7 principal
components as in @ellinghaus_analysis_2016. Essentially, we derive adjusted 
phenotypes from the logistic regression model by subtracting model-fitted 
values from observed phenotype values (i.e. response residuals).

### Gene interaction detection procedure {#methods:pipeline}

As we describe below, we applied four different functional filters to
the dataset. The functional filter used known interactions between
genes, and the three different ways of mapping SNPs to genes described in Section
\@ref(intro-interpretability), and hence, to
these interactions. In essence, in each of the filtered datasets we only tested
the interactions between SNPs which can be mapped in a particular way to
a pair of interacting genes. The resulting four datasets
(including an unfiltered dataset) were analyzed separately. For
convenience, we will refer to their associated analyses using the terms
*Standard*, *physical*, *eQTL* and *chromatin*. For each dataset, the
entire pipeline described below is applied and the four obtained outputs
are subsequently compared. An overview of the whole pipeline is
available in Figure \@ref(fig:pipeline).

(ref:fig-pipeline-caption) Overview of the gene-gene interaction detection procedure (from @duroux_2019_iges). The whole protocol is described in Section \@ref(methods:pipeline).

```{r pipeline, echo=FALSE, fig.cap='(ref:fig-pipeline-caption)'}
knitr::include_graphics('fig/pipeline2.PNG', dpi = NA)
```

#### Functional SNP pre-filtering {#methods:prefiltering}

The initial step of the protocol is a functional SNP pre-filtering,
which has three stages. First, we mapped the SNPs in the dataset to
genes using FUMA [@watanabe_functional_2017]. FUMA is a post-GWAS
annotation tool. Its SNP2GENE function takes GWAS summary statistics and
maps significant SNPs to genes according to both physical and functional
criteria specified by the user. We created an artificial input where
every SNP is significant in order to perform such mapping on all the
SNPs. We performed three SNP-gene mappings using SNP2GENE: physical,
eQTL and 3D chromatin interaction. In the physical mapping, we mapped a
SNP to a gene when the former was within the boundaries of the latter
$\pm$ 10 kb. The eQTL mapping uses eQTLs obtained
from GTEx [@gtex_consortium_genetic_2017]. We mapped an eQTL SNP to its target gene when the
association P-value was significant in any tissue ($\text{FDR} < 0.05$).
Lastly, in the 3D chromatin interaction mapping, we mapped a SNP to a
gene when a contact had been observed between the former and the region
around the latter's promoter (250 bp upstream and 500 bp downstream from
the transcription start site) in any of the Hi-C datasets included in
FUMA ($\text{FDR} < 10^{-6}$). The chromatin mapping might contain new,
undiscovered, regulatory variants which, as eQTL, regulate the expression
of a gene.

Second, after obtaining the SNP-to-gene mappings, we used Biofilter 2.4
[@pendergrass_genomic_2013] to obtain the candidate gene-pairs
subsequently investigated for epistasis evidence. Biofilter generates
pairs of genes with evidence of co-function across multiple publicly
available biological databases. It includes genomic locations of SNPs
and genes, as well as known relationships among genes and proteins such
as interaction pairs, pathways and ontological categories.
Notably, it does not use trait information.
Specifically, we considered only pairs of genes for which both genes
could be mapped, using any of the mappings, to a SNP in our GWAS
dataset. We used only gene pairs supported by evidence in at least 2
databases. When the two SNPs of a pair were located in the HLA region,
we removed the pair, as this complex genomic region is currently not
well understood. Additionally, we removed self-interactions, as
detection of within-gene epistasis requires special considerations and
is beyond the scope of this paper.

Lastly, we filtered the datasets again to explore exclusively
interactions between SNPs mapped to genes known to interact, according
to Biofilter. For that purpose we first converted the Biofilter
gene-pair models into SNP-pair models separately via each of the FUMA
SNP-gene mappings described above. Then, from these SNP pairs sets, we
built the four datasets enumerated above: one without any filter (*Standard*); 
and one for each SNP to gene mapping (*Physical*, *eQTL*,
*Chromatin*). It is worth pointing out that SNP-pair models
were also built based exclusively on the corresponding mapping e.g.
*physical* contains exclusively pairs of SNPs which can be associated to
pairs of genes via a physical mapping. This helps interpretability and
keeps the number of tested interactions under control.

#### Post-filtering quality control

Additional quality controls were performed on each of the four generated
datasets. As motivated in @gusareva_practical_2014, only common variants
($\text{MAF} > 5$%) and in Hardy--Weinberg equilibrium ($\text{P-value} > 0.001$) are
considered. Also, we pruned SNPs that are in linkage equilibrium 
($R^{2} > 0.75$). Lastly all risk SNP described in @liu2015association were 
included.

#### SNP-level epistasis detection and multiple test correction {#methods:snp-epistasis}

We used PLINK 1.9 to detect epistasis through a linear regression on the adjusted phenotypes:

$$Y = \beta_0 + \beta_1 g_A + \beta_2 g_B + \beta_3 g_A g_B,$$

where $g_A$ and $g_B$ are the genotypes under additive encoding for SNPs
A and B respectively; $\beta_0$, $\beta_1$, $\beta_2$, $\beta_3$, are
the regression coefficients. PLINK performs a statistical test to
evaluate if $\beta_3 \neq 0$. Crucially, it only returns SNP-pairs
with a P-value lower than a specified threshold. We used the default
0.0001.

We only considered pairs of SNPs not in strong LD ($R^{2}<0.75$) and
that could be mapped to the corresponding SNP-model obtained from
Biofilter, with self-gene interactions removed (Section
\@ref(methods:prefiltering)). This impacted the total number of
statistical tests. 

To correctly account for multiple testing, the P-value threshold of
significance had to be dataset-dependent as the number of tested SNP
pairs changed from dataset to dataset (Section
\@ref(methods:prefiltering)). We obtained the threshold through a
permutation analysis as in @hemani_detection_2014. In essence,
for each dataset, we permuted the phenotypes 400 times and measured SNP
association as above. This produced a null distribution of the extreme
P-values for this number of tests given the LD structure. For each
dataset, we took the most extreme P-value from each of the 400
permutations and set the threshold for 5% family-wise error rate
(FWER) to be the 95% percentile of these most extreme P-values.

#### From SNP-level to gene-level epistasis {#methods:gene-networks}

Next we converted SNP pairs into gene-pairs using their respective FUMA 
and Biofilter mappings (Section \@ref(methods:prefiltering)). 
The exception was the *Standard*
dataset, where SNPs are physically mapped to genes, and no Biofilter
restriction on which pairs of genes can interact was applied. Still,
self-interactions were removed.

Then, we computed gene-level statistics from the respective SNP-level
statistics of the involved SNPs. In this regard, all $N$ pairs of SNPs
mapped to a gene pair are taken as a set of tests on the same global
null hypothesis $H_{0i}$, where $i=1,2,\dots ,N$. @zaykin2002truncated
developed the truncated product method (TPM) as a method to combine
P-values on a same global hypothesis. It does so by computing the
statistic $W(\tau)=\prod_{i=1}^{N} p_i^{I(p_{i}\le\tau)}$ where
$I(\cdot)$ is the indicator function and $\tau$ is the truncation point.
A P-value $\hat{s}(\tau)$ can be the estimated for a given $W(\tau)$.
TPM was interesting since we did not not have
P-values for every SNP pair but for the most strongly associated ones
only (Section \@ref(methods:snp-epistasis)). However, TPM requires setting the
truncation point $\tau$, a parameter that is arbitrary and might be
gene-pair specific. On top of that, the null distribution of $W(\tau)$
is unknown when P-values are correlated, as is the case. To solve these
problems the adaptive truncated product method (ATPM) was proposed
[@sheng2013adaptive]. ATPM explores different $\tau$, choosing the one
that produces the minimum P-value $\hat{s}(\tau)$. We estimated the
distribution of the ATPM using permutations as in @ge2003resampling.
Specifically, we created $B = 999$ permuted datasets by permuting the
phenotype vector. Based on the $b^{th}$ permuted dataset,
$1 \leq b \leq B$, we perform the N individual tests. We used three
values for $\tau$ (0.001, 0.01, 0.05) and set the significance level
$\alpha=0.05$. The specific procedure goes as follows:

1.  For each gene-pair in the output of the original data analysis,
    based on $p{_1}^{(b)},...,p{_N}^{(b)},1\le b\le B$, calculate the
    truncated product statistics for each candidate truncation threshold
    for the original data and B permuted datasets.

2.  Use Ge's algorithm to obtain the estimated P-value
    $\hat{s}_{k}^{b}=\frac{\sum_{l=0}^B I(W(\tau_k)_b \geq W(\tau_k)_l)}{B+1}$,
    $1 \le k \le 3$, $0 \le b \le B$

3.  Calculate $M{_b}=min_{1 \le k \le 3}\hat{s}_{k}^{b}$

4.  The adjusted P-value for the adaptive truncated product statistic
    $M$ is estimated as
    $P_{minP^{(b)}}=\frac{\sum_{b=0}^B I(M \geq M_b)}{B+1}$.

5.  We reject the joint null hypothesis if the adjusted P-value is
    smaller than the global significance level $\alpha$.

## Results

### Type I error

The "multi-stage" nature of the protocol presented in Section
\@ref(methods:pipeline) required controlling the type I error. For
that purpose, we performed 1\,000 permutation analyses for each of the
four datasets, permuting the phenotypes and running the entire protocol
to detect significant gene pairs. When at least one significant
gene-pair is observed in a permutation, that permutation is considered a
false positive (FP). This allowed to compute the type I error rate as
$\frac{ \text{\# FP}}{1000}$. We observed that the type I error was
under control for all four datasets (3.6%, 3.7%, 6.1%, and 3.9% for *Standard*, 
*Physical*, *Chromatin*, and *eQTL*, respectively).

### Chromatin contacts map more SNPs per gene than other mappings {#results-power-gain}

We considered three procedures to map SNPs to their gene, as a proxy for
their link to functionality (Section
\@ref(methods:prefiltering)): *physical*, *eQTL* and *chromatin*.
*Chromatin* produced the largest number of mappings (2\,394\,589), an
order of magnitude more than *eQTL* (411\,120) and *physical* (174\,879)
(Table \@ref(tab:datasets-snps)). Similarly, *chromatin* has 
the largest number of SNPs mapped to an individual gene, followed by *eQTL* and
*physical* (Figure \@ref(fig:snps-per-mapping)A). Nonetheless, 
different genes had very unequal contributions from each of the mappings 
(Figure \@ref(fig:snps-per-mapping)B). This is consistent with 
the striking differences between the ranking of genes according to the number of 
SNPs mapped on them (Figures \@ref(fig:snps-per-mapping)C, D and E): 
in general, the genes with the most SNPs mapped using the eQTL mapping had 
relatively few SNPs mapped in the chromatin mapping, and viceversa.

(ref:fig-snps-per-mapping-caption) **(A)** Number of SNPs per gene for each of the three mappings described in Section \@ref(methods:prefiltering). Outliers are not displayed to facilitate visualization. **(B)** Ranking of genes with most SNPs mapped using any of the mappings, colored by mapping. Only genes with more than 100 SNPs mapped to it are displayed. **(C,D,E)** Comparison between the rank of each gene according to the number of SNPs mapped to it using each mapping.

```{r snps-per-mapping, echo=FALSE, fig.cap='(ref:fig-snps-per-mapping-caption)'}
knitr::include_graphics('fig/mappings_per_gene.png', dpi = NA)
```

The number of mappings is directly linked the number of SNPs and 
interactions tested per dataset (Table \@ref(tab:datasets-snps)). As it can be
observed, restricting our search exclusively to Biofilter-plausible interactions
leads to an increase in statistical power with respect to the *Standard* 
protocol. Specifically, the number of tests are between 1 and 2 orders of 
magnitude smaller.

```{r datasets-snps, echo=FALSE}
datasets_snps <- rbind(
      c('# SNPs', 38225, 16417, 30146, 16652),
      c('# SNP-gene mappings', format(c(NA, 1.7e5, 2.4e6, 4.1e5), scientific = T)),
      c('# tests', format(c(7.2e8, 4.6e6, 8.9e7, 2.2e7)))
  )
colnames(datasets_snps) <- c('', 'Standard', 'Physical' ,'Chromatin', 'eQTL')

kable(datasets_snps, 
      caption = 'Properties of the different SNP-gene mappings and the filtered datasets.', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = "striped")
```

### The *Physical* protocol does not recover any SNP interaction

We searched SNP epistasis in the four datasets (Section 
\@ref(methods:snp-epistasis)). The different epistatic SNP-SNP networks are 
described on Table \@ref(tab:snp-network) and Figure 
\@ref(fig:snp-networks). Strikingly, while the *Standard* protocol generated the 
largest network (55 nodes), the *eQTL* one was the largest by number of 
interactions (64). The *Physical* protocol produced no significant pairs.

```{r snp-network, echo=FALSE}
snp_networks <- rbind(
      c('# significant pairs', 57, 0, 19, 64),
      c('# nodes', 55, 0, 20, 46),
      c('# connected components', 12, NA, 5, 6),
      c('Size of the largest component', 25, NA, 11, 17),
      c('Average degree', 2.07, NA, 1.90, 2.78)
  )
colnames(snp_networks) <- c('', 'Standard', 'Physical' ,'Chromatin', 'eQTL')

kable(snp_networks, 
      caption = 'Properties of the SNP networks from the different datasets.',
      booktabs = TRUE) %>% 
  kable_styling(latex_options = "striped")
```

(ref:fig-snp-networks-caption) SNP-level epistasis networks for *Standard* (orange), *eQTL* (green), and *Chromatin* (violet) (Sections \@ref(methods:prefiltering) and \@ref(methods:snp-epistasis)). The *Physical* dataset is absent, as no SNP pairs were significant.

```{r snp-networks, echo=FALSE, fig.cap='(ref:fig-snp-networks-caption)'}
knitr::include_graphics('fig/snp_networks.pdf', dpi = NA)
```

### Gene-level network {#results:gene-networks}

We converted the aforementioned SNP-pair networks into gene-pair epistasis
networks, estimating their significance through ATPM (Section 
\@ref(methods:gene-networks)). Most of the SNP-pairs mapped to exclusively one 
gene pair in *eQTL* and *Standard*, removing possible sources of ambivalence 
(Figure \@ref(fig:sign-snps-per-gene-and-viceversa)A). That was 
not the case under the chromatin mapping, where it was more common for the same
SNP pair to map to different gene pairs. We then compared the relationship
between significant gene-pairs and the number of significant SNP pairs that map
to them (Figure \@ref(fig:sign-snps-per-gene-and-viceversa)B). 
Interestingly, most significant gene pairs are supported by 
relatively small number of SNPs: either few in number, or few with respect to 
the total number of SNP pairs for that gene pair.

(ref:fig-sign-snps-per-gene-and-viceversa-caption) Relationship between the number of significant SNP pairs and of significant gene pairs. **(A)** Histogram of the number of significant gene pairs mapped to the same SNP pair. **(B)** Relationship between the total number of SNP pairs mapped to the same gene pair (y-axis), and the percentage of all significant SNP-pairs between all the SNP-pairs mapped to the same gene (x-axis). Data points are semi-transparent, so multiple points stacked result in a darker shade.

```{r sign-snps-per-gene-and-viceversa, echo=FALSE, fig.cap='(ref:fig-sign-snps-per-gene-and-viceversa-caption)'}
knitr::include_graphics('fig/sign_snps_per_gene_and_viceversa.pdf', dpi = NA)
```

We built an epistatic gene network from the significant gene pairs
(Methods \@ref(methods:gene-networks)), shown in Figure \@ref(fig:gene-networks)
and Table \@ref(tab:gene-networks-stats). Overall, the *Standard* protocol still
produces the largest network, and contains more connected components and
significant gene-pairs. On the other hand, *Chromatin* and *eQTL* mappings
produce similar networks in terms of sizes, number of gene-pairs and
connected components. However, both *Chromatin* and *eQTL*'s networks are
notably smaller than *Standard*'s (11 and 10 nodes versus 29,
respectively). 

*Standard*'s nodes are proportionally more clustered in
connected components, while most *eQTL* and *Chromatin*'s connected
components are composed of only a pair of genes. Although this might
reveal the affection of a common mechanism, it is likely a result of the
overlap in the genome of multiple genes, which are mapped to highly
overlapping sets of SNP.

```{r gene-networks-stats, echo=FALSE}
gene_networks <- rbind(
      c('# significant pairs', 26, 0, 5, 7),
      c('# nodes', 29, 0, 10, 11),
      c('# connected components', 8, NA, 5, 5),
      c('Size of the largest component', 6, NA, 2, 3),
      c('Average degree', 1.79, NA, 1, 1.27)
  )
colnames(gene_networks) <- c('', 'Standard', 'Physical' ,'Chromatin', 'eQTL')

kable(gene_networks, 
      caption = 'Properties of the gene networks from the different datasets.', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = "striped")
```

(ref:fig-gene-networks-caption) Gene-level epistasis networks for *Standard* (orange), *eQTL* (green), and *Chromatin* (violet) (Section \@ref(methods:prefiltering)).

```{r gene-networks, echo=FALSE, fig.cap='(ref:fig-gene-networks-caption)'}
knitr::include_graphics('fig/gene_networks.pdf', dpi = NA)
```

A hub is node with a number of links that greatly exceeds the average.
For this application, we define a hub as a node having a degree strictly
superior to three. Only the *Standard* process contains such hubs: *P4HA2*,
*NKD1*, *RNU4ATAC4P*, *C1orf141*, *IL12RB2*, *IL23R* and *USP4*.

Jointly, 38 significant gene pairs are involved in at least one method,
involving 46 unique genes. Seven chromosomes are involved in epistasis.
Notably, 39% of the epistatic genes are located in chromosome 3 and 22%
in chromosome 10.

### *Chromatin* and *Standard* mappings partially replicate previous studies on IBD {#results-ibd-soa}

Several genetic studies studying epistasis on IBD have been conducted
[@lin2017genetic; @vermeire2004genome; @pedros2015epistatic; @mcgovern2009genetic; @glas2009novel; @lin2013nod2].
We compared them to our results at the gene level, the minimal
functional unit at which we expect genetic studies on different
populations to converge. For instance several studies showed epistatic
alterations involving interleukins, like *IL-10* [@lin2017genetic],
*IL-17* and *IL-23* [@mcgovern2009genetic], and *IL-2/IL-21* and
*IL-23R* [@glas2009novel]. Encouragingly, *Standard*'s results include
interactions involving both *IL-10* and *IL-23*, although we do not find
support for the specific interactions described in the aforementioned
studies. In fact, the *Standard* protocol highlights the relevance of
interleukins as hubs (Section \@ref(results:gene-networks)). Out of the five 
gene interactions retrieved in *Chromatin* pipeline, three of them involve at 
least one interleukin. @lin2013nod2 detected interactions involving *NOD2*, 
with both *IL-23R* and other genes. Our *Standard* protocol also detects two
potentially new epistasis interactions involving *NOD2*.

## Discussion

In this chapter we explore protocols for functional filtering for
epistasis detection on an IBD dataset (Section
\@ref(methods:pipeline)). This is expected to bring two
advantages. The first one is an increase in statistical power.
The high dimensionality of GWAIS data requires a conservative
multiple testing correction and limits the detection of epistasis with low effect
sizes. The proposed protocol tackles this issue, while controlling for
type I error. It does so by limiting the number of tests by filtering
the dataset with functional filters. As we observe in Section 
\@ref(results-power-gain), the reduction is notable. The second advantage is an
improvement of the interpretability of the results, by examining only
statistical interactions that map to a known biological interaction. As shown
in Section \@ref(results-ibd-soa), the proposed eQTL and, specially, chromatin 
mappings provide results which match the biology of IBD, while corresponding 
to known interactions. On the other hand, the *Standard* protocol detects 
multiple interactions that are hard to interpret. For instance, several 
interactions involve RNA genes of unknown function (e.g. *LOC101927272* or 
*LINC02178*). Hence, our results stress the relevance of regulatory variants
in susceptibility to IBD.

In this chapter, we aim at developing a set of guidelines for the detection of 
gene epistasis, with an application to IBD. Nonetheless, epistasis detection 
at the gene level still requires making many
choices which were out of the scope of this work. One instance is the
choice of encoding for the genotypes. In this work we used the additive
encoding, which can lead to an increased false positive rate
[@van2019increase]. Also we focused on linear regression as epistasis
detection algorithm, as it accepts a continuous outcome variable,
corrects for main effects, and is computationally efficient. However,
other algorithms which share these properties but make different 
assumptions about epistasis (like the ones used in Chapter \@ref(epi-network)), 
would have been suitable as well. 

With regards to the aspects of GWAIS that are within the scope of this work,
it would interesting for the community for us 
to provide a set of recommendations based on our experience, and the results
justifying each of them. Before we reach that point, a few extra experiments 
are required.

Our current protocol produces compelling hypotheses,
and shows the benefits of functional filtering with regards to statistical power and 
interpretability. However, the multi-stage nature of the process makes it
impossible to find out what each of the step brings. For instance, if the
detected interactions in any dataset are just a subset of the interactions that could be 
found without filtering out the interactions not in Biofilter; if they are
a subset of the interactions that could be found by a joint *physical+chromatin+eQTL*
mapping; or how often are interactions between genes mapped to SNPs through 
different mechanisms (e.g. a gene regulated by eQTL and a gene physically mapped). 
The answers to such questions are relevant to the community, 
and cannot be answered without isolating their effect from the Biofilter 
interactions. In other words, it would be useful to just map the results of a 
conventional GWAIS result, to observe how they differ from a conventional 
mapping. Related to this point, risk SNPs from GWAS are often located
in chromatin that is active in the tissues involved in the disease 
[@boyle_expanded_2017]. Hence, the presented protocol might lead to the most biologically
plausible epistatic interactions while boosting the power if it focused exclusively
on eQTLs and chromatin mappings obtained in the tissues relevant 
to IBD (intestines and leukocytes).

It would also be interesting to explore alternative sources of known
interactions. In this chapter we worked
exclusively on interactions from Biofilter, which compiles multiple databases.
The database that Biofilter built contained 37\,266 interactions. This
is notably smaller than other gene interaction databases, like HINT 
(@das_hint_2012, 173\,797 interactions, used in Chapter \@ref(genesis)), or 
STRING (@szklarczyk_string_2019, 11\,759\,455 interactions). Hence changing
databases might result in more, equally interpretable, detected interactions.

Pathway enrichment analyses can inform about the broader framework 
in which the observed gene epistasis occurs. I would like to adapt the "network neighborhood 
search" procedure from @yip2018network to build appropriate gene sets. In 
summary, given reference biological network (e.g. the Biofilter network), a 
gene set for a given pair of genes is obtained in three steps:

1. Remove the edge connecting the two genes in the reference network.
2. Find the shortest path between them in the reference network.
3. Create a gene set including the initial two genes and all the genes in the
shortest path that are part of the epistasis network as well.

Another important question is which null hypothesis
to test in the pathway enrichment analyses. 
In this regard, the literature often distinguishes two kinds of test:
self-contained and competitive [@wang_gene_2011]. As in our study we do not
have gene-wise statistics, we are restricted to the former.
Those tests compare the overlap between a pathway and the gene set to 
the expected overlap from taking equally-sized random sets from the universe
of genes. This is often tested using a hypergeometric test. However, this 
approach requires deciding *a priori* what that gene universe is. Selecting
all the known genes is not an option, as a GWAIS experiment surveys all the 
genome unbiasedly, but not necessarily so all the genes. This is specially true
in an array focused on immunogenomics. Indeed, we observe this bias across all
mappings in the genes mapped to the SNPs with main effects 
(Figure \@ref(fig:platform-bias)).

(ref:fig-platform-bias-caption) Plaform bias in the genes hit by SNPs with significant main effects. Every data point is a gene. The *x*-axis represents the number of significant SNPs mapped to that gene. The *y*-axis represents the expected number, if a random set of SNPs of the same size were selected $\left( \frac {\text{(# significant SNPs)} \times \text{(# SNPs mapped to gene)}} {\text{# SNPS}} \right)$. In red, a line with a slope of 1 representing the case were SNPs are selected at random.

```{r platform-bias, echo=FALSE, fig.cap='(ref:fig-platform-bias-caption)'}
knitr::include_graphics('fig/main_effect_bias.png', dpi = NA)
```

Hence, I propose computing the gene 
background in a dataset specific way. For instance, the *Chromatin* 
results are analyzed in a gene universe where only the genes with a chromatin
mapping to the chip are used. However not all genes are surveyed at the same
resolution, as we observe in Figure \@ref(fig:sign-snps-per-gene-and-viceversa).
If two genes are equally involved in a disease, we are more likely to find an 
association in the gene which we are testing more often. Hence, I would like to
weigh every gene by the number of SNPs that map to it, which should provide 
a conservative null hypothesis. 

Lastly, the protocol presented here is a
complex, multi-stage approach which can be useful to any researcher with any
GWAIS dataset. In consequence, it would be useful to provide a dataset-agnostic
computational pipeline. The user would just need to provide a gene-gene network, 
a SNP-gene mapping, and a GWAIS dataset, and would be given two epistasic
networks, SNP- and gene-based respectively. Generating such a pipeline would 
also allow us to answer the questions outlined above faster.

<!--chapter:end:04-snp2gene.Rmd-->

# High-order epistasis detection through fusion of epistasis networks {#epi-network}

The contents of this chapter are based on *Héctor Climente-González, Diane Duroux, Chloé-Agathe Azencott, Kristel Van Steen*, High-order epistasis detection through fusion of epistasis networks (manuscript in preparation).

```{r echo=FALSE, results = "asis", eval='is_latex_output'}
cat("\\begin{chapabstract}{Abstract:}
In this chapter we explore the detection of high-order epistasis in inflammatory bowel disease (IBD). We model high-order epistasis through a linear regression with interaction terms of orders up to 4. The number of tests increases exponentially with the order of the interactions, resulting in low statistical power and computational challenges. To tackle this, we devised a two-step approach to restrict the number of tests. In a first step, we apply a variety of second-order epistasis detection methods, and build an epistasis network that aggregates all the detected interactions. In a second step, we take all star subnetworks composed of 3 and 4 nodes, and search for high-order epistasis between them. The rationale is that SNPs involved in high-order epistasis are more likely to display second-order as well. The advantage of this approach is that we keep the number of tests under control, performing 124 tests instead of the $5 \\times 10^{17}$ required to exhaustively search order 4 interactions. Indeed, we are able to detect 37 high-order interactions (FDR 10\\%). Mapping these SNP-SNP interactions to gene-gene interactions suggests the involvement of the complement system in susceptibility to IBD.
\\end{chapabstract}
\\vskip 0.2in
\\begin{chapabstract}{Résumé :}
Dans ce chapitre nous nous int\\'eressons \\`a la d\\'etection d'interactions 
\\'epistatiques d'ordre sup\\'erieur dans les maladies chroniques 
inflammatoires de l'intestin (MICI). Nous mod\\'elisions ces interactions 
par une r\\'egression lin\\'eaire comportant des termes d'interaction allant 
jusqu'\\`a l'ordre 4. Le nombre de tests statistiques augmentant de façon 
exponentielle avec l'ordre des interactions, cela r\\'eduit fortement la 
puissance statistiques tout en cr\\'eant des difficult\\'es calculatoires. 
Pour r\\'esoudre ces probl\\`emes, nous proposons une approche en deux \\'etapes 
pour r\\'eduire le nombre de tests. Dans la premi\\`ere \\'etape, nous appliquons 
diverses m\\'ethodes de d\\'etection d'\\'epistasie d'ordre 2, et construisons un 
r\\'eseau d'\\'epistasie qui agr\\`ege toutes les interactions ainsi d\\'ecouvertes. 
Dans un second temps, nous testons tous les sous-r\\'eseaux en \\'etoiles de 3 
ou 4 nœuds. Nous faisons ici l'hypoth\\`ese que les SNPs impliqu\\'es dans des 
interactions d'ordre 3 ou 4 sont plus susceptibles d'apparaître aussi 
dans des interactions d'ordre 2. L'avantage de notre approche est de 
contr\\^oler fortement le nombre de tests statistiques, ici 124 au lieu des 
$5 \\times 10^{17}$ requis pour une recherche exhaustive. Cela nous 
permet de d\\'etecter 37 interaction d'ordre 3 ou 4 (FDR 10\\%). Traduire 
ces interactions SNP-SNP en interactions g\\`ene-g\\`ene sugg\\`ere l'implication 
du syst\\`eme du compl\\'ement dans la susceptibilit\\'e aux MICI.
\\end{chapabstract}")
```

## Introduction

Although in the previous chapter we focused on interactions between two loci, some studies suggest that high-order epistasis, interactions between more than two loci, that is to say, strongly shapes complex traits [@taylor_higher-order_2015]. However, exhaustively searching for high-order epistasis aggravates the challenges of GWAIS (Section \@ref(gwais)): a large number of statistical tests to consider, higher diversity of ways an interaction can occur, and results harder to interpret. Under the rationale that high-order interactions are likely to show also second-order effects, second-order epistasis networks have been leveraged on in the past to detect high-order epistasis [@hu_statistical_2012]. In this chapter we explore high-order epistasis interactions in IBD using a similar approach. We do so in two steps. First, we apply a representative variety of second-order epistasis detection methods, and construct an epistasis network. Then we search for high-order epistasis between the SNPs in the network and their neighbors. As in Chapter \@ref(genesis), the first step also allows us to compare the different methods.

## Materials and methods

### Data, quality control and preprocessing

We studied genetic data from the IIBDGC (Section \@ref(iibdgc-dataset)), and performed a quality control and preprocessing as in Section \@ref(snp2gene-qc). Lastly, we performed LD pruning $R^{2}>0.75$ to remove SNPs in strong LD, taking down the final number of SNPs to 38\,225, which yields $7.3 \times 10^8$ pairs to test. In this chapter we did not apply any functional pre-filter to the data.

### Epistasis detection methods {#methods-methods}

We applied a selection of epistasis detection methods that can process continuous phenotypes and that had a ready-to-use implementation.

#### Linear regression

As in Section \@ref(methods:snp-epistasis), we explored two-way epistasis using a linear regression. In this case, we accounted for multiple testing by setting the family-wise error rate to 0.05.

#### MB-MDR

MB-MDR [@lishout_efficient_2013] is an extension of the classical MDR method [@moore_flexible_2006]. For a pair of SNPs, MDR classifies every pair of alleles as either high risk, or low risk, depending on whether for the examined combination, the cases/controls ratio is positive or negative, respectively. In a cross-validation setting, the SNP-pairs are ranked using the classification accuracy on unseen samples. MB-MDR includes improvements over the MDR algorithm. First, it accounts for ambiguous pairs of alleles where the number of cases and controls are similar, boosting generalization. Second, MB-MDR extends MDR to continuous phenotypes. This is done by performing a *t*-test that compares the continuous trait between the samples with a specific a pair of alleles with that  of the samples with any other combination. If such difference is significant, the combination of alleles is classified as high risk or low risk, depending on the directionality of the change; else, it is considered ambiguous.

We ran MB-MDR with 999 permutations, and a codominant encoding. We kept interactions with a P-value \< 0.05 as in @lishout_efficient_2013.

#### EpiHSIC

EpiHSIC [@kam-thong_epistasis_2011] is a generalization to continuous phenotypes of EPIBLASTER [@kam-thong_epiblaster-fast_2011], which works on case-control studies. EPIBLASTER detects epistasis in a two stage approach. First all SNP pairs are screened using the difference in Pearson's correlation between two SNPs between cases and controls. Then, a logistic regression is used to detect epistasis among the candidate SNP pairs, similar to our what we do in Section \@ref(methods:snp-epistasis). @kam-thong_epistasis_2011 show that the difference in correlation coefficients is an instance of the Hilbert–Schmidt Independence Criterion (HSIC) which a particular choice of kernels. In EpiHSIC they propose a different selection of kernels that extends the screening step to continuous phenotypes. We used *gpuEpiScan*'s R implementation of this algorithm [@gpuEpiScan].

### High-order epistasis detection {#methods-high-order}

We built a SNP-SNP epistasis network using all the significant interactions from the presented methods. On this network, we selected star networks (one central node connected to all the others) as candidate sets for high-order epistasis detection. For each candidate set we built a linear model that included all possible interaction terms. For instance, for a network with three SNPs, *A*, *B* and *C*, we built the following model:

$$Y = \beta_0 + \beta_1 g_A + \beta_2 g_B + \beta_3 g_A g_B + \beta_4 g_A g_C + \beta_5 g_B g_C + \beta_6 g_A g_B g_C,$$

where the coefficients are the $\beta$ terms, and $g_A$, $g_B$ and $g_C$ are the genotypes under additive encoding for SNPs *A*, *B* and *C* respectively. We consider that there is a 3-way epistatic interaction between these three genes if $\beta_6$ is significantly different from 0. As under the null hypothesis the $\beta$ coefficients follow a standard Normal distribution, we evaluated a departure from it using a *t*-test. We restricted the sets to star subnetworks of size 3 and 4, as the number of tests grows exponentially with the number of SNPs. We accounted for multiple testing by setting the false discovery rate to 10%. For this purpose, the total number of tests was the number of $\beta$ for terms of order 3 and 4 for which a $\beta$ was obtained.

### Code availability

The different epistasis detection methods are implemented in pipelines with a similar interface, available on GitHub (https://github.com/hclimente/gwas-tools). The code necessary to reproduce this chapter's results, altogether with the corresponding analyses, are available on GitHub as well (https://github.com/hclimente/waffles).

## Results

### Epistasis detection methods produce relatively similar results

We applied different methods to search for epistatic interactions in the IIBDGC dataset (Section \@ref(methods-methods)). From these interactions, we built an epistasis network that included all significant pairs from any of the methods. In total they retrieved 823 epistatic interactions involving 459 SNPs (Figure \@ref(fig:overview)). MB-MDR is the method that detects the largest number of interactions (741), followed by logistic regression (68) and EpiHSIC (14). This is not surprising, since we did not apply any multiple testing correction to MB-MDR's output. Despite difference in size, MB-MDR's solutions greatly overlap with logistic regression's (37, 54.4%) and EpiHSIC (3, 21.4%) (Figure \@ref(fig:solutions-overlap)). In the same vein, 110 of the 459 SNPs in epistasis have significant main effects themselves (Fisher's Exact Test $\text{P-value} < 2.2 \times 10^{-16}$ on the overlap), despite both MB-MDR and logistic regression accounting for main effects. This large overlap points to a large agreement on the biological mechanisms of the disease. It also motivates searching for high-order epistasis preferentially among the subset of SNPs involved in second order interactions.

(ref:fig-overview-caption) Overview of the SNP epistasis networks produced by different epistasis detection (Section \@ref(methods-methods)). The network is split into several panels to depict the source of each interaction.

```{r overview, echo=FALSE, fig.cap='(ref:fig-overview-caption)'}
knitr::include_graphics('fig/overview_snp_solutions.png', dpi = NA)
```

(ref:fig-solutions-overlap-caption) Overlap between the solutions from the different epistasis detection methods. Note that the *y*-axis is in logarithmic scale.

```{r solutions-overlap, echo=FALSE, fig.cap='(ref:fig-solutions-overlap-caption)', out.width = '60%', fig.align = 'center'}
knitr::include_graphics('fig/overlap_solutions.png', dpi = NA)
```

(ref:fig-main-effect-overlap-caption) Main effects on IBD, measured using a logistic regression on the case-control phenotype. The 7 largest principal components are included in the model as covariates. SNPs involved in an epistasis interaction are colored in dark grey.

```{r main-effect-overlap, echo=FALSE, fig.cap='(ref:fig-main-effect-overlap-caption)'}
knitr::include_graphics('fig/manhattan_epistasis.png', dpi = NA)
```

### High-order epistasis interactions in IBD

We searched for 3rd and 4th order epistasis in 124 star subnetworks from the aggregated network (Section \@ref(methods-high-order)). Out of them 37 are significant at FDR 10% (Figure \@ref(fig:high-order)). The high proportion of the tested interactions that are significant (29.8%) supports the pertinence of our approach. Only three of the tested star subnetworks involved edges from different sources, probably due to the difference in solution size between MB-MDR and the other methods.

(ref:fig-high-order-caption) Overview of the detected high-order epistasis interactions, shown on the binary epistasis network. The edges involved in the 124 tested 3rd and 4th order interactions are highlighted in blue; the 37 among those that were significant, in red.

```{r high-order, echo=FALSE, fig.cap='(ref:fig-high-order-caption)', out.width = '60%', fig.align = 'center'}
knitr::include_graphics('fig/high_order_significant.png', dpi = NA)
```

### Mapping SNP to genes involves the complement system

We mapped the 823 SNP-SNP interactions to gene-gene interactions. We used any of the maps described in Section \@ref(methods:prefiltering), using exclusively genes with a HUGO symbol. As in Chapter \@ref(epi-snp2gene), we removed genes in the HLA regions and self-interactions. But, in contrast with that chapter, we make use of all three mappings at once, and hence expect gene-gene interactions where the genes were mapped via two different mappings. This resulted in 1\,113 different genes and 20\,826 different gene-gene interactions. When we restricted ourselves only to those binary interactions involved in one of the 37 significant high-order interactions, we obtained 162 genes and 5\,273 candidate gene-pairs. This one-to-many mapping illustrates well the problem of linking SNP-SNP interactions to their functional consequences. Restricting the 5\,273 interactions only to those that exist on STRING [@szklarczyk_string_2019], we reduced the number to 892 interactions involving 94 genes. We decided to focus on this latter set of interactions. Although it is doubtful that all these gene-gene interactions are truly linked to the disease, we conjecture that it will be enriched in relationships truly associated to the disease. The high interconnectedness of the network (Figure \@ref(fig:overview)) gives support to this idea, seeming to capture a reduced number of susceptibility mechanisms.

We first studied how many SNP-SNP interactions supported every gene-gene interaction. The gene-gene interactions with higher support involve immunity-related genes. More specifically, they involve multiple genes from the complement system (*C4A*, *C4B* and, to a lesser extent, *C2*). This set of genes is part of the innate system, and is part of the organism's reaction to pathogens. The involvement of the complement system in IBD has received some attention lately [@jain_complement_2014;@heybeli_complement_2016;@sina_intestinal_2018], as mice lacking complement proteins suffer intestinal infammation similar to IBD. In our results, we observe an interaction between the three aforementioned genes of the complement, and in greater numbers with other genes, the one supported by most SNP-SNP interactions being *NOTCH4*.

## Discussion

In this chapter I apply three methods for second order epistasis detection to the IIBDGC dataset. Most of the detected interactions came from the same method, MB-MDR, which exhibited a high sensitivity. Consistent with the observation that SNPs with main effects are involved in such epistatic interactions, we detected 3rd and 4th order interactions between the many SNPs with two-way epistasis. When we mapped the SNP pairs to candidate gene pairs, we observed that many gene-gene interactions involved genes involved in the complement system. This gives a genetic rationale to the recent observations of the involvement of the complement system in the etiology of IBD.

However, our approach has some shortcomings. First, searching for high-order epistasis in hubs from the epistatis network is not possible under the current approach. For instance, the node with the highest degree has 39 neighbors, which implies building a model which includes interaction terms up to order 39. In this regard, a possible way forward would be dividing such star subnetworks into sets of 3 and 4 nodes, and looking exclusively for 3rd and 4th order interactions respectively within them. Another issue is the mapping of SNPs pairs to gene pairs. In this regard, our work in Chapter \@ref(epi-snp2gene) is promising, and answering the questions remaining in that project will pave the way to this one. Some of these questions are whether it is beneficial to use tissue-specific SNP-gene mappings, when should a gene-gene filter be used, or how to appropriately carry out a pathway enrichment analysis on epistasis results. However, this chapter presents new challenges in these fronts as well, like how to use binary interactions when we work on ternary interactions. Although here I decomposed them into binary interactions, it is worth considering higher order gene-interactions obtained from databases, namely co-complexes. Another open question is how the results of the presented approach compare to the results of existing methods to detect high-order interactions [@llinares-lopez_casmap_2018;@suzumura_selective_nodate;@terada_lamplink_2016].

Additionally, in this study we faced a scarcity of methods that work on continuous phenotypes or accept covariates. Such methods are necessary to account for population structure in epistasis detection. In this chapter I have presented three such methods, and we have located a few more: GenEpi [@chang_genepi:_2018], and several approaches based on random forest-based approaches [@yoshida_snpinterforest:_2011; @li_detecting_2016] or mutual information [@pensar_genome-wide_2019]. Adding some of these methods to protocol is likely to increase the number of detected high-order interactions. Additionally, I see this as a good opportunity to compare the methods, as has been done already on methods that work on case-control phenotypes [@Shang2011;@chatelain_performance_2018].

<!--chapter:end:05-epi-network.Rmd-->

# Conclusions

In the last 10 years great progress has been made in the understanding of the genetic architecture of complex diseases through GWA(I)S. That success would not have been possible without the effort of multiple national and international consortia who coordinated research groups to tackle specific diseases. Nonetheless, the GWA(I)S experimental settings present challenges that hinder the discovery of susceptibility mechanisms, our understanding of biology and, ultimately, improving diagnosis and clinical care. Some of these challenges are low statistical power, the difficulty to link associated genotypes to functional changes, the multiplicity of ways to detect epistasis, and properly accounting for LD in the statistical treatment of the data. During my PhD I have explored how network methods can be used to tackle these challenges.

In **Chapter \@ref(genesis)** we applied and critically evaluated different high-score subnetwork search methods to the GENESIS breast cancer GWAS dataset. Such methods are particularly relevant in GWAS, as they address some of the drawbacks of the experimental setting: low statistical power and interpretability. Most network methods produced a biologically plausible answer, which by itself deepens our understanding of the susceptibility mechanisms acting out in this specific trait. However, this methodological comparison also highlights how radically different the solutions are from each other. This is the product of different ways of modeling the informal concept of high-score subnetwork mathematically. We combined the different solutions into a consensus network, which was simpler than the largest individual solutions, but preserved their most important topological and biological properties. Network aggregation is a promising approach to the multiplicity of network methods.

In **Chapter \@ref(martini)** I presented *martini*, an R implementation of SConES, one of the methods I used in **Chapter \@ref(genesis)**, that addresses some of its initial shortcomings, extending it to case-control phenotypes, and adding hyperparameter selection options and support for covariates. As I show Chapter \@ref(genesis), and expand in this chapter, SConES is a particularly flexible algorithm among high-score subnetwork search methods. Specifically, it has two hyperparameters that allow to fine-tune the topology and the sparsity of the selected subnetwork. Unfortunately, this flexibility comes at the price of appropriately tuning these parameters, an issue that is exacerbated by the instability of the solutions. In this regard, the feature selection scores I implemented helped finding more realistic solutions in some simulations, although one could still improve upon this proposition. One possibility would be to score the solution using the topology (edge density, centrality betweenness, number of connected components).

In **Chapter \@ref(epi-snp2gene)** we propose a protocol to improve both the statistical power and the interpretability of GWAIS, and apply it to the IIBDGC dataset. In essence, we employ three different ways of mapping SNPs to known gene interactions (physical, eQTL, and chromatin), and only test the corresponding pairs of SNPs. Indeed, we observe that the proposed protocols reduce the number of tests, and detect interpretable interactions which are not found using a conventional approach. For instance, the eQTL filter produces more SNP interactions than any other mapping, while the chromatin filter detected gene interactions that are strongly related to interleukins, a immune system family of molecules involved in IBD. Despite the promising results, there are two aspects of the protocol that need to be refined: decoupling SNP-gene mapping from known gene interactions, to study the contribution of each to the positive results; and topping the protocol with an approach to map the detected interactions to affected pathways.

In **Chapter \@ref(epi-network)** I apply a strategy for high-order epistasis detection to IBD. In essence, it consists of two steps. The first step involves building a SNP-SNP epistasis network using multiple epistasis detection tools. The second one consists on searching high-order epistasis between specific nodes of the network and their neighbors. Indeed, this approach boosts the statistical power of high-order epistasis detection by reducing the number of tests by 15 orders of magnitude, and allows us to detect 37 high-order interactions. In an exercise similar in its spirit to that of Chapter \@ref(epi-snp2gene), we map these interactions to known gene-gene interactions using functional SNP-gene mappings. In this way we link the detected interactions to the complement system, an emergent candidate susceptibility mechanism to IBD.

In this thesis I illustrated several hurdles in our way to identifying and understanding the genetics of complex diseases. I now conclude with a look into the state of the field, and future research directions related to my contribution.

**The future of network methods for GWAS.** The network methods I worked with during my PhD are notably heterogeneous. Although that heterogeneity stems from divergences in what different researchers aim to find, being able to obtain different points of view from a disease is a strength. However, clearer language and more exhaustive comparisons to other methods would be well-received in the methods' publications. With the exception of SConES, all high-score subnetwork search methods I tested work on gene-gene interaction networks. SConES on the other hand uses SNP-SNP networks. The latter kind of networks operate at a lower level than the gene and, hence, potentially can handle more information. For instance, it could contain information about the specific protein residues that participate in a protein-protein interface. It could also encode LD blocks, by weighing the edges of the network proportionally to the correlation between SNPs. The benefits from using SNP-SNP networks are highlighted by the the biologically relevant results obtained with SConES even when no protein-protein information was added to the SNP network. On the other hand, the SNP networks that I handled in this thesis were orders of magnitude more complex than the corresponding protein-protein interaction networks for the same dataset. This increases runtime prohibitively, and new methods, or faster versions of the current ones, are required to explore the potential of such, more informative, SNP networks.

**The role of network methods in bioinformatics.** My experience with network methods sometimes clashed with other people's impressions of the field during informal exchanges. In particular, some researchers receive network results and methods with a mix of puzzlement in the face of the overwhelming amount of information they represent, and a subsequent skepticism. In my opinion, this anecdotal evidence speaks to the state of network biology as a field, which lacks clear, agreed on, protocols and goals. Hence, I believe a multi-front effort must be made to close this gap. A first line of attack involves more accessible tools, with similar interfaces and proper documentation. One of my goals in Chapter \@ref(genesis) was to create such  an interface for the tools I used. Another subject to tackle is the development of better tools to visualize and manipulate network results. Innovative visualizations like hive plots [@krzywinski_hive_2012], and the package *tidygraph* to manipulate networks [@pedersen_tidygraph_2019] are moves in the right direction. Interaction databases are also part of the inaccessibility issue: @huang_systematic_2018 evaluated 21 such databases, often collecting overlapping information, and with unclear definitions of what *interaction* means. Efforts must be unified towards a single database, with a user-friendly interface and clarity about its contents -- with the caveat that such efforts may simply result in yet another database that will be used in parallel with the others^[https://xkcd.com/927/]. In this regard, I believe HINT [@das_hint_2012], which aggregates multiple databases, is a step in the right direction. This relates to the issue, also discussed in Chapter \@ref(genesis), of the different types of biological interactions: despite the preponderance of protein-protein interactions, other types of  interactions (like RNA -- RNA or transcription factor -- DNA) need to be better compiled and characterized.

**The future of GWAS.** Despite the issues I discussed, GWAS have had a sound success in identifying genetic associations with complex traits, therefore helping understanding their genetic basis. Many past and present initiatives are leveraging on more data to solve some of these issues: massive prospective cohort studies like UK Biobank [@sudlow_uk_2015], more GWAS on understudied populations, and initiatives in functional genomics like ENCODE [@the_encode_project_consortium_integrated_2012]. However, many statistical questions remain to be solved. Despite some efforts to unify practices [@wang_statistical_2018], the field is dominated by laboratory- and method-specific practices, which are rarely justified in published articles. For instance, whether to LD prune or not; map SNPs to the exact coordinates of a gene, or include some kilobases upstream and downstream; or what is the minimum MAF of SNPs to analyze. An open conversation is needed which confronts different protocols, and reaches consensuses where possible, and justified guidelines when it is not. On this front, network methods can be powerful allies, as shown in Chapter \@ref(genesis), but work in other fields, both experimental and *in silico*, is also required. For instance a proper incorporation of LD patterns to processing analysis might lead to both an increase in statistical power (by reducing the number of tests to one per independent test) and interpretability (by considering the true unit of variation), while accounting for population structure by default (as it is caused by LD). Also, the technology of choice for GWAS is slowly shifting from SNP arrays to the increasingly affordable whole genome-sequencing. This in itself is bringing a substantial change to the GWAS scene, by providing a deeper coverage which includes more rare variants. Nonetheless, this increase in the number of variants studied only increases our need for appropriately treating LD.

**The future of GWAIS.** Epistasis detection is an open and promising field. As in high-score subnetwork search, and as I describe in Chapter \@ref(epi-network), epistasis detection can be achieved through a multiplicity of tools that model it in different ways. As with the aforementioned network methods, we can exploit that multiplicity by collapsing the results into a unified view of disease. Yet, a few challenges remain ahead. The first one, developed above, is the lack of general guidelines for GWA(I)S. The second one, is the inability of most methods to account for population structure, either by accepting covariates, or by accepting an adjusted, continuous phenotype. This last problem also comes up in the study of continuous phenotypes, which were beyond the scope of this thesis. In Chapter \@ref(epi-snp2gene) I explored the possible contribution of functional pre-filtering to epistasis detection. In this regard, and encouraged by my experiences described in Chapters \@ref(genesis) and \@ref(martini), I believe in the potential of epistasis-detection methods that exploit prior knowledge in the form of a network.

**Open science with sensitive data.** I would like to finish reflecting about my experience on *open science* as a GWAS researcher. Understandably, GWAS data from human samples requires a careful treatment, as it contains very sensitive information about both the participants and their families. For that reason, working on GWAS data requires compromising on open data, one of the pillars of open science. Hence, at the moment, work on genetic data requires trust from all the other researchers who do not have access to a particular dataset. This might change in the future, when data obfuscation solutions that allow to operate on a dataset while hiding the identifiable information [@azencott_machine_2018] are fully developed. Yet, for the moment, the lack of open data should lead to a stronger commitment to open algorithms and results, to facilitate as much as possible the application of published tools to other datasets and study if the conclusions hold.

<!--chapter:end:06-conclusions.Rmd-->

# Funding acknowledgments {-}

This thesis was funded by the European Union's Horizon 2020 research and innovation program under the Marie Skłodowska-Curie [666003]. Additionally, the work presented in Appendix \@ref(block-hsic-lasso) was made possible by a scholarship from RIKEN AIP, which allowed me to spend 3 months in Makoto Yamada's laboratory in Kyoto. 

The co-authors of some chapters, and more generally the work presented in them, received additional sources of funding. Regarding Chapter \@ref(genesis), it was supported by funding from Agence Nationale de la Recherche (ANR-18-CE45-0021-01). Financial support for GENESIS resource and genotyping was provided by the Ligue Nationale contre le Cancer (grants PRE05/DSL, PRE07/DSL, PRE11/NA), the French National Institute of Cancer (INCa grant No b2008-029/LL-LC) and the comprehensive cancer center SiRIC, (Site de Recherche Intégrée sur le Cancer: Grant INCa-DGOS-4654). Computational resources for Chapters \@ref(epi-snp2gene) and \@ref(epi-network) have been provided by the Consortium des Équipements de Calcul Intensif (CÉCI), funded by the Fonds de la Recherche Scientifique de Belgique (F.R.S.-FNRS) under Grant No. 2.5020.11 and by the Walloon Region.

<!--chapter:end:07-funding.Rmd-->

# References {-}

<div id="refs"></div>

# (APPENDIX) Appendix {-}

# Block HSIC Lasso: model-free biomarker detection for ultra-high dimensional data {#block-hsic-lasso}

\includepdf[pages=-, scale=1]{articles/climente_2019.pdf}

`r if (knitr::is_html_output()) '
This manuscript is published in https://academic.oup.com/bioinformatics/article/35/14/i427/5529193.
'`

<!--chapter:end:11-block_hsic_lasso.Rmd-->

# The Functional Impact of Alternative Splicing in Cancer {#smartas}

\includepdf[pages=-, scale=1]{articles/climente_2017.pdf}

`r if (knitr::is_html_output()) '
This manuscript is published in https://www.cell.com/cell-reports/fulltext/S2211-1247(17)31104-X
'`


<!--chapter:end:12-smartas.Rmd-->

# Systematic Analysis of Splice-Site-Creating Mutations in Cancer {#scm}

\includepdf[pages=-, scale=1]{articles/jayasinghe_2018.pdf}

`r if (knitr::is_html_output()) '
This manuscript is published in https://www.cell.com/cell-reports/fulltext/S2211-1247(18)30397-8
'`

<!--chapter:end:13-scm.Rmd-->

# Susceptibility genes to breast cancer {#hboc-genes}

Nearly all known HBOC susceptibility genes encode tumor suppressors that participate in genome stability pathways (homologous recombination repair, replication fork stability, transcription-replication collisions, mismatch repair, and DNA damage signaling, checkpoints and cell death). 

## Homologous recombination repair

The homologous recombination repair pathway (HRR) deals with double strand DNA breaks by using the undamaged chromosome as template for error-free repair. After a DSB occurs, the MRN complex (MRE11, RAD50 and NDN) detects and binds the free DNA ends. Then, it promotes DNA damage checkpoint signaling.

HRR involves *BRCA1*, *BRCA2* and, actually, most of the HBOC genes. Because of its ability to interact with a wide range of proteins, BRCA1 is hypothetized to act as a recruitment scaffold. A deficiency of *BRCA1* is linked to the inability to trigger HRR. Mutations in the MRN complex have also been clinically associated to breast cancer, although dubiously so in the case of *RAD50* variants. Reassuringly, some other HBOC genes are interactors of the MRN complex and BRCA1/2.

## Replication fork stability

BRCA1 and BRCA2 protect newly synthesized DNA and promote the restart of stalled forks in an HRR-independent manner. In the absence of these proteins, newly synthesized DNA in a stalled fork would get degraded, leading to genome instability and increasing the risk of cancer.

## Transcription-replication collisions 

Collisions between transcription and replication are emerging as a source of genome instability. In particular, RNA-DNA hybrids called R-loops can form between the nascent transcript and the DNA template. They can lead to double-strand breaks and mutations. Both BRCA1 and BRCA2 participate in the resolution of R-loops, preventing their accumulation. In consequence, BRCA-deficient cells tend to suffer transcriptional stress that leads to genome instability. Nonetheless the relationship between this mechanism and proneness to HBOC is yet to be proven, and the genes involved further investigated.

## Mismatch repair

DNA mismatch repair (MMR) corrects base-base mispairs. When MMR is faulty, accumulations point mutations and genetic changes in repeated nucleotide sequences (microsatellite instability) occur. MMR also plays a role in error-free HRR.

## DNA damage signaling, checkpoints and cell death

Pathways involved in genome maintenance, cell cycle checkpoints and cell death usually eliminate cells with damaged DNA. When proteins involved in them are not active, some processes such as cell cycle arrest, apoptosis and senescence will not occur. In consequence, cells that undergo genomic alterations are allowed to proliferate. The most famous case of HBOC in this pathway is TP53, which coordinates the transcriptional induction of many genome stability factors.

(ref:tab-hboc-genes-caption) Overview of HBOC genes: estimated lifetime risk of breast cancer (age in years) and tumorogenic molecular mechanisms that involves them: homologous recombination repair (HRR), replication fork stability, transcription-replication collisions, mismatch repair (MMR), DNA damage signaling, checkpoints and cell death, and/or others. Adapted from @nielsen_hereditary_2016.

(ref:checkmark) $\checkmark$

```{r hboc-genes, echo=FALSE}
library(magrittr)
library(kableExtra)
library(readr)

hboc_genes <- read_table2(file = "
ATM 60%_by_age_80 (ref:checkmark) - - - (ref:checkmark) -
BARD1 Unknown (ref:checkmark) - - - - -
BLM Unknown - (ref:checkmark) - - - -
BRCA1 57-65%_by_age_70 (ref:checkmark) (ref:checkmark) (ref:checkmark) - (ref:checkmark) -
BRCA2 45-55%_by_age_70 (ref:checkmark) (ref:checkmark) (ref:checkmark) - (ref:checkmark) -
BRIP1 OR:_<_2.0 - - - - - -
CDH1 42%_by_age_80 - - - - - (ref:checkmark)
CHEK2 37%_by_age_70 - - - - (ref:checkmark) -
FAM175A Unknown (ref:checkmark) - - - - -
FANCC Unknown - (ref:checkmark) - - - -
FANCM Unknown - (ref:checkmark) - - - -
MLH1 ~19%_by_age_70 - (ref:checkmark) - (ref:checkmark) - -
MRE11 Unknown - - - (ref:checkmark) - -
MSH2 ~11%_by_age_70 - - - (ref:checkmark) - -
NBN OR:_3.0 (ref:checkmark) - - - - -
NF1 6.5-fold_up_ages_30-39 - - - - - (ref:checkmark)
PALB2 35%_by_age_70 (ref:checkmark) (ref:checkmark) - - - -
PMS2 SIR:_3.8 - - - (ref:checkmark) - -
PTEN 85%_by_age_70 - - - - - (ref:checkmark)
RAD51B Unknown (ref:checkmark) - - - - -
RAD51C Unknown (ref:checkmark) - - - - -
RAD51D Unknown (ref:checkmark) - - - - -
RECQL Unknown - (ref:checkmark) - - - -
RINT1 Unknown (ref:checkmark) - - - - -
STK11 32%_by_age_60 - - - - - (ref:checkmark)
TP53 25%_by_age_70 - - - - (ref:checkmark) -
", col_names = FALSE, col_types = 'cccccccc')

colnames(hboc_genes) <- c('Gene', 'Lifetime risk', 'HRR', 'Rep. fork stab.', 'Tr.-rep. clash', 'MMR', 'DNA Damage, apoptosis', 'Other')

hboc_genes <- data.frame(lapply(hboc_genes, function(x) {
  gsub("^-$", "", x)
  }))

hboc_genes <- data.frame(lapply(hboc_genes, function(x) {
  gsub("_", " ", x)
  }))

kable(hboc_genes, 
      caption = '(ref:tab-hboc-genes-caption)', 
      booktabs = TRUE) %>% 
  kable_styling(latex_options = "striped") %>%
  landscape()
```

<!--chapter:end:14-hboc.Rmd-->

