\usepackage{booktabs}
\usepackage{pdfpages}
\usepackage{env/psl-cover}
\usepackage{rotating}
\usepackage{fancyhdr}
\usepackage{bibentry}

% abstract for each chapter
\newenvironment{chapabstract}[1]%
{\leftskip1in\textbf{#1 }\itshape}%

%%% Fancy Header %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Fancy Header Style Options

\pagestyle{fancy}                       % Sets fancy header and footer
\fancyfoot{}                            % Delete current footer settings

%\renewcommand{\chaptermark}[1]{         % Lower Case Chapter marker style
%  \markboth{\chaptername\ \thechapter.\ #1}}{}} %

%\renewcommand{\sectionmark}[1]{         % Lower case Section marker style
%  \markright{\thesection.\ #1}}         %

\fancyhead[LE,RO]{\bfseries\thepage}    % Page number (boldface) in left on even
% pages and right on odd pages
\fancyhead[RE]{\bfseries\nouppercase{\leftmark}}      % Chapter in the right on even pages
\fancyhead[LO]{\bfseries\nouppercase{\rightmark}}     % Section in the left on odd pages

\author{\mbox{Héctor CLIMENTE GONZÁLEZ}}

\institute{MINES ParisTech}
\doctoralschool{Ingénierie des Systèmes, Matériaux, Mécanique, Énergétique}{621}
\specialty{Bio-informatique}
\date{1 Février 2020}

%% cotutelle
% \entitle{Thesis Subject in English}
% \otherinstitute{Institut Curie}
% \logootherinstitute{logo-institute}

%\jurymember{1}{Nadine ANDRIEU}{Mme., Institut Curie}{Présidente}
\jurymember{1}{Nadine ANDRIEU}{Mme., Institut Curie}{Examinatrice}
\jurymember{2}{Kristel VAN STEEN}{Mme., Université de Liège}{Rapporteuse}
\jurymember{3}{Antonio RAUSELL}{M., Imagine Institute}{Rapporteur}
\jurymember{4}{Laura FURLONG}{Mme., Pompeu Fabra University}{Examinatrice}
\jurymember{5}{Véronique STOVEN}{Mme., MINES ParisTech}{Directrice de thèse}
\jurymember{6}{Chloé-Agathe AZENCOTT}{Mme., MINES ParisTech}{Co-encadrante}

\frabstract{
  Cuius acerbitati uxor grave accesserat incentivum,
  germanitate Augusti turgida supra modum, quam Hannibaliano
  regi fratris filio antehac Constantinus iunxerat pater,
  Megaera quaedam mortalis, inflammatrix saevientis adsidua,
  humani cruoris avida nihil mitius quam maritus; qui paulatim
  eruditiores facti processu temporis ad nocendum per
  clandestinos versutosque rumigerulos conpertis leviter
  addere quaedam male suetos falsa et placentia sibi
  discentes, adfectati regni vel artium nefandarum calumnias
  insontibus adfligebant.

  Saraceni tamen nec amici nobis umquam nec hostes optandi,
  ultro citroque discursantes quicquid inveniri poterat
  momento temporis parvi vastabant milvorum rapacium similes,
  qui si praedam dispexerint celsius, volatu rapiunt celeri,
  aut nisi impetraverint, non inmorantur.

  Vita est illis semper in fuga uxoresque mercenariae
  conductae ad tempus ex pacto atque, ut sit species
  matrimonii, dotis nomine futura coniunx hastam et
  tabernaculum offert marito, post statum diem si id elegerit
  discessura, et incredibile est quo ardore apud eos in
  venerem uterque solvitur sexus.

  Sed tamen haec cum ita tutius observentur, quidam vigore artuum
  inminuto rogati ad nuptias ubi aurum dextris manibus cavatis
  offertur, inpigre vel usque Spoletium pergunt.\ haec nobilium sunt
  instituta.
}

\enabstract{
This thesis tackles methodologies to identify the genetic causes of complex diseases. This is usually done via genome-wide association studies (GWAS), when univariate association is studied, and genome-wide association interaction studies, when interactions between genetic factors are considered (GWAIS). However, both settings present some challenges, namely low statistical power, difficult interpretation, and arbitrary choices at multiple points of the study. In this thesis I study how a framework that uses biological networks can help overcome these issues and boost biomarker discovery. This is done by incorporating prior knowledge into the statistical analysis and putting every SNP and gene in relation to their biological context. By analyzing two datasets, on breast cancer and inflammatory bowel disease, I demonstrate the utility of networks to discover new mechanisms of susceptibility. These involve individual SNPs, as well as groups of SNPs in epistasis, two-way and higher. I also show how including networks in GWAS and GWAIS boosts the interpretability of the results and produces compelling biological hypotheses.
}

\frkeywords{ Caesar licentia post honoratis haec adhibens urbium
  honoratis nullum Caesar.}
\enkeywords{ gwas network epistasis machine-learning inflammatory-bowel-disease breast-cancer}


\frontmatter